<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using R for proteomics data analysis • RforProteomics</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using R for proteomics data analysis">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RforProteomics</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.19.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RforProteomics.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/RProtVis.html">Visualisation of proteomics data using R and Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="RforProteomics_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="RforProteomics_files/jquery-1.12.4/jquery.min.js"></script><link href="RforProteomics_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="RforProteomics_files/datatables-binding-0.4/datatables.js"></script><link href="RforProteomics_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="RforProteomics_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="RforProteomics_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="RforProteomics_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="RforProteomics_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Using R for proteomics data analysis</h1>
                        <h4 class="author">Laurent Gatto</h4>
                                    <h4 class="author">Lisa Breckels</h4>
                                    <h4 class="author">Vlad Petyuk</h4>
                                    <h4 class="author">Thomas Lin Pedersen</h4>
                                    <h4 class="author">Sebastian Gibb</h4>
                        
      
      
      <div class="hidden name"><code>RforProteomics.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This is the companion vignette to the ‘Visualisation of proteomics data using R and Bioconductor’ manuscript that presents an overview of R and Bioconductor software for mass spectrometry and proteomics data. It provides the code to reproduce the figures in the paper.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="sec:intro" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:intro" class="anchor"></a>Introduction</h1>
<p>This document illustrates some existing R infrastructure for the analysis of proteomics data. It presents the code for the use cases taken from <span class="citation">(Laurent Gatto and Christoforou 2013,Gatto:2015)</span>. A pre-print of <span class="citation">(Laurent Gatto and Christoforou 2013)</span> available on <a href="http://arxiv.org/abs/1305.6559">arXiv</a> and <span class="citation">(L Gatto et al. 2015)</span> is <a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201400392/full">open access</a>.</p>
<p>There are however numerous additional R resources distributed by the <a href="http://www.bioconductor.org">Bioconductor</a> and <a href="http://cran.r-project.org/web/packages/">CRAN</a> repositories, as well as packages hosted on personal websites. Section @ref(sec:packages) tries to provide a wider picture of available packages, without going into details.</p>
<div id="sec:rres" class="section level2">
<h2 class="hasAnchor">
<a href="#sec:rres" class="anchor"></a>General R resources</h2>
<p>The reader is expected to have basic R knowledge to find the document helpful. There are numerous R introductions freely available, some of which are listed below.</p>
<p>From the R project web-page:</p>
<ul>
<li>
<em>An Introduction to R</em> is based on the former <em>Notes on R</em>, gives an introduction to the language and how to use R for doing statistical analysis and graphics (<a href="http://cran.r-project.org/doc/manuals/R-intro.html">html</a> and <a href="http://cran.r-project.org/doc/manuals/R-intro.pdf">pdf</a>.</li>
<li>Several introductory tutorials in the <a href="http://cran.r-project.org/other-docs.html">contributed documentation</a> section.</li>
<li>The <a href="https://github.com/lgatto/TeachingMaterial">TeachingMaterial repository</a> contains several sets of slides and vignettes about R programming.</li>
</ul>
<p>Relevant background on the R software and its application to computational biology in general and proteomics in particular can also be found in <span class="citation">(Laurent Gatto and Christoforou 2013)</span>. For details about the Bioconductor project, the reader is referred to <span class="citation">(Gentleman et al. 2004)</span>.</p>
</div>
<div id="sec:biocres" class="section level2">
<h2 class="hasAnchor">
<a href="#sec:biocres" class="anchor"></a>Bioconductor resources</h2>
<p>The Bioconductor offers many educational resources on its <a href="http://bioconductor.org/help/">help page</a>, in addition the package’s vignettes (vignettes are a requirement for Bioconductor packages). We want to draw the attention to the Bioconductor work flows that offer a cross-package overview about a specific topic. In particular, there is now a <a href="http://bioconductor.org/help/workflows/proteomics/"><em>Mass spectrometry and proteomics data analysis</em></a> work flow.</p>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-help" class="anchor"></a>Getting help</h2>
<p>All R packages come with ample documentation. Every command (function, class or method) a user is susceptible to use is documented. The documentation can be accessed by preceding the command by a <code>?</code> in the R console. For example, to obtain help about the <code>library</code> function, that will be used in the next section, one would type <code>?library</code>. In addition, all Bioconductor packages come with at least one vignette (this document is the vignette that comes with the <em><a href="http://bioconductor.org/packages/RforProteomics">RforProteomics</a></em> package), a document that combines text and R code that is executed before the pdf is assembled. To look up all vignettes that come with a package, say <em><a href="http://bioconductor.org/packages/RforProteomics">RforProteomics</a></em> and then open the vignette of interest, one uses the <code>vignette</code> function as illustrated below. More details can be found in <code>?vignette</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## list all the vignettes in the RforProteomics package
<span class="kw">vignette</span>(<span class="dt">package =</span> <span class="st">"RforProteomics"</span>)
## Open the vignette called RforProteomics
<span class="kw">vignette</span>(<span class="st">"RforProteomics"</span>, <span class="dt">package =</span> <span class="st">"RforProteomics"</span>)
## or just
<span class="kw">vignette</span>(<span class="st">"RforProteomics"</span>)</code></pre></div>
<p>R has several <a href="http://www.r-project.org/mail.html">mailing lists</a>. The most relevant here being the main <code>R-help</code> list, <em>for discussion about problem and solutions using R</em>, ideal for general R content and is not suitable for bioinformatics or proteomics questions. Bioconductor also offers several resources dedicated to bioinformatics matters and Bioconductor packages, in particular the main <a href="https://support.bioconductor.org/">Bioconductor support forum</a> for Bioconductor-related queries.</p>
<p>It is advised to read and comply to the <a href="http://www.r-project.org/posting-guide.html">posting guides</a> (and <a href="http://bioconductor.org/help/mailing-list/posting-guide/">here</a> to maximise the chances to obtain good responses. It is important to specify the software versions using the <code>sessionInfo()</code> functions (see an example output at the end of this document. It the question involves some code, make sure to isolate the relevant portion and report it with your question, trying to make your <a href="https://github.com/hadley/devtools/wiki/Reproducibility%3E">code/example reproducible</a>.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The package should be installed using as described below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## only first time you install Bioconductor packages
<span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))
    <span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>)
## else
<span class="kw">library</span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"RforProteomics"</span>)</code></pre></div>
<p>To install all dependencies and reproduce the code in the vignette, replace the last line in the code chunk above with:)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"RforProteomics"</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Finally, the package can be loaded with</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"RforProteomics"</span>)</code></pre></div>
<pre><code>## 
## This is the 'RforProteomics' version 1.19.4.
## 
##   To get started, visit
##     http://lgatto.github.com/RforProteomics/
## 
##   or, in R, open package vignettes by typing
##     RforProteomics() # R/Bioc for proteomics overview
##     RProtVis()       # R/Bioc for proteomics visualisation
## 
##   For a full list of available documents:
##     vignette(package='RforProteomics')</code></pre>
<pre><code>## 
## Attaching package: 'RforProteomics'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     spectrum</code></pre>
<p>See also the <em><a href="http://bioconductor.org/packages/RforProteomics">RforProteomics</a></em> <a href="http://lgatto.github.io/RforProteomics/">web page</a> for more information on installation.</p>
</div>
<div id="external-dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#external-dependencies" class="anchor"></a>External dependencies</h2>
<p>Some packages used in the document depend on external libraries that need to be installed prior to the R packages:</p>
<ul>
<li>
<em><a href="http://bioconductor.org/packages/mzR">mzR</a></em> depends on the <a href="http://cdf.gsfc.nasa.gov/">Common Data Format</a> (CDF) to CDF-based raw mass-spectrometry data. On Linux, the <code>libcdf</code> library is required. On Debian-based systems, for instance, one needs to install the <code>libnetcdf-dev</code> package.</li>
<li>
<em><a href="http://bioconductor.org/packages/IPPD">IPPD</a></em> (and others) depend on the <em><a href="https://CRAN.R-project.org/package=XML">XML</a></em> package which requires the <code>libxml2</code> infrastructure on Linux. On Debian-based systems, one needs to install <code>libxml2-dev</code>.</li>
<li>
<em><a href="http://bioconductor.org/packages/biomaRt">biomaRt</a></em> performs on-line requests using the <a href="http://curl.haxx.se/"><code>curl</code></a> infrastructure. On Debian-based systems, you one needs to install <code>libcurl-dev</code> or <code>libcurl4-openssl-dev</code>.</li>
<li>
<em><a href="http://bioconductor.org/packages/MSGFplus">MSGFplus</a></em> is based on the <code>MS-GF+</code> java program and thus requires <a href="https://java.com">Java 1.7</a> in order to work.</li>
</ul>
</div>
<div id="obtaining-the-code" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-the-code" class="anchor"></a>Obtaining the code</h2>
<p>The code in this document describes all the examples presented in <span class="citation">(Laurent Gatto and Christoforou 2013)</span> and can be copy, pasted and executed. It is however more convenient to have it in a separate text file for better interaction with R to easily modify and explore it. This can be achieved with the <code>Stangle</code> function. One needs the Sweave source of this document (a document combining the narration and the R code) and the <code>Stangle</code> then specifically extracts the code chunks and produces a clean R source file. If the package is installed, the following code chunk will direct you to the <code>RforProteomics.R</code> file containing all the annotated source code contained in this document.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## gets the vignette source
rfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"doc/RforProteomics.R"</span>,
                     <span class="dt">package =</span> <span class="st">"RforProteomics"</span>)
rfile</code></pre></div>
<pre><code>## [1] ""</code></pre>
</div>
<div id="prepare-the-working-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-the-working-environment" class="anchor"></a>Prepare the working environment</h2>
<p>The packages that we will depend on to execute the examples will be loaded in the respective sections. Here, we pre-load packages that provide general functionality used throughout the document.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"RColorBrewer"</span>) ## Color palettes</code></pre></div>
<pre><code>## Warning: package 'RColorBrewer' was built under R version 3.6.0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"ggplot2"</span>)  ## Convenient and nice plotting
<span class="kw">library</span>(<span class="st">"reshape2"</span>) ## Flexibly reshape data</code></pre></div>
<pre><code>## Warning: package 'reshape2' was built under R version 3.6.0</code></pre>
</div>
</div>
<div id="data-standards-and-inputoutput" class="section level1">
<h1 class="hasAnchor">
<a href="#data-standards-and-inputoutput" class="anchor"></a>Data standards and input/output</h1>
<div id="sec:mzr" class="section level2">
<h2 class="hasAnchor">
<a href="#sec:mzr" class="anchor"></a>The mzR package</h2>
<div id="raw-ms-data" class="section level3">
<h3 class="hasAnchor">
<a href="#raw-ms-data" class="anchor"></a>Raw MS data</h3>
<p>The <em><a href="http://bioconductor.org/packages/mzR">mzR</a></em> package <span class="citation">(Chambers et al. 2012)</span> provides a unified interface to various mass spectrometry open formats. This code chunk, taken from the <code>openMSfile</code> documentation, illustrated how to open a connection to an raw data file. The example <code>mzML</code> data is taken from the <em><a href="http://bioconductor.org/packages/msdata">msdata</a></em> data package. The code below would also be applicable to an <code>mzXML</code>, <code>mzData</code> or <code>netCDF</code> file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## load the required packages
<span class="kw">library</span>(<span class="st">"mzR"</span>) ## the software package
<span class="kw">library</span>(<span class="st">"msdata"</span>) ## the data package
## below, we extract the releavant example file
## from the local 'msdata' installation
filepath &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"microtofq"</span>, <span class="dt">package =</span> <span class="st">"msdata"</span>)
file &lt;-<span class="st"> </span><span class="kw">list.files</span>(filepath, <span class="dt">pattern=</span><span class="st">"MM14.mzML"</span>,
                   <span class="dt">full.names=</span><span class="ot">TRUE</span>, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)
## creates a commection to the mzML file
mz &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/openMSfile">openMSfile</a></span>(file)
## demonstraction of data access
<span class="kw">basename</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/openMSfile">fileName</a></span>(mz))</code></pre></div>
<pre><code>## [1] "MM14.mzML"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">runInfo</a></span>(mz)</code></pre></div>
<pre><code>## $scanCount
## [1] 112
## 
## $lowMz
## [1] 0
## 
## $highMz
## [1] 0
## 
## $dStartTime
## [1] 270.334
## 
## $dEndTime
## [1] 307.678
## 
## $msLevels
## [1] 1
## 
## $startTimeStamp
## [1] NA</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">instrumentInfo</a></span>(mz)</code></pre></div>
<pre><code>## $manufacturer
## [1] "Unknown"
## 
## $model
## [1] "instrument model"
## 
## $ionisation
## [1] "electrospray ionization"
## 
## $analyzer
## [1] "mass analyzer type"
## 
## $detector
## [1] "detector type"
## 
## $software
## [1] "so_in_0 "
## 
## $sample
## [1] "MM14_20uMsa_0"
## 
## $source
## [1] ""</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## once finished, it is good to explicitely
## close the connection
<span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/mzR-class">close</a></span>(mz)</code></pre></div>
<p><em><a href="http://bioconductor.org/packages/mzR">mzR</a></em> is used by other packages, like <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> <span class="citation">(L Gatto and Lilley 2012)</span>, <em><a href="http://bioconductor.org/packages/TargetSearch">TargetSearch</a></em> <span class="citation">(Cuadros-Inostroza et al. 2009)</span> and <em><a href="http://bioconductor.org/packages/xcms">xcms</a></em> <span class="citation">(C. A. Smith et al. 2006, Benton2008, Tautenhahn2008)</span>, that provide a higher level abstraction to the data.</p>
</div>
<div id="identification-data" class="section level3">
<h3 class="hasAnchor">
<a href="#identification-data" class="anchor"></a>Identification data</h3>
<p>The <em><a href="http://bioconductor.org/packages/mzR">mzR</a></em> package also provides very fast access to <code>mzIdentML</code> data by leveraging proteowizard’s <code>C++</code> parser.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"mzid"</span>, <span class="st">"Tandem.mzid.gz"</span>, <span class="dt">package=</span><span class="st">"msdata"</span>)
mzid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/openMSfile">openIDfile</a></span>(file)
mzid</code></pre></div>
<pre><code>## Identification file handle.
## Filename:  Tandem.mzid.gz 
## Number of psms:  171</code></pre>
<p>Once and <code>mzRident</code> identification file handle has been established, various data and metadata can be extracted, as illustrated below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">softwareInfo</a></span>(mzid)</code></pre></div>
<pre><code>## [1] "xtandem x! tandem CYCLONE (2010.06.01.5) "  
## [2] "ProteoWizard MzIdentML 3.0.501 ProteoWizard"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">enzymes</a></span>(mzid)</code></pre></div>
<pre><code>##      name nTermGain cTermGain minDistance missedCleavages
## 1 Trypsin         H        OH           0               1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">psms</a></span>(mzid))</code></pre></div>
<pre><code>##  [1] "spectrumID"               "chargeState"             
##  [3] "rank"                     "passThreshold"           
##  [5] "experimentalMassToCharge" "calculatedMassToCharge"  
##  [7] "sequence"                 "modNum"                  
##  [9] "isDecoy"                  "post"                    
## [11] "pre"                      "start"                   
## [13] "end"                      "DatabaseAccess"          
## [15] "DBseqLength"              "DatabaseSeq"             
## [17] "DatabaseDescription"      "spectrum.title"          
## [19] "acquisitionNum"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/mzR/topics/metadata">psms</a></span>(mzid))[, <span class="dv">1</span><span class="op">:</span><span class="dv">13</span>]</code></pre></div>
<pre><code>##   spectrumID chargeState rank passThreshold experimentalMassToCharge
## 1   index=12           3    1         FALSE                 903.7209
## 2  index=285           3    1         FALSE                 792.3792
## 3   index=83           3    1         FALSE                 792.5295
## 4   index=21           3    1         FALSE                 850.0782
## 5  index=198           3    1         FALSE                 527.2592
## 6   index=13           2    1         FALSE                 724.8816
##   calculatedMassToCharge                 sequence modNum isDecoy post pre
## 1               903.4032  LCYIALDFDEEMKAAEDSSDIEK      2   FALSE    S   K
## 2               792.3899   KDLYGNVVLSGGTTMYEGIGER      1   FALSE    L   R
## 3               792.3899   KDLYGNVVLSGGTTMYEGIGER      1   FALSE    L   R
## 4               849.7635 VIDENFGLVEGLMTTVHAATGTQK      1   FALSE    V   K
## 5               527.2849          GVGGAIVLVLYDEMK      1   FALSE    R   R
## 6               724.3771            HAVGGRYSSLLCK      1    TRUE    D   K
##   start end
## 1   217 239
## 2   292 313
## 3   292 313
## 4   842 865
## 5   297 311
## 6   392 404</code></pre>
</div>
</div>
<div id="subsec:mzID" class="section level2">
<h2 class="hasAnchor">
<a href="#subsec:mzID" class="anchor"></a>Handling MS2 identification data with <em><a href="http://bioconductor.org/packages/mzID">mzID</a></em>
</h2>
<p>The <em><a href="http://bioconductor.org/packages/mzID">mzID</a></em> package allows to load and manipulate MS2 data in the <code>mzIdentML</code> format. The main <code>mzID</code> function reads such a file and constructs an instance of class <code>mzID</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"mzID"</span>)
mzids &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="dt">package =</span> <span class="st">'mzID'</span>),
                    <span class="dt">pattern =</span> <span class="st">'*.mzid'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)
mzids</code></pre></div>
<pre><code>## [1] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/55merge_omssa.mzid"                      
## [2] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/55merge_tandem.mzid"                     
## [3] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/Mascot_MSMS_example1.0.mzid"             
## [4] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/Mascot_MSMS_example.mzid"                
## [5] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/Mascot_NA_example.mzid"                  
## [6] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/mascot_pmf_example.mzid"                 
## [7] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/Mascot_top_down_example.mzid"            
## [8] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/MPC_example_Multiple_search_engines.mzid"
## [9] "/home/lg390/R/x86_64-pc-linux-gnu-library/3.5/mzID/extdata/Sequest_example_ver1.1.mzid"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">id &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mzID/topics/mzID">mzID</a></span>(mzids[<span class="dv">1</span>])</code></pre></div>
<pre><code>## reading 55merge_omssa.mzid... DONE!</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">id</code></pre></div>
<pre><code>## An mzID object
## 
## Software used:   OMSSA (version: NA)
## 
## Rawfile:         D:/TestSpace/NeoTestMarch2011/55merge.mgf
## 
## Database:        D:/Software/Databases/Neospora_3rndTryp/Neo_rndTryp_3times.fasta
## 
## Number of scans: 39
## Number of PSM's: 99</code></pre>
<p>Multiple files can be parsed in one go, possibly in parallel if the environment supports it. When this is done an mzIDCollection object is returned:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ids &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mzID/topics/mzID">mzID</a></span>(mzids[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>])
ids</code></pre></div>
<pre><code>## An mzIDCollection object containing 2 samples</code></pre>
<p>Peptides, scans, parameters, … can be extracted with the respective <code>peptides</code>, <code>scans</code>, <code>parameters</code>, … functions. The <code>mzID</code> object can also be converted into a <code>data.frame</code> using the <code>flatten</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/mzID/topics/flatten">flatten</a></span>(id)
<span class="kw">names</span>(fid)</code></pre></div>
<pre><code>##  [1] "spectrumid"               "spectrum title"          
##  [3] "acquisitionnum"           "passthreshold"           
##  [5] "rank"                     "calculatedmasstocharge"  
##  [7] "experimentalmasstocharge" "chargestate"             
##  [9] "omssa:evalue"             "omssa:pvalue"            
## [11] "isdecoy"                  "post"                    
## [13] "pre"                      "end"                     
## [15] "start"                    "accession"               
## [17] "length"                   "description"             
## [19] "pepseq"                   "modified"                
## [21] "modification"             "idFile"                  
## [23] "spectrumFile"             "databaseFile"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(fid)</code></pre></div>
<pre><code>## [1] 101  24</code></pre>
</div>
</div>
<div id="raw-data-abstraction-with-msnexp-objects" class="section level1">
<h1 class="hasAnchor">
<a href="#raw-data-abstraction-with-msnexp-objects" class="anchor"></a>Raw data abstraction with <code>MSnExp</code> objects</h1>
<p><em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> <span class="citation">(L Gatto and Lilley 2012)</span> provides base functions and classes for MS-based proteomics that allow facile data and meta-data processing, manipulation and plotting (see for instance figure below).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MSnbase"</span>)
## uses a simple dummy test included in the package
mzXML &lt;-<span class="st"> </span><span class="kw">dir</span>(<span class="kw">system.file</span>(<span class="dt">package=</span><span class="st">"MSnbase"</span>,<span class="dt">dir=</span><span class="st">"extdata"</span>),
             <span class="dt">full.name=</span><span class="ot">TRUE</span>,
             <span class="dt">pattern=</span><span class="st">"mzXML$"</span>)
<span class="kw">basename</span>(mzXML)</code></pre></div>
<pre><code>## [1] "dummyiTRAQ.mzXML"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## reads the raw data into and MSnExp instance
raw &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/readMSData">readMSData</a></span>(mzXML, <span class="dt">verbose =</span> <span class="ot">FALSE</span>, <span class="dt">centroided =</span> <span class="ot">TRUE</span>)
raw</code></pre></div>
<pre><code>## MSn experiment data ("MSnExp")
## Object size in memory: 0.18 Mb
## - - - Spectra data - - -
##  MS level(s): 2 
##  Number of spectra: 5 
##  MSn retention times: 25:1 - 25:2 minutes
## - - - Processing information - - -
## Data loaded: Tue Oct  2 20:43:38 2018 
##  MSnbase version: 2.7.4 
## - - - Meta data  - - -
## phenoData
##   rowNames: dummyiTRAQ.mzXML
##   varLabels: sampleNames
##   varMetadata: labelDescription
## Loaded from:
##   dummyiTRAQ.mzXML 
## protocolData: none
## featureData
##   featureNames: F1.S1 F1.S2 ... F1.S5 (5 total)
##   fvarLabels: spectrum
##   fvarMetadata: labelDescription
## experimentData: use 'experimentData(object)'</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Extract a single spectrum
raw[[<span class="dv">3</span>]]</code></pre></div>
<pre><code>## Object of class "Spectrum2"
##  Precursor: 645.3741 
##  Retention time: 25:2 
##  Charge: 2 
##  MSn level: 2 
##  Peaks count: 2125 
##  Total ion count: 150838188</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(raw, <span class="dt">full =</span> <span class="ot">TRUE</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(raw[[<span class="dv">3</span>]], <span class="dt">full =</span> <span class="ot">TRUE</span>, <span class="dt">reporters =</span> iTRAQ4)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/msnexp-1.png" alt="The `plot` method can be used on experiments, i.e. spectrum collections (top), or individual spectra (bottom)." width="700"><p class="caption">
The <code>plot</code> method can be used on experiments, i.e. spectrum collections (top), or individual spectra (bottom).
</p>
</div>
<div id="mgf-readwrite-support" class="section level2">
<h2 class="hasAnchor">
<a href="#mgf-readwrite-support" class="anchor"></a><code>mgf</code> read/write support</h2>
<p>Read and write support for data in the <a href="http://www.matrixscience.com/help/data_file_help.html#GEN"><code>mgf</code></a> and <code>mzTab</code> formats are available via the <code>readMgfData</code>/<code>writeMgfData</code> and <code>readMzTabData</code>/<code>writeMzTabData</code> functions, respectively. An example for the latter is shown in the next section.</p>
</div>
</div>
<div id="quantitative-proteomics" class="section level1">
<h1 class="hasAnchor">
<a href="#quantitative-proteomics" class="anchor"></a>Quantitative proteomics</h1>
<p>As an running example throughout this document, we will use a TMT 6-plex data set, <code>PXD000001</code> to illustrate quantitative data processing. The code chunk below first downloads this data file from the ProteomeXchange server using the <em><a href="http://bioconductor.org/packages/rpx">rpx</a></em> package.</p>
<div id="the-mztab-format" class="section level2">
<h2 class="hasAnchor">
<a href="#the-mztab-format" class="anchor"></a>The <code>mzTab</code> format</h2>
<p>The first code chunk downloads the <code>mzTab</code> data from the ProteomeXchange repository <span class="citation">(Vizcaino et al. 2014)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Experiment information
<span class="kw">library</span>(<span class="st">"rpx"</span>)
px1 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rpx/topics/PXDataset-class">PXDataset</a></span>(<span class="st">"PXD000001"</span>)
px1</code></pre></div>
<pre><code>## Object of class "PXDataset"
##  Id: PXD000001 with 12 files
##  [1] 'F063721.dat' ... [12] 'generated'
##  Use 'pxfiles(.)' to see all files.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/rpx/topics/PXDataset-class">pxfiles</a></span>(px1)</code></pre></div>
<pre><code>##  [1] "F063721.dat"                                                         
##  [2] "F063721.dat-mztab.txt"                                               
##  [3] "PRIDE_Exp_Complete_Ac_22134.xml.gz"                                  
##  [4] "PRIDE_Exp_mzData_Ac_22134.xml.gz"                                    
##  [5] "PXD000001_mztab.txt"                                                 
##  [6] "README.txt"                                                          
##  [7] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML" 
##  [8] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML"
##  [9] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML"         
## [10] "TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw"           
## [11] "erwinia_carotovora.fasta"                                            
## [12] "generated"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Downloading the mzTab data
mztab &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rpx/topics/PXDataset-class">pxget</a></span>(px1, <span class="st">"PXD000001_mztab.txt"</span>)</code></pre></div>
<pre><code>## Downloading 1 file</code></pre>
<pre><code>## /home/lg390/dev/RforProteomics/vignettes/PXD000001_mztab.txt already present.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mztab</code></pre></div>
<pre><code>## [1] "/home/lg390/dev/RforProteomics/vignettes/PXD000001_mztab.txt"</code></pre>
<p>The code below loads the <code>mzTab</code> file into R and generates an <code>MSnSet</code> instance<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a>, removes missing values and calculates protein intensities by summing the peptide quantitation data. The figure below illustrates the intensities for 5 proteins.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Load mzTab peptide data
qnt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/readMzTabData">readMzTabData</a></span>(mztab, <span class="dt">what =</span> <span class="st">"PEP"</span>, <span class="dt">version =</span> <span class="st">"0.9"</span>)</code></pre></div>
<pre><code>## Warning: Version 0.9 is deprecated. Please see '?readMzTabData' and '?
## MzTab' for details.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sampleNames</span>(qnt) &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/ReporterIons-class">reporterNames</a></span>(TMT6)
<span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(qnt))</code></pre></div>
<pre><code>##   TMT6.126 TMT6.127 TMT6.128 TMT6.129 TMT6.130 TMT6.131
## 1       NA       NA       NA       NA       NA       NA
## 2 10630132 11238708 12424917 10997763  9928972 10398534
## 3       NA       NA       NA       NA       NA       NA
## 4       NA       NA       NA       NA       NA       NA
## 5 11105690 12403253 13160903 12229367 11061660 10131218
## 6  1183431  1322371  1599088  1243715  1306602  1159064</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## remove missing values
qnt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/MSnSet-class">filterNA</a></span>(qnt)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/pSet-class">processingData</a></span>(qnt)</code></pre></div>
<pre><code>## - - - Processing information - - -
## mzTab read: Tue Oct  2 20:43:49 2018 
## Subset [2351,6][1504,6] Tue Oct  2 20:43:49 2018 
## Removed features with more than 0 NAs: Tue Oct  2 20:43:49 2018 
## Dropped featureData's levels Tue Oct  2 20:43:49 2018 
##  MSnbase version: 2.7.4</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## combine into proteins
## - using the 'accession' feature meta data
## - sum the peptide intensities
protqnt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/combineFeatures">combineFeatures</a></span>(qnt,
                           <span class="dt">groupBy =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>accession,
                           <span class="dt">fun =</span> sum)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cls &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">5</span>, <span class="st">"Set1"</span>)
<span class="kw">matplot</span>(<span class="kw">t</span>(<span class="kw">tail</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(protqnt), <span class="dt">n =</span> <span class="dv">5</span>)), <span class="dt">type =</span> <span class="st">"b"</span>,
        <span class="dt">lty =</span> <span class="dv">1</span>, <span class="dt">col =</span> cls,
        <span class="dt">ylab =</span> <span class="st">"Protein intensity (summed peptides)"</span>,
        <span class="dt">xlab =</span> <span class="st">"TMT reporters"</span>)
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="kw">tail</span>(<span class="kw">featureNames</span>(protqnt), <span class="dt">n=</span><span class="dv">5</span>),
       <span class="dt">lty =</span> <span class="dv">1</span>, <span class="dt">bty =</span> <span class="st">"n"</span>, <span class="dt">cex =</span> .<span class="dv">8</span>, <span class="dt">col =</span> cls)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/matplot-1.png" alt="Protein quantitation data." width="700"><p class="caption">
Protein quantitation data.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qntS &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/normalise-methods">normalise</a></span>(qnt, <span class="st">"sum"</span>)
qntV &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/normalise-methods">normalise</a></span>(qntS, <span class="st">"vsn"</span>)
qntV2 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/normalise-methods">normalise</a></span>(qnt, <span class="st">"vsn"</span>)

acc &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"P00489"</span>, <span class="st">"P00924"</span>,
         <span class="st">"P02769"</span>, <span class="st">"P62894"</span>,
         <span class="st">"ECA"</span>)

idx &lt;-<span class="st"> </span><span class="kw">sapply</span>(acc, grep, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)
idx2 &lt;-<span class="st"> </span><span class="kw">sapply</span>(idx, head, <span class="dv">3</span>)
small &lt;-<span class="st"> </span>qntS[<span class="kw">unlist</span>(idx2), ]

idx3 &lt;-<span class="st"> </span><span class="kw">sapply</span>(idx, head, <span class="dv">10</span>)
medium &lt;-<span class="st"> </span>qntV[<span class="kw">unlist</span>(idx3), ]

m &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(medium)
<span class="kw">colnames</span>(m) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"126"</span>, <span class="st">"127"</span>, <span class="st">"128"</span>,
                 <span class="st">"129"</span>, <span class="st">"130"</span>, <span class="st">"131"</span>)
<span class="kw">rownames</span>(m) &lt;-<span class="st"> </span><span class="kw">fData</span>(medium)<span class="op">$</span>accession
<span class="kw">rownames</span>(m)[<span class="kw">grep</span>(<span class="st">"CYC"</span>, <span class="kw">rownames</span>(m))] &lt;-<span class="st"> "CYT"</span>
<span class="kw">rownames</span>(m)[<span class="kw">grep</span>(<span class="st">"ENO"</span>, <span class="kw">rownames</span>(m))] &lt;-<span class="st"> "ENO"</span>
<span class="kw">rownames</span>(m)[<span class="kw">grep</span>(<span class="st">"ALB"</span>, <span class="kw">rownames</span>(m))] &lt;-<span class="st"> "BSA"</span>
<span class="kw">rownames</span>(m)[<span class="kw">grep</span>(<span class="st">"PYGM"</span>, <span class="kw">rownames</span>(m))] &lt;-<span class="st"> "PHO"</span>
<span class="kw">rownames</span>(m)[<span class="kw">grep</span>(<span class="st">"ECA"</span>, <span class="kw">rownames</span>(m))] &lt;-<span class="st"> "Background"</span>

cls &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/pSet-class">length</a></span>(<span class="kw">unique</span>(<span class="kw">rownames</span>(m)))<span class="op">-</span><span class="dv">1</span>, <span class="st">"Set1"</span>),
         <span class="st">"grey"</span>)
<span class="kw">names</span>(cls) &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">rownames</span>(m))
wbcol &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">"white"</span>, <span class="st">"darkblue"</span>))(<span class="dv">256</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">heatmap</span>(m, <span class="dt">col =</span> wbcol, <span class="dt">RowSideColors=</span>cls[<span class="kw">rownames</span>(m)])</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/heatmap-1.png" alt="A heatmap." width="700"><p class="caption">
A heatmap.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dfr &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(small),
                  <span class="dt">Protein =</span> <span class="kw">as.character</span>(<span class="kw">fData</span>(small)<span class="op">$</span>accession),
                  <span class="dt">Feature =</span> <span class="kw">featureNames</span>(small),
                  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="kw">colnames</span>(dfr) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"126"</span>, <span class="st">"127"</span>, <span class="st">"128"</span>, <span class="st">"129"</span>, <span class="st">"130"</span>, <span class="st">"131"</span>,
                   <span class="st">"Protein"</span>, <span class="st">"Feature"</span>)
dfr<span class="op">$</span>Protein[dfr<span class="op">$</span>Protein <span class="op">==</span><span class="st"> "sp|P00924|ENO1_YEAST"</span>] &lt;-<span class="st"> "ENO"</span>
dfr<span class="op">$</span>Protein[dfr<span class="op">$</span>Protein <span class="op">==</span><span class="st"> "sp|P62894|CYC_BOVIN"</span>]  &lt;-<span class="st"> "CYT"</span>
dfr<span class="op">$</span>Protein[dfr<span class="op">$</span>Protein <span class="op">==</span><span class="st"> "sp|P02769|ALBU_BOVIN"</span>] &lt;-<span class="st"> "BSA"</span>
dfr<span class="op">$</span>Protein[dfr<span class="op">$</span>Protein <span class="op">==</span><span class="st"> "sp|P00489|PYGM_RABIT"</span>] &lt;-<span class="st"> "PHO"</span>
dfr<span class="op">$</span>Protein[<span class="kw">grep</span>(<span class="st">"ECA"</span>, dfr<span class="op">$</span>Protein)] &lt;-<span class="st"> "Background"</span>
dfr2 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/reshape2/topics/melt">melt</a></span>(dfr)</code></pre></div>
<pre><code>## Using Protein, Feature as id variables</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggplot">ggplot</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/aes">aes</a></span>(<span class="dt">x =</span> variable, <span class="dt">y =</span> value, <span class="dt">colour =</span> Protein),
       <span class="dt">data =</span> dfr2) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_point">geom_point</a></span>() <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_path">geom_line</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/aes">aes</a></span>(<span class="dt">group=</span><span class="kw">as.factor</span>(Feature)), <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/facet_grid">facet_grid</a></span>(. <span class="op">~</span><span class="st"> </span>Protein) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme">theme</a></span>(<span class="dt">legend.position=</span><span class="st">"none"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/labs">labs</a></span>(<span class="dt">x =</span> <span class="st">"Reporters"</span>, <span class="dt">y =</span> <span class="st">"Normalised intensity"</span>)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/spikes-1.png" alt="Spikes plot using `r CRANpkg('ggplot2')`." width="700"><p class="caption">
Spikes plot using <code>r CRANpkg('ggplot2')</code>.
</p>
</div>
</div>
<div id="third-party-data" class="section level2">
<h2 class="hasAnchor">
<a href="#third-party-data" class="anchor"></a>Third-party data</h2>
<p>It is possible to import any arbitrary text-based spreadsheet as <em>MSnSet</em> object using either <code>readMSnSet</code> or <code>readMSnSet2</code>. The former takes three spreadsheets as input (for the expression data and the feature and sample meta-data). The latter uses a single spreadsheet and a vector of expression columns to populate the assay data and the feature meta-data. Detailed examples are provided in the <code>MSnbase-io</code> vignette, that can be consulted from R with <code>vignette("MSnbase-io")</code> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/MSnbase-io.html">online</a>.</p>
</div>
<div id="working-with-raw-data" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-raw-data" class="anchor"></a>Working with raw data</h2>
<p>We reuse our dedicated <code>px1</code> ProteomeXchange data object to download the raw data (in <code>mzXML</code> format) and load it with the <code>readMSData</code> from the <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> package that produces a raw data experiment object of class <code>MSnExp</code> (a new <em>on-disk</em> infrastructure is now available to access the raw data on disk on demand, rather than loading it all in memory, enabling the management of more and larger files - see the <code>benchmarking</code> vignette in the <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> package for details). The raw data is then quantified using the <code>quantify</code> method specifying the TMT 6-plex isobaric tags and a 7th peak of interest corresponding to the un-dissociated reporter tag peaks (see the <code>MSnbase-demo</code> vignette in <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> for details).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mzxml &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rpx/topics/PXDataset-class">pxget</a></span>(px1, <span class="st">"TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML"</span>)</code></pre></div>
<pre><code>## Downloading 1 file</code></pre>
<pre><code>## /home/lg390/dev/RforProteomics/vignettes/TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML already present.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rawms &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/readMSData">readMSData</a></span>(mzxml, <span class="dt">centroided =</span> <span class="ot">TRUE</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)
qntms &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/quantify-methods">quantify</a></span>(rawms, <span class="dt">reporters =</span> TMT7, <span class="dt">method =</span> <span class="st">"max"</span>)
qntms</code></pre></div>
<pre><code>## MSnSet (storageMode: lockedEnvironment)
## assayData: 6103 features, 7 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: TMT7.126 TMT7.127 ... TMT7.230 (7 total)
##   varLabels: mz reporters
##   varMetadata: labelDescription
## featureData
##   featureNames: F1.S0001 F1.S0002 ... F1.S6103 (6103 total)
##   fvarLabels: spectrum fileIdx ... collision.energy (12 total)
##   fvarMetadata: labelDescription
## experimentData: use 'experimentData(object)'
## Annotation: No annotation 
## - - - Processing information - - -
## Data loaded: Tue Oct  2 15:16:36 2018 
## TMT7 quantification by max: Tue Oct  2 15:18:42 2018 
##  MSnbase version: 2.7.4</code></pre>
<p>Identification data in the <code>mzIdentML</code> format can be added to <code>MSnExp</code> or <code>MSnSet</code> instances with the <code>addIdentificationData</code> function. See the function documentation for examples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Signal =</span> <span class="kw">rowSums</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(qntms)[, <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]),
                <span class="dt">Incomplete =</span> <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(qntms)[, <span class="dv">7</span>])
d &lt;-<span class="st"> </span><span class="kw">log</span>(d)
cls &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">"#00000050"</span>, <span class="kw">nrow</span>(qnt))
pch &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(qnt))
cls[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> "gold4"</span> ## BSA
cls[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> "dodgerblue"</span> ## ENO
cls[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> "springgreen4"</span> ## CYT
cls[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> "darkorchid2"</span> ## PHO
pch[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> </span><span class="dv">19</span>
pch[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> </span><span class="dv">19</span>
pch[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> </span><span class="dv">19</span>
pch[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)] &lt;-<span class="st"> </span><span class="dv">19</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mzp &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plotMzDelta-methods">plotMzDelta</a></span>(rawms, <span class="dt">reporters =</span> TMT6, <span class="dt">verbose =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/labs">ggtitle</a></span>(<span class="st">""</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mzp</code></pre></div>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_vline).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_text).</code></pre>
<div class="figure">
<img src="RforProteomics_files/figure-html/plotmzdelta-1.png" alt="A m/z delta plot." width="700"><p class="caption">
A m/z delta plot.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(Signal <span class="op">~</span><span class="st"> </span>Incomplete, <span class="dt">data =</span> d,
     <span class="dt">xlab =</span> <span class="kw">expression</span>(Incomplete<span class="op">~</span>dissociation),
     <span class="dt">ylab =</span> <span class="kw">expression</span>(Sum<span class="op">~</span>of<span class="op">~</span>reporters<span class="op">~</span>intensities),
     <span class="dt">pch =</span> <span class="dv">19</span>,
     <span class="dt">col =</span> <span class="st">"#4582B380"</span>)
<span class="kw">grid</span>()
<span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dt">lty =</span> <span class="st">"dotted"</span>)
<span class="kw">abline</span>(<span class="kw">lm</span>(Signal <span class="op">~</span><span class="st"> </span>Incomplete, <span class="dt">data =</span> d), <span class="dt">col =</span> <span class="st">"darkblue"</span>)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/incompl-1.png" alt="Incomplete dissociation." width="700"><p class="caption">
Incomplete dissociation.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">MAplot</span>(qnt[, <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">2</span>)], <span class="dt">cex =</span> .<span class="dv">9</span>, <span class="dt">col =</span> cls, <span class="dt">pch =</span> pch, <span class="dt">show.statistics =</span> <span class="ot">FALSE</span>)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/maplot-1.png" alt="MAplot on an `MSnSet` instance." width="700"><p class="caption">
MAplot on an <code>MSnSet</code> instance.
</p>
</div>
</div>
<div id="the-maldiquant-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-maldiquant-package" class="anchor"></a>The <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> package</h2>
<p>This section illustrates some of <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em>’s data processing capabilities <span class="citation">(Gibb and Strimmer 2012)</span>. The code is taken from the <code>processing-peaks.R</code> script downloaded from the <a href="http://strimmerlab.org/software/maldiquant/">package homepage</a>.</p>
<div id="loading-the-data" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#loading-the-data" class="anchor"></a>Loading the data</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## load packages
<span class="kw">library</span>(<span class="st">"MALDIquant"</span>)
<span class="kw">library</span>(<span class="st">"MALDIquantForeign"</span>)
## getting test data
datapath &lt;-
<span class="st">  </span><span class="kw">file.path</span>(<span class="kw">system.file</span>(<span class="st">"Examples"</span>,
                        <span class="dt">package =</span> <span class="st">"readBrukerFlexData"</span>),
            <span class="st">"2010_05_19_Gibb_C8_A1"</span>)
<span class="kw">dir</span>(datapath)</code></pre></div>
<pre><code>## [1] "0_A1" "0_A2"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sA1 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquantForeign/topics/importBrukerFlex-functions">importBrukerFlex</a></span>(datapath, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co"># in the following we use only the first spectrum</span>
s &lt;-<span class="st"> </span>sA1[[<span class="dv">1</span>]]

<span class="kw">summary</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/AbstractMassObject-class">mass</a></span>(s))</code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   999.9  2373.3  4331.4  4721.3  6874.2 10001.9</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/AbstractMassObject-class">intensity</a></span>(s))</code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##       4     180    1562    2841    4656   32594</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">as.matrix</span>(s))</code></pre></div>
<pre><code>##           mass intensity
## [1,]  999.9388     11278
## [2,] 1000.1316     11350
## [3,] 1000.3244     10879
## [4,] 1000.5173     10684
## [5,] 1000.7101     10740
## [6,] 1000.9030     10947</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/mqplot1-1.png" alt="Spectrum plotting in `r CRANpkg('MALDIquant')`." width="700"><p class="caption">
Spectrum plotting in <code>r CRANpkg('MALDIquant')</code>.
</p>
</div>
</div>
<div id="preprocessing" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>{Preprocessing}</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## sqrt transform (for variance stabilization)
s2 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/transformIntensity-methods">transformIntensity</a></span>(s, <span class="dt">method=</span><span class="st">"sqrt"</span>)
s2</code></pre></div>
<pre><code>## S4 class type            : MassSpectrum            
## Number of m/z values     : 22431                   
## Range of m/z values      : 999.939 - 10001.925     
## Range of intensity values: 2e+00 - 1.805e+02       
## Memory usage             : 361.133 KiB             
## Name                     : 2010_05_19_Gibb_C8_A1.A1
## File                     : /home/lg390/R/x86_64-pc-linux-gnu-library/3.5/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## smoothing - 5 point moving average
s3 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/smoothIntensity-methods">smoothIntensity</a></span>(s2, <span class="dt">method=</span><span class="st">"MovingAverage"</span>, <span class="dt">halfWindowSize=</span><span class="dv">2</span>)
s3</code></pre></div>
<pre><code>## S4 class type            : MassSpectrum            
## Number of m/z values     : 22431                   
## Range of m/z values      : 999.939 - 10001.925     
## Range of intensity values: 3.606e+00 - 1.792e+02   
## Memory usage             : 361.133 KiB             
## Name                     : 2010_05_19_Gibb_C8_A1.A1
## File                     : /home/lg390/R/x86_64-pc-linux-gnu-library/3.5/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## baseline subtraction
s4 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/removeBaseline-methods">removeBaseline</a></span>(s3, <span class="dt">method=</span><span class="st">"SNIP"</span>)
s4</code></pre></div>
<pre><code>## S4 class type            : MassSpectrum            
## Number of m/z values     : 22431                   
## Range of m/z values      : 999.939 - 10001.925     
## Range of intensity values: 0e+00 - 1.404e+02       
## Memory usage             : 361.133 KiB             
## Name                     : 2010_05_19_Gibb_C8_A1.A1
## File                     : /home/lg390/R/x86_64-pc-linux-gnu-library/3.5/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</code></pre>
</div>
<div id="peak-picking" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#peak-picking" class="anchor"></a>Peak picking</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## peak picking
p &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/detectPeaks-methods">detectPeaks</a></span>(s4)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/pSet-class">length</a></span>(p) <span class="co"># 181</span></code></pre></div>
<pre><code>## [1] 186</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peak.data &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(p) <span class="co"># extract peak information</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))
xl &lt;-<span class="st"> </span><span class="kw">range</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/AbstractMassObject-class">mass</a></span>(s))
<span class="co"># use same xlim on all plots for better comparison</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"1: raw"</span>, <span class="dt">xlim=</span>xl)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s2, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"2: variance stabilisation"</span>, <span class="dt">xlim=</span>xl)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s3, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"3: smoothing"</span>, <span class="dt">xlim=</span>xl)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s4, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"4: base line correction"</span>, <span class="dt">xlim=</span>xl)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(s4, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"5: peak detection"</span>, <span class="dt">xlim=</span>xl)
<span class="kw">points</span>(p)
top20 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/AbstractMassObject-class">intensity</a></span>(p) <span class="op">%in%</span><span class="st"> </span><span class="kw">sort</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/AbstractMassObject-class">intensity</a></span>(p), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>]
<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/labelPeaks-methods">labelPeaks</a></span>(p, <span class="dt">index=</span>top20, <span class="dt">underline=</span><span class="ot">TRUE</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(p, <span class="dt">sub=</span><span class="st">""</span>, <span class="dt">main=</span><span class="st">"6: peak plot"</span>, <span class="dt">xlim=</span>xl)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MALDIquant/topics/labelPeaks-methods">labelPeaks</a></span>(p, <span class="dt">index=</span>top20, <span class="dt">underline=</span><span class="ot">TRUE</span>)</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/mqplot-1.png" alt="Spectrum plotting in `r CRANpkg('MALDIquant')`." width="700"><p class="caption">
Spectrum plotting in <code>r CRANpkg('MALDIquant')</code>.
</p>
</div>
</div>
</div>
<div id="working-with-peptide-sequences" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-peptide-sequences" class="anchor"></a>Working with peptide sequences</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(IPPD)
<span class="kw">library</span>(BRAIN)
atoms &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/BRAIN/topics/getAtomsFromSeq">getAtomsFromSeq</a></span>(<span class="st">"SIVPSGASTGVHEALEMR"</span>)
<span class="kw">unlist</span>(atoms)</code></pre></div>
<pre><code>##   C   H   N   O   S 
##  77 129  23  27   1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Rdisop)
pepmol &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/Rdisop/topics/getMolecule">getMolecule</a></span>(<span class="kw">paste0</span>(<span class="kw">names</span>(atoms),
                             <span class="kw">unlist</span>(atoms),
                             <span class="dt">collapse =</span> <span class="st">""</span>))
pepmol</code></pre></div>
<pre><code>## $formula
## [1] "C77H129N23O27S"
## 
## $score
## [1] 1
## 
## $exactmass
## [1] 1839.915
## 
## $charge
## [1] 0
## 
## $parity
## [1] "e"
## 
## $valid
## [1] "Valid"
## 
## $DBE
## [1] 25
## 
## $isotopes
## $isotopes[[1]]
##              [,1]         [,2]         [,3]         [,4]         [,5]
## [1,] 1839.9148973 1840.9177412 1841.9196777 1.842921e+03 1.843923e+03
## [2,]    0.3427348    0.3353456    0.1960976 8.474135e-02 2.952833e-02
##              [,6]         [,7]         [,8]         [,9]        [,10]
## [1,] 1.844925e+03 1.845927e+03 1.846928e+03 1.847930e+03 1.848932e+03
## [2,] 8.691735e-03 2.226358e-03 5.066488e-04 1.040196e-04 1.949686e-05</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">##
<span class="kw">library</span>(OrgMassSpecR)</code></pre></div>
<pre><code>## Loading required package: grid</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(itraqdata)

simplottest &lt;-
<span class="st">  </span>itraqdata[<span class="kw">featureNames</span>(itraqdata) <span class="op">%in%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">"X"</span>, <span class="dv">46</span><span class="op">:</span><span class="dv">47</span>)]
sim &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/OrgMassSpecR/topics/SpectrumSimilarity">SpectrumSimilarity</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(simplottest[[<span class="dv">1</span>]], <span class="st">"data.frame"</span>),
                          <span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(simplottest[[<span class="dv">2</span>]], <span class="st">"data.frame"</span>),
                          <span class="dt">top.lab =</span> <span class="st">"itraqdata[['X46']]"</span>,
                          <span class="dt">bottom.lab =</span> <span class="st">"itraqdata[['X47']]"</span>,
                          <span class="dt">b =</span> <span class="dv">25</span>)
<span class="kw">title</span>(<span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">"Spectrum similarity"</span>, <span class="kw">round</span>(sim, <span class="dv">3</span>)))</code></pre></div>
<p><img src="RforProteomics_files/figure-html/isotopes-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/OrgMassSpecR/topics/MonoisotopicMass">MonoisotopicMass</a></span>(<span class="dt">formula =</span> <span class="kw">list</span>(<span class="dt">C =</span> <span class="dv">2</span>, <span class="dt">O =</span> <span class="dv">1</span>, <span class="dt">H=</span><span class="dv">6</span>))</code></pre></div>
<pre><code>## [1] 46.04186</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">molecule &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/Rdisop/topics/getMolecule">getMolecule</a></span>(<span class="st">"C2H5OH"</span>)
molecule<span class="op">$</span>exactmass</code></pre></div>
<pre><code>## [1] 46.04186</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## x11()
## plot(t(.pepmol$isotopes[[1]]), type = "h")

## x &lt;- IsotopicDistribution(formula = list(C = 2, O = 1, H=6))
## t(molecule$isotopes[[1]])
## par(mfrow = c(2,1))
## plot(t(molecule$isotopes[[1]]), type = "h")
## plot(x[, c(1,3)], type = "h")

## data(myo500)
## masses &lt;- c(147.053, 148.056)
## intensities &lt;- c(93, 5.8)
## molecules &lt;- decomposeIsotopes(masses, intensities)

## experimental eno peptides
exppep &lt;-
<span class="st">  </span><span class="kw">as.character</span>(<span class="kw">fData</span>(qnt[<span class="kw">grep</span>(<span class="st">"ENO"</span>, <span class="kw">fData</span>(qnt)[, <span class="dv">2</span>]), ])[, <span class="dv">1</span>]) ## 13
minlength &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">nchar</span>(exppep))


<span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(<span class="st">"P00924.fasta"</span>))
    eno &lt;-<span class="st"> </span><span class="kw">download.file</span>(<span class="st">"http://www.uniprot.org/uniprot/P00924.fasta"</span>,
                         <span class="dt">destfile =</span> <span class="st">"P00924.fasta"</span>)
eno &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">readLines</span>(<span class="st">"P00924.fasta"</span>)[<span class="op">-</span><span class="dv">1</span>], <span class="dt">collapse =</span> <span class="st">""</span>)
enopep &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/OrgMassSpecR/topics/Digest">Digest</a></span>(eno, <span class="dt">missed =</span> <span class="dv">1</span>)
<span class="kw">nrow</span>(enopep) ## 103</code></pre></div>
<pre><code>## [1] 103</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(<span class="kw">nchar</span>(enopep<span class="op">$</span>peptide) <span class="op">&gt;=</span><span class="st"> </span>minlength) ## 68</code></pre></div>
<pre><code>## [1] 0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pepcnt &lt;-<span class="st"> </span>enopep[enopep[, <span class="dv">1</span>] <span class="op">%in%</span><span class="st"> </span>exppep, ]
<span class="kw">nrow</span>(pepcnt) ## 13</code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>The following code chunks demonstrate how to use the <em><a href="http://bioconductor.org/packages/cleaver">cleaver</a></em> package for in-silico cleavage of polypeptides, e.g. cleaving of <em>Gastric juice peptide 1 (P01358)</em> using <em>Trypsin</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cleaver)
<span class="kw"><a href="http://www.rdocumentation.org/packages/cleaver/topics/cleave-methods">cleave</a></span>(<span class="st">"LAAGKVEDSD"</span>, <span class="dt">enzym =</span> <span class="st">"trypsin"</span>)</code></pre></div>
<pre><code>## $LAAGKVEDSD
## [1] "LAAGK" "VEDSD"</code></pre>
<p>Sometimes cleavage is not perfect and the enzym miss some cleavage positions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## miss one cleavage position
<span class="kw"><a href="http://www.rdocumentation.org/packages/cleaver/topics/cleave-methods">cleave</a></span>(<span class="st">"LAAGKVEDSD"</span>, <span class="dt">enzym =</span> <span class="st">"trypsin"</span>, <span class="dt">missedCleavages =</span> <span class="dv">1</span>)</code></pre></div>
<pre><code>## $LAAGKVEDSD
## [1] "LAAGKVEDSD"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## miss zero or one cleavage positions
<span class="kw"><a href="http://www.rdocumentation.org/packages/cleaver/topics/cleave-methods">cleave</a></span>(<span class="st">"LAAGKVEDSD"</span>, <span class="dt">enzym =</span> <span class="st">"trypsin"</span>, <span class="dt">missedCleavages =</span> <span class="dv">0</span><span class="op">:</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## $LAAGKVEDSD
## [1] "LAAGK"      "VEDSD"      "LAAGKVEDSD"</code></pre>
<p>Example code to generate an Texshade image to be included directly in a Latex document or R vignette is presented below. The R code generates a Texshade environment and the annotated sequence display code that is written to a TeX file that can itself be included into a LaTeX or Sweave document.</p>
<pre><code>seq1file &lt;- "seq1.tex"
cat("\\begin{texshade}{Figures/P00924.fasta}
     \\setsize{numbering}{footnotesize}
     \\setsize{residues}{footnotesize}
     \\residuesperline*{70}
     \\shadingmode{functional}
     \\hideconsensus
     \\vsepspace{1mm}
     \\hidenames
     \\noblockskip\n", file = seq1file)
tmp &lt;- sapply(1:nrow(pepcnt), function(i) {
  col &lt;- ifelse((i %% 2) == 0, "Blue", "RoyalBlue")
  cat("\\shaderegion{1}{", pepcnt$start[i], "..", pepcnt$stop[i], "}{White}{", col, "}\n",
      file = seq1file, append = TRUE)
})
cat("\\end{texshade}
    \\caption{Visualising observed peptides for the Yeast enolase protein. Peptides are shaded in blue and black.
              The last peptide is a mis-cleavage and overlaps with \`IEEELGDNAVFAGENFHHGDK`.}
    \ {#fig:seq}
  \\end{center}
\\end{figure}\n\n",
    file = seq1file, append = TRUE)</code></pre>
<div id="n15-incorporation" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#n15-incorporation" class="anchor"></a>N15 incorporation</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## 15N incorporation rates from 0, 0.1, ..., 0.9, 0.95, 1
incrate &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">seq</span>(<span class="dv">0</span>, <span class="fl">0.9</span>, <span class="fl">0.1</span>), <span class="fl">0.95</span>, <span class="dv">1</span>)
inc &lt;-<span class="st"> </span><span class="kw">lapply</span>(incrate, <span class="cf">function</span>(inc)
              <span class="kw"><a href="http://www.rdocumentation.org/packages/OrgMassSpecR/topics/IsotopicDistributionN">IsotopicDistributionN</a></span>(<span class="st">"YEVQGEVFTKPQLWP"</span>, inc))
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">3</span>))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/pSet-class">length</a></span>(inc))
  <span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/plot-methods">plot</a></span>(inc[[i]][, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)], <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">1823</span>, <span class="dv">1848</span>), <span class="dt">type =</span> <span class="st">"h"</span>,
       <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">"15N incorporation at "</span>, incrate[i]<span class="op">*</span><span class="dv">100</span>, <span class="st">"%"</span>))</code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/n15-1.png" alt="Isotopic envelope for the `YEVQGEVFTKPQLWP` peptide at different N15 incorporation rates " width="700"><p class="caption">
Isotopic envelope for the <code>YEVQGEVFTKPQLWP</code> peptide at different N15 incorporation rates
</p>
</div>
</div>
</div>
<div id="the-isobar-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-isobar-package" class="anchor"></a>The <em><a href="http://bioconductor.org/packages/isobar">isobar</a></em> package</h2>
<p>The <em><a href="http://bioconductor.org/packages/isobar">isobar</a></em> package <span class="citation">(Breitwieser et al. 2011)</span> provides methods for the statistical analysis of isobarically tagged MS2 experiments.</p>
<div id="note" class="section level4">
<h4 class="hasAnchor">
<a href="#note" class="anchor"></a>Note</h4>
<p>The <code>isobar</code> code chunks were failing in the context of the vignette and aren’t being evaludated at the moment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"isobar"</span>)

## Prepare the PXD000001 data for isobar analysis
.ions &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(qnt)
.mass &lt;-<span class="st"> </span><span class="kw">matrix</span>(TMT6<span class="op">@</span>mz, <span class="kw">nrow</span>(qnt), <span class="dt">byrow=</span><span class="ot">TRUE</span>, <span class="dt">ncol =</span> <span class="dv">6</span>)
<span class="kw">colnames</span>(.ions) &lt;-<span class="st"> </span><span class="kw">colnames</span>(.mass) &lt;-
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/IBSpectra-class">reporterTagNames</a></span>(<span class="kw">new</span>(<span class="st">"TMT6plexSpectra"</span>))
<span class="kw">rownames</span>(.ions) &lt;-<span class="st"> </span><span class="kw">rownames</span>(.mass) &lt;-
<span class="st">  </span><span class="kw">paste</span>(<span class="kw">fData</span>(qnt)<span class="op">$</span>accession, <span class="kw">fData</span>(qnt)<span class="op">$</span>sequence, <span class="dt">sep =</span> <span class="st">"."</span>)
pgtbl &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">spectrum =</span> <span class="kw">rownames</span>(.ions),
                    <span class="dt">peptide =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>sequence,
                    <span class="dt">modif =</span> <span class="st">":"</span>,
                    <span class="dt">start.pos =</span> <span class="dv">1</span>,
                    <span class="dt">protein =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>accession,
                    <span class="dt">accession =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>accession)
x &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">"TMT6plexSpectra"</span>, pgtbl, .ions, .mass)
<span class="kw">featureData</span>(x)<span class="op">$</span>proteins &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">fData</span>(qnt)<span class="op">$</span>accession)

x &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/isobar-preprocessing">correctIsotopeImpurities</a></span>(x) ## using identity matrix here
x &lt;-<span class="st"> </span>isobar<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/isobar-preprocessing">normalize</a></span>(x, <span class="dt">per.file =</span> <span class="ot">FALSE</span>)
## spikes
spks &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/ProteinGroup-class">protein.g</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/IBSpectra-class">proteinGroup</a></span>(x), <span class="st">"P00489"</span>),
          <span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/ProteinGroup-class">protein.g</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/IBSpectra-class">proteinGroup</a></span>(x), <span class="st">"P00924"</span>),
          <span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/ProteinGroup-class">protein.g</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/IBSpectra-class">proteinGroup</a></span>(x), <span class="st">"P02769"</span>),
          <span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/ProteinGroup-class">protein.g</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/IBSpectra-class">proteinGroup</a></span>(x), <span class="st">"P62894"</span>))

cls2 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">"#00000040"</span>, <span class="kw">nrow</span>(x))
pch2 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(x))
cls2[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> "gold4"</span> ## BSA
cls2[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> "dodgerblue"</span> ## ENO
cls2[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> "springgreen4"</span> ## CYT
cls2[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> "darkorchid2"</span> ## PHO
pch2[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch2[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch2[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch2[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">featureNames</span>(x))] &lt;-<span class="st"> </span><span class="dv">19</span>

nm &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/NoiseModel-class">NoiseModel</a></span>(x)
ib.background &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/subsetIBSpectra">subsetIBSpectra</a></span>(x, <span class="dt">protein=</span>spks,
                                 <span class="dt">direction =</span> <span class="st">"exclude"</span>)
nm.background &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/NoiseModel-class">NoiseModel</a></span>(ib.background)
ib.spks &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/subsetIBSpectra">subsetIBSpectra</a></span>(x, <span class="dt">protein =</span> spks,
                           <span class="dt">direction=</span><span class="st">"include"</span>,
                           <span class="dt">specificity=</span><span class="st">"reporter-specific"</span>)
nm.spks &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/NoiseModel-class">NoiseModel</a></span>(ib.spks, <span class="dt">one.to.one=</span><span class="ot">FALSE</span>, <span class="dt">pool=</span><span class="ot">TRUE</span>)

ratios &lt;-<span class="st"> </span><span class="dv">10</span><span class="op">^</span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/isobar-analysis">estimateRatio</a></span>(x, nm,
                           <span class="dt">channel1=</span><span class="st">"127"</span>, <span class="dt">channel2=</span><span class="st">"129"</span>,
                           <span class="dt">protein =</span> spks,
                           <span class="dt">combine =</span> <span class="ot">FALSE</span>)[, <span class="st">"lratio"</span>]

res &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/isobar-analysis">estimateRatio</a></span>(x, nm,
                     <span class="dt">channel1=</span><span class="st">"127"</span>, <span class="dt">channel2=</span><span class="st">"129"</span>,
                     <span class="dt">protein =</span> <span class="kw">unique</span>(<span class="kw">fData</span>(x)<span class="op">$</span>proteins),
                     <span class="dt">combine =</span> <span class="ot">FALSE</span>,
                     <span class="dt">sign.level =</span> <span class="fl">0.01</span>)[, <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">8</span>)]
res &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(res)
res<span class="op">$</span>lratio &lt;-<span class="st"> </span><span class="op">-</span>(res<span class="op">$</span>lratio)

cls3 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">"#00000050"</span>, <span class="kw">nrow</span>(res))
pch3 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">nrow</span>(res))
cls3[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> "gold4"</span> ## BSA
cls3[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> "dodgerblue"</span> ## ENO
cls3[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> "springgreen4"</span> ## CYT
cls3[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> "darkorchid2"</span> ## PHO
pch3[<span class="kw">grep</span>(<span class="st">"P02769"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch3[<span class="kw">grep</span>(<span class="st">"P00924"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch3[<span class="kw">grep</span>(<span class="st">"P62894"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> </span><span class="dv">19</span>
pch3[<span class="kw">grep</span>(<span class="st">"P00489"</span>, <span class="kw">rownames</span>(res))] &lt;-<span class="st"> </span><span class="dv">19</span>

rat.exp &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">PHO =</span> <span class="dv">2</span><span class="op">/</span><span class="dv">2</span>,
             <span class="dt">ENO =</span> <span class="dv">5</span><span class="op">/</span><span class="dv">1</span>,
             <span class="dt">BSA =</span> <span class="fl">2.5</span><span class="op">/</span><span class="dv">10</span>,
             <span class="dt">CYT =</span> <span class="dv">1</span><span class="op">/</span><span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/isobar/topics/isobar-plots">maplot</a></span>(x,
       <span class="dt">noise.model =</span> <span class="kw">c</span>(nm.background, nm.spks, nm),
       <span class="dt">channel1=</span><span class="st">"127"</span>, <span class="dt">channel2=</span><span class="st">"129"</span>,
       <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">col =</span> cls2,
       <span class="dt">main =</span> <span class="st">"Spectra MA plot"</span>)
<span class="kw">abline</span>(<span class="dt">h =</span> <span class="dv">1</span>, <span class="dt">lty =</span> <span class="st">"dashed"</span>, <span class="dt">col =</span> <span class="st">"grey"</span>)
<span class="kw">legend</span>(<span class="st">"topright"</span>,
       <span class="kw">c</span>(<span class="st">"BSA"</span>, <span class="st">"ENO"</span>, <span class="st">"CYT"</span>, <span class="st">"PHO"</span>),
       <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">"gold4"</span>, <span class="st">"dodgerblue"</span>,
                   <span class="st">"springgreen4"</span>, <span class="st">"darkorchid2"</span>),
       <span class="dt">bty =</span> <span class="st">"n"</span>, <span class="dt">cex =</span> .<span class="dv">7</span>)</code></pre></div>
<div class="figure">
<img src="figures/isobar-fig.png" alt="Result from the isobar pipeline."><p class="caption">Result from the isobar pipeline.</p>
</div>
</div>
</div>
<div id="the-dep-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-dep-package" class="anchor"></a>The <em><a href="http://bioconductor.org/packages/DEP">DEP</a></em> package</h2>
<p>The <em><a href="http://bioconductor.org/packages/DEP">DEP</a></em> package supports analysis of label-free and TMT pipelines using, as described in its vignette. These can be used with <code>MSnSet</code> objects by converting them to/from <code>SummarizedExperiment</code> objects:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(msnset)
se &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(msnset, <span class="st">"SummarizedExperiment"</span>)</code></pre></div>
<pre><code>## Loading required namespace: SummarizedExperiment</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se</code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 55 4 
## metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion
## assays(1): ''
## rownames(55): X1 X10 ... X8 X9
## rowData names(15): spectrum ProteinAccession ...
##   acquisition.number collision.energy
## colnames(4): iTRAQ4.114 iTRAQ4.115 iTRAQ4.116 iTRAQ4.117
## colData names(2): mz reporters</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ms &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(se, <span class="st">"MSnSet"</span>)
ms</code></pre></div>
<pre><code>## MSnSet (storageMode: lockedEnvironment)
## assayData: 55 features, 4 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: iTRAQ4.114 iTRAQ4.115 iTRAQ4.116 iTRAQ4.117
##   varLabels: mz reporters
##   varMetadata: labelDescription
## featureData
##   featureNames: X1 X10 ... X9 (55 total)
##   fvarLabels: spectrum ProteinAccession ... collision.energy (15
##     total)
##   fvarMetadata: labelDescription
## experimentData: use 'experimentData(object)'
## Annotation:  
## - - - Processing information - - -
##  MSnbase version: 2.7.4</code></pre>
</div>
<div id="the-synapter-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-synapter-package" class="anchor"></a>The <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> package</h2>
<p>The <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> <span class="citation">(Bond et al. 2013)</span> package comes with a detailed vignette that describes how to prepare the MSE data and then process it in R. Several interfaces are available provided the user with maximum control, easy batch processing capabilities or a graphical user interface. The conversion into <code>MSnSet</code> instances and filter and combination thereof as well as statistical analysis are also described.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## open the synapter vignette
<span class="kw">library</span>(<span class="st">"synapter"</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/synapter/topics/synapterGuide">synapterGuide</a></span>()</code></pre></div>
</div>
</div>
<div id="ms2-spectra-identification" class="section level1">
<h1 class="hasAnchor">
<a href="#ms2-spectra-identification" class="anchor"></a>MS2 spectra identification</h1>
<p>At the moment two packages allow the user to run peptide identifications from within R. Each of the packages interface to an external peptide database search tool and have more or less the same workflow, though their syntax differs:</p>
<ul>
<li>Prepare the input data.</li>
<li>Run the search.</li>
<li>Import the search results and extract the peptides and proteins</li>
</ul>
<div id="x-tandem" class="section level2">
<h2 class="hasAnchor">
<a href="#x-tandem" class="anchor"></a>X! Tandem</h2>
<p>Following Bioconductor 2.12 the <em><a href="http://bioconductor.org/packages/rTANDEM">rTANDEM</a></em> package provides the means to run the popular X!Tandem software <span class="citation">(Craig and Beavis 2004)</span>.</p>
<p>Using example code/data from the <em><a href="http://bioconductor.org/packages/rTANDEM">rTANDEM</a></em> vignette/package, the following is an example of a typical workflow</p>
<div id="preparation-of-the-input-data" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#preparation-of-the-input-data" class="anchor"></a>Preparation of the input data</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(rTANDEM)
taxonomy &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">rTTaxo</a></span>(<span class="dt">taxon =</span> <span class="st">"yeast"</span>,
                   <span class="dt">format =</span> <span class="st">"peptide"</span>,
                   <span class="dt">URL =</span> <span class="kw">system.file</span>(
                     <span class="st">"extdata/fasta/scd.fasta.pro"</span>,
                     <span class="dt">package=</span><span class="st">"rTANDEM"</span>))
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">rTParam</a></span>()
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param,
                       <span class="st">'protein'</span>, <span class="st">'taxon'</span>,
                       <span class="dt">value=</span><span class="st">"yeast"</span>)
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param, <span class="st">'list path'</span>,
                       <span class="st">'taxonomy information'</span>, taxonomy)
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param,
                       <span class="st">'list path'</span>, <span class="st">'default parameters'</span>,
                       <span class="dt">value =</span> <span class="kw">system.file</span>(
                         <span class="st">"extdata/default_input.xml"</span>,
                         <span class="dt">package=</span><span class="st">"rTANDEM"</span>))
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param, <span class="st">'spectrum'</span>, <span class="st">'path'</span>,
                       <span class="dt">value =</span> <span class="kw">system.file</span>(
                         <span class="st">"extdata/test_spectra.mgf"</span>,
                         <span class="dt">package=</span><span class="st">"rTANDEM"</span>))
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param, <span class="st">'output'</span>, <span class="st">'xsl path'</span>,
                       <span class="dt">value =</span> <span class="kw">system.file</span>(
                         <span class="st">"extdata/tandem-input-style.xsl"</span>,
                         <span class="dt">package=</span><span class="st">"rTANDEM"</span>))
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/parameters">setParamValue</a></span>(param, <span class="st">'output'</span>, <span class="st">'path'</span>,
                       <span class="dt">value =</span> <span class="kw">paste</span>(<span class="kw">getwd</span>(),
                         <span class="st">"output.xml"</span>, <span class="dt">sep=</span><span class="st">"/"</span>))</code></pre></div>
</div>
<div id="performing-the-search" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#performing-the-search" class="anchor"></a>Performing the search</h3>
<p>The analysis is run using the <code>tandem</code> function (see also the <code>rtandem</code> function), which returns the results data file path (only the file name is displayed below).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">resultPath &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/tandem">tandem</a></span>(param)</code></pre></div>
<pre><code>## Loading spectra
##  (mgf). loaded.
## Spectra matching criteria = 242
## Starting threads . started.
## Computing models:
##  testin
##      sequences modelled = 5 ks
## Model refinement:
##  partial cleavage ..... done.
##  unanticipated cleavage ..... done.
##  modified N-terminus ..... done.
##  finishing refinement ... done.
## Creating report:
##  initial calculations  ..... done.
##  sorting  ..... done.
##  finding repeats ..... done.
##  evaluating results ..... done.
##  calculating expectations ..... done.
##  writing results ..... done.
## 
## Valid models = 40
## Unique models = 41
## Estimated false positives = 1 +/- 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">basename</span>(resultPath)</code></pre></div>
<pre><code>## [1] "output.2018_10_02_20_44_09.t.xml"</code></pre>
</div>
<div id="import-and-analyse-results" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#import-and-analyse-results" class="anchor"></a>Import and analyse results</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/converters">GetResultsFromXML</a></span>(resultPath)
## the inferred proteins
proteins &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/accessors">GetProteins</a></span>(res,
                        <span class="dt">log.expect =</span> <span class="op">-</span><span class="fl">1.3</span>,
                        <span class="dt">min.peptides =</span> <span class="dv">2</span>)
proteins[, <span class="op">-</span>(<span class="dv">4</span><span class="op">:</span><span class="dv">5</span>), with =<span class="st"> </span><span class="ot">FALSE</span>]</code></pre></div>
<pre><code>##      uid expect.value   label description num.peptides
##  1:  576        -27.2 YCR012W     YCR012W            5
##  2: 1811        -14.5 YFR053C     YFR053C            3
##  3: 2301        -12.8 YGR254W     YGR254W            3
##  4:    4        -12.0 YAL005C     YAL005C            3
##  5: 3517        -12.0 YLL024C     YLL024C            3
##  6: 3328        -10.3 YKL152C     YKL152C            2
##  7: 3386        -10.1 YKL216W     YKL216W            2
##  8: 2281         -7.9 YGR234W     YGR234W            2
##  9: 2568         -7.5 YHR174W     YHR174W            2
## 10: 2044         -7.1 YGL253W     YGL253W            2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## the identified peptides for YFR053C
peptides &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/rTANDEM/topics/accessors">GetPeptides</a></span>(<span class="dt">protein.uid =</span> <span class="dv">1811</span>,
                        <span class="dt">results =</span> res,
                        <span class="dt">expect =</span> <span class="fl">0.05</span>)
peptides[, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">9</span>, <span class="dv">10</span><span class="op">:</span><span class="dv">16</span>), with =<span class="st"> </span><span class="ot">FALSE</span>]</code></pre></div>
<pre><code>##     pep.id prot.uid spectrum.id spectrum.mh expect.value tandem.score
## 1: 102.1.1     1811         102    942.5147      0.00660         31.9
## 2: 250.1.1     1811         250   1212.5610      0.00043         35.0
## 3:  60.1.1     1811          60    863.4933      0.00870         21.7
##           mh   delta peak.count missed.cleavages start.position
## 1:  942.5370 -0.0220         NA                0            166
## 2: 1212.5531  0.0079         NA                0            437
## 3:  863.4985 -0.0052         NA                0            309
##    end.position
## 1:          173
## 2:          447
## 3:          315</code></pre>
<p>More details are provided in the vignette available with <code>vignette("rTANDEM")</code>, for instance the extraction of degenerated peptides, i.e. peptides found in multiple proteins.</p>
<p>The <em><a href="http://bioconductor.org/packages/shinyTANDEM">shinyTANDEM</a></em> package offers a web-based graphical interface to <em><a href="http://bioconductor.org/packages/rTANDEM">rTANDEM</a></em>.</p>
</div>
</div>
<div id="ms-gf" class="section level2">
<h2 class="hasAnchor">
<a href="#ms-gf" class="anchor"></a>MS-GF+</h2>
<p>With the release of Bioconductor 3.0 the <em><a href="http://bioconductor.org/packages/MSGFplus">MSGFplus</a></em> package has provided an interface to MS-GF+ <span class="citation">(Kim, Gupta, and Pevzner 2008, Kim:2010ks)</span>. The package vignette describe in detail the different ways an MS-GF+ analysis can be initiated and only a simple example will be given here:</p>
<div id="preparation-of-the-input-data-1" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#preparation-of-the-input-data-1" class="anchor"></a>Preparation of the input data</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MSGFplus"</span>)
## Create a parameter object with a set of parameters
param &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/msgfPar">msgfPar</a></span>(<span class="dt">database =</span> <span class="kw">system.file</span>(<span class="st">'extdata'</span>,
                                        <span class="st">'milk-proteins.fasta'</span>,
                                        <span class="dt">package=</span><span class="st">'MSGFplus'</span>),
                 <span class="dt">tolerance =</span> <span class="st">'10 ppm'</span>,
                 <span class="dt">enzyme =</span> <span class="st">'Trypsin'</span>)

## Add parameters after creation
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/instrument">instrument</a></span>(param) &lt;-<span class="st"> 'QExactive'</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/tda">tda</a></span>(param) &lt;-<span class="st"> </span><span class="ot">TRUE</span>
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/ntt">ntt</a></span>(param) &lt;-<span class="st"> </span><span class="dv">2</span>

## Add expected modifications
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/mods">mods</a></span>(param)[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/msgfParModification-class">msgfParModification</a></span>(<span class="st">'Carbamidomethyl'</span>,
                                        <span class="dt">composition =</span> <span class="st">'C2H3N1O1'</span>,
                                        <span class="dt">residues =</span> <span class="st">'C'</span>,
                                        <span class="dt">type =</span> <span class="st">'fix'</span>,
                                        <span class="dt">position =</span> <span class="st">'any'</span>)

<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/mods">mods</a></span>(param)[[<span class="dv">2</span>]] &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/msgfParModification-class">msgfParModification</a></span>(<span class="dt">name =</span> <span class="st">'Oxidation'</span>,
                                        <span class="dt">mass =</span> <span class="fl">15.994915</span>,
                                        <span class="dt">residues =</span> <span class="st">'M'</span>,
                                        <span class="dt">type =</span> <span class="st">'opt'</span>,
                                        <span class="dt">position =</span> <span class="st">'any'</span>)

<span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/mods">nMod</a></span>(param) &lt;-<span class="st"> </span><span class="dv">2</span>    <span class="co"># Number of allowed modifications per peptide</span>

## Get a summary of your parameters
<span class="kw">show</span>(param)</code></pre></div>
<pre><code>## An msgfPar object
## 
## Database:                   /home/lg390/R/x86_64-pc-linux-gnu-library/3.5/MSGFplus/extdata/milk-proteins.fasta
## Tolerance:                  10 ppm 
## TDA:                        TRUE 
## Instrument:                 3:   QExactive
## Enzyme:                     1:   Trypsin
## No. tolerable termini:      2 
## 
## Modifications:
## 
## Number of modifications per peptide: 2
## 
## Carbamidomethyl: C2H3N1O1, C, fix, any
## Oxidation:   15.99492, M, opt, any</code></pre>
</div>
<div id="performing-the-search-1" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#performing-the-search-1" class="anchor"></a>Performing the search</h3>
<p>Initiating the search is done using the <code>runMSGF</code> method. As a minimum it takes a parameter object and a list of raw data files and performs the search for each data file in sequence. More specialised operations are also possible such as running it asynchronously, but interested readers should refer to the <em><a href="http://bioconductor.org/packages/MSGFplus">MSGFplus</a></em> vignette for additional information.</p>
<p>The first time a search is initialised the MS-GF+ code is downloaded, so be sure to have an active internet connection (only applies to the first time a search is run).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSGFplus/topics/runMSGF">runMSGF</a></span>(param, <span class="st">'path/to/a/rawfile.mzML'</span>)</code></pre></div>
</div>
<div id="import-and-analyse-results-1" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#import-and-analyse-results-1" class="anchor"></a>Import and analyse results</h3>
<p>By default MSGFplus imports the results automatically using <em><a href="http://bioconductor.org/packages/mzID">mzID</a></em>. If only one file was analysed, the return value is an mzID object; if multiple files are analysed at once the return value is an mzIDCollection object.</p>
<p>If <code>import=FALSE</code> the results are not imported and can be accessed at a later time using the <em><a href="http://bioconductor.org/packages/mzID">mzID</a></em> package.</p>
</div>
<div id="running-ms-gf-through-a-gui" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#running-ms-gf-through-a-gui" class="anchor"></a>Running MS-GF+ through a GUI</h3>
<p><em><a href="http://bioconductor.org/packages/MSGFplus">MSGFplus</a></em> comes with a sister package, <em><a href="http://bioconductor.org/packages/MSGFgui">MSGFgui</a></em>, which provide a graphic interface to setting up and running MS-GF+ through R. Besides facilitating MS-GF+ analyses, which is arguably just as easy from the command line, it provides an intuitive way to investigate and evaluate the resulting identification data.</p>
<div class="figure">
<img src="figures/MSGFgui.png" alt="A screenshot of MSGFgui"><p class="caption">A screenshot of MSGFgui</p>
</div>
<p>The figure above shows an example of using <em><a href="http://bioconductor.org/packages/MSGFgui">MSGFgui</a></em>. It is possible to gradually drill down in the results starting from the protein level and ending at the raw spectrum level. mzIdentML files already created with MS-GF+ (using <em><a href="http://bioconductor.org/packages/MSGFplus">MSGFplus</a></em> or in other ways) can easily be imported into the gui to take advantage of the visualisation features, and results can be exported as either rds (for R), xlsx (for excel) or txt (for everything else) files.</p>
</div>
</div>
<div id="post-search-filtering-of-msms-ids-using-msnid" class="section level2">
<h2 class="hasAnchor">
<a href="#post-search-filtering-of-msms-ids-using-msnid" class="anchor"></a>Post-search Filtering of MS/MS IDs Using <em><a href="http://bioconductor.org/packages/MSnID">MSnID</a></em>
</h2>
<p>The main purpose of <em><a href="http://bioconductor.org/packages/MSnID">MSnID</a></em> package is to make sure that the peptide and protein identifications resulting from MS/MS searches are sufficiently confident for a given application.} MS/MS peptide and protein identification is a process that prone to uncertanities. A typical and currently most reliable way to quantify uncertainty in the list of identify spectra, peptides or proteins relies on so-called decoy database. For bottom-up (i.e. involving protein digestion) approaches a common way to construct a decoy database is simple inversion of protein amino-acid sequences. If the spectrum matches to normal protein sequence it can be true or false match. Matches to decoy part of the database are false only (excluding the palindromes). Therefore the false discovery rate (FDR) of identifications can be estimated as ratio of hits to decoy over normal parts of the protein sequence database. There are multiple levels of identification that FDR can be estimated for. First, is at the level of peptide/protein- to-spectrum matches. Second is at the level of unique peptide sequences. Note, true peptides tend to be identified by more then one spectrum. False peptide tend to be sporadic. Therefore, after collapsing the redundant peptide identifications from multiple spectra to the level of unique peptide sequence, the FDR typically increases. The extend of FDR increase depends on the type and complexity of the sample. The same trend is true for estimating the identification FDR at the protein level. True proteins tend to be identified with multiple peptides, while false protein identifications are commonly covered only by one peptide. Therefore FDR estimate tend to be even higher for protein level compare to peptide level. The estimation of the FDR is also affected by the number of LC-MS (runs) datasets in the experiment. Again, true identifications tend to be more consistent from run to run, while false are sporadic. After collapsing the redundancy across the runs, the number of true identification reduces much stronger compare to false identifications. Therefore, the peptide and protein FDR estimates need to be re-evaluated. The main objective of the MSnID package is to provide convenience tools for handling tasks on estimation of FDR, defining and optimizing the filtering criteria and ensuring confidence in MS/MS identification data. The user can specify the criteria for filtering the data (e.g. goodness or p-value of matching of experimental and theoretical fragmentation mass spectrum, deviation of theoretical from experimentally measured mass, presence of missed cleavages in the peptide sequence, etc), evaluate the performance of the filter judging by FDRs at spectrum, peptide and protein levels, and finally optimize the filter to achieve the maximum number of identifications while not exceeding maximally allowed FDR upper threshold.</p>
<div id="starting-project-and-importing-data" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#starting-project-and-importing-data" class="anchor"></a>Starting Project and Importing Data</h3>
<p>To start a project one have to specify a directory. Currently the only use of the directory is for storing cached results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MSnID"</span>)</code></pre></div>
<pre><code>## 
## Attaching package: 'MSnID'</code></pre>
<pre><code>## The following object is masked from 'package:ProtGenerics':
## 
##     peptides</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/MSnID-class">MSnID</a></span>(<span class="st">"."</span>)</code></pre></div>
<pre><code>## Note, the anticipated/suggested columns in the
## peptide-to-spectrum matching results are:
## -----------------------------------------------
## accession
## calculatedMassToCharge
## chargeState
## experimentalMassToCharge
## isDecoy
## peptide
## spectrumFile
## spectrumID</code></pre>
<p>Data can imported as <code>data.frame</code> or read from mzIdentML file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">PSMresults &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"human_brain.txt"</span>,
                                     <span class="dt">package=</span><span class="st">"MSnID"</span>),
                         <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/psms">psms</a></span>(msnid) &lt;-<span class="st"> </span>PSMresults
<span class="kw">show</span>(msnid)</code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  1 
## #PSMs: 767 at 49 % FDR
## #peptides: 687 at 57 % FDR
## #accessions: 665 at 65 % FDR</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mzids &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="st">"c_elegans.mzid.gz"</span>, <span class="dt">package=</span><span class="st">"MSnID"</span>)
msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/read_mzIDs">read_mzIDs</a></span>(msnid, mzids)</code></pre></div>
<pre><code>## Reading from mzIdentMLs ...</code></pre>
<pre><code>## reading c_elegans.mzid.gz... DONE!</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">show</span>(msnid)</code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  1 
## #PSMs: 12263 at 36 % FDR
## #peptides: 9489 at 44 % FDR
## #accessions: 7414 at 76 % FDR</code></pre>
</div>
<div id="analysis-of-peptide-sequences" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#analysis-of-peptide-sequences" class="anchor"></a>Analysis of Peptide Sequences</h3>
<p>A particular properties of peptide sequences we are interested in are</p>
<ol style="list-style-type: decimal">
<li>irregular cleavages at the termini of the peptides and</li>
<li>missing cleavage site within the peptide sequences.</li>
</ol>
<p>A particular properties of peptide sequences we are interested in are (1) irregular cleavages at the termini of the peptides and (2) missing cleavage site within the peptide sequences:</p>
<ul>
<li>Counting the number of irregular cleavage termimi (0, 1 or 2) in peptides sequence creates a new column <code>numIrregCleavages</code>.</li>
<li>Counting the number of missed cleavages in peptides sequence correspondingly creates a <code>numMissCleavages</code> column.</li>
</ul>
<p>The default regular expressions for the <code>validCleavagePattern</code> and <code>missedCleavagePattern</code> correspond to trypsin specificity.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/assess_termini">assess_termini</a></span>(msnid, <span class="dt">validCleavagePattern=</span><span class="st">"[KR]</span><span class="ch">\\</span><span class="st">.[^P]"</span>)
msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/assess_missed_cleavages">assess_missed_cleavages</a></span>(msnid, <span class="dt">missedCleavagePattern=</span><span class="st">"[KR](?=[^P$])"</span>)
<span class="kw">prop.table</span>(<span class="kw">table</span>(msnid<span class="op">$</span>numIrregCleavages))</code></pre></div>
<pre><code>## 
##           0           1           2 
## 0.801574390 0.189294149 0.009131462</code></pre>
<p>Now the object has two more columns, <code>numIrregCleavages</code> and <code>numMissCleavages</code>, evidently corresponding to the number of termini with irregular cleavages and number of missed cleavages within the peptide sequence. The figure below shows that peptides with 2 or more missed cleavages are likely to be false identifications.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pepCleav &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/psms">psms</a></span>(msnid)[,<span class="kw">c</span>(<span class="st">"numMissCleavages"</span>, <span class="st">"isDecoy"</span>, <span class="st">"peptide"</span>)])
pepCleav &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(pepCleav[,<span class="kw">c</span>(<span class="st">"numMissCleavages"</span>, <span class="st">"isDecoy"</span>)]))
<span class="kw">library</span>(<span class="st">"ggplot2"</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggplot">ggplot</a></span>(pepCleav, <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/aes">aes</a></span>(<span class="dt">x=</span>numMissCleavages, <span class="dt">y=</span>Freq, <span class="dt">fill=</span>isDecoy)) <span class="op">+</span>
<span class="st">    </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_bar">geom_bar</a></span>(<span class="dt">stat=</span><span class="st">'identity'</span>, <span class="dt">position=</span><span class="st">'dodge'</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/labs">ggtitle</a></span>(<span class="st">"Number of Missed Cleavages"</span>)</code></pre></div>
<p><img src="RforProteomics_files/figure-html/MSnIDmissedCleavagesPlot-1.png" width="700"></p>
</div>
<div id="defining-the-filter" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#defining-the-filter" class="anchor"></a>Defining the Filter</h3>
<p>The criteria that will be used for filtering the MS/MS data has to be present in the <code>MSnID</code> object. We will use -log10 transformed MS-GF+ Spectrum E-value, reflecting the goodness of match experimental and theoretical fragmentation patterns as one the filtering criteria. Let’s store it under the “msmsScore” name. The score density distribution shows that it is a good discriminant between non-decoy (red) and decoy hits (green).</p>
<p>For alternative MS/MS search engines refer to the engine-specific manual for the names of parameters reflecting the quality of MS/MS spectra matching. Examples of such parameters are <code>E-Value</code> for X!Tandem and <code>XCorr</code> and <code>$\Delta$Cn2</code> for SEQUEST.</p>
<p>As a second criterion we will be using the absolute mass measurement error (in ppm units) of the parent ion. The mass measurement errors tend to be small for non-decoy (enriched with real identificaiton) hits (red line) and is effectively uniformly distributed for decoy hits.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid<span class="op">$</span>msmsScore &lt;-<span class="st"> </span><span class="op">-</span><span class="kw">log10</span>(msnid<span class="op">$</span><span class="st">`</span><span class="dt">MS-GF:SpecEValue</span><span class="st">`</span>)
msnid<span class="op">$</span>absParentMassErrorPPM &lt;-<span class="st"> </span><span class="kw">abs</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/mass_measurement_error">mass_measurement_error</a></span>(msnid))</code></pre></div>
<p>MS/MS fiters are handled by a special <em>MSnIDFilter</em> class objects. Individual filtering criteria can be set by name (that is present in <code>names(msnid)</code>), comparison operator (&gt;, &lt;, = , …) defining if we should retain hits with higher or lower given the threshold and finally the threshold value itself. The filter below set in such a way that retains only those matches that has less then 5 ppm of parent ion mass measurement error and more the <span class="math inline">\(10^7\)</span> MS-GF:SpecEValue.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filtObj &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/MSnIDFilter-class">MSnIDFilter</a></span>(msnid)
filtObj<span class="op">$</span>absParentMassErrorPPM &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">comparison=</span><span class="st">"&lt;"</span>, <span class="dt">threshold=</span><span class="fl">5.0</span>)
filtObj<span class="op">$</span>msmsScore &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">comparison=</span><span class="st">"&gt;"</span>, <span class="dt">threshold=</span><span class="fl">8.0</span>)
<span class="kw">show</span>(filtObj)</code></pre></div>
<pre><code>## MSnIDFilter object
## (absParentMassErrorPPM &lt; 5) &amp; (msmsScore &gt; 8)</code></pre>
<p>The stringency of the filter can be evaluated at different levels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj, <span class="dt">level=</span><span class="st">"PSM"</span>)</code></pre></div>
<pre><code>##            fdr    n
## PSM 0.00272745 5147</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj, <span class="dt">level=</span><span class="st">"peptide"</span>)</code></pre></div>
<pre><code>##                fdr    n
## peptide 0.00424371 3313</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj, <span class="dt">level=</span><span class="st">"accession"</span>)</code></pre></div>
<pre><code>##                  fdr    n
## accession 0.01770658 1207</code></pre>
</div>
<div id="optimizing-the-filter" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#optimizing-the-filter" class="anchor"></a>Optimizing the Filter</h3>
<p>The threshold values in the example above are not necessarily optimal and set just be in the range of probable values. Filters can be optimized to ensure maximum number of identifications (peptide-to-spectrum matches, unique peptide sequences or proteins) within a given FDR upper limit.</p>
<p>First, the filter can be optimized simply by stepping through individual parameters and their combinations. The idea has been described in <span class="citation">(Piehowski et al. 2013)</span>. The resulting <code>MSnIDFilter</code> object can be used for final data filtering or can be used as a good starting parameters for follow-up refining optimizations with more advanced algorithms.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filtObj.grid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/optimize_filter">optimize_filter</a></span>(filtObj, msnid, <span class="dt">fdr.max=</span><span class="fl">0.01</span>,
                                <span class="dt">method=</span><span class="st">"Grid"</span>, <span class="dt">level=</span><span class="st">"peptide"</span>,
                                <span class="dt">n.iter=</span><span class="dv">500</span>)
<span class="kw">show</span>(filtObj.grid)</code></pre></div>
<pre><code>## MSnIDFilter object
## (absParentMassErrorPPM &lt; 10) &amp; (msmsScore &gt; 7.8)</code></pre>
<p>The resulting <code>filtObj.grid</code> can be further fine tuned with such optimization routines as simulated annealing or Nelder-Mead optimization.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filtObj.nm &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/optimize_filter">optimize_filter</a></span>(filtObj.grid, msnid, <span class="dt">fdr.max=</span><span class="fl">0.01</span>,
                                <span class="dt">method=</span><span class="st">"Nelder-Mead"</span>, <span class="dt">level=</span><span class="st">"peptide"</span>,
                                <span class="dt">n.iter=</span><span class="dv">500</span>)
<span class="kw">show</span>(filtObj.nm)</code></pre></div>
<pre><code>## MSnIDFilter object
## (absParentMassErrorPPM &lt; 10) &amp; (msmsScore &gt; 7.8)</code></pre>
<p>Evaluate non-optimized and optimized filters.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj, <span class="dt">level=</span><span class="st">"peptide"</span>)</code></pre></div>
<pre><code>##                fdr    n
## peptide 0.00424371 3313</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj.grid, <span class="dt">level=</span><span class="st">"peptide"</span>)</code></pre></div>
<pre><code>##                 fdr    n
## peptide 0.009220702 3393</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/evaluate_filter">evaluate_filter</a></span>(msnid, filtObj.nm, <span class="dt">level=</span><span class="st">"peptide"</span>)</code></pre></div>
<pre><code>##                 fdr    n
## peptide 0.009777778 3408</code></pre>
<p>Finally applying filter to remove predominantly false identifications.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/apply_filter">apply_filter</a></span>(msnid, filtObj.nm)
<span class="kw">show</span>(msnid)</code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  1 
## #PSMs: 5288 at 0.63 % FDR
## #peptides: 3408 at 0.98 % FDR
## #accessions: 1253 at 3.8 % FDR</code></pre>
<p>Removing hits to decoy and contaminant sequences using the same <code>apply_filter</code> method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/apply_filter">apply_filter</a></span>(msnid, <span class="st">"isDecoy == FALSE"</span>)
<span class="kw">show</span>(msnid)</code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  1 
## #PSMs: 5255 at 0 % FDR
## #peptides: 3375 at 0 % FDR
## #accessions: 1207 at 0 % FDR</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnid &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/apply_filter">apply_filter</a></span>(msnid, <span class="st">"!grepl('Contaminant',accession)"</span>)
<span class="kw">show</span>(msnid)</code></pre></div>
<pre><code>## MSnID object
## Working directory: "."
## #Spectrum Files:  1 
## #PSMs: 5246 at 0 % FDR
## #peptides: 3368 at 0 % FDR
## #accessions: 1205 at 0 % FDR</code></pre>
</div>
<div id="interface-with-other-bioconductor-packages" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#interface-with-other-bioconductor-packages" class="anchor"></a>Interface with Other Bioconductor Packages</h3>
<p>One can extract the entire PSMs tables as <code>data.frame</code> or <code>data.table</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">psm.df &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/psms">psms</a></span>(msnid)
psm.dt &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(msnid, <span class="st">"data.table"</span>)</code></pre></div>
<p>If only interested in the non-redundant list of confidently identified peptides or proteins</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">peps &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/peptides">peptides</a></span>(msnid)
<span class="kw">head</span>(peps)</code></pre></div>
<pre><code>## [1] "K.AISQIQEYVDYYGGSGVQHIALNTSDIITAIEALR.A"      
## [2] "K.SAGSGYLVGDSLTFVDLLVAQHTADLLAANAALLDEFPQFK.A"
## [3] "K.NSIFTNVAETANGEYFWEGLEDEIADKNVDITTWLGEK.W"   
## [4] "R.VFCLLGDGESAEGSVWEAAAFASIYKLDNLVAIVDVNR.L"   
## [5] "R.TTDSDGNNTGLDLYTVDQVEHSNYVEQNFLDFIFVFR.K"    
## [6] "R.KFDADGSGKLEFDEFCALVYTVANTVDKETLEKELR.E"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prots &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/accessions">accessions</a></span>(msnid)
<span class="kw">head</span>(prots)</code></pre></div>
<pre><code>## [1] "CE02347" "CE07055" "CE12728" "CE36358" "CE36359" "CE36360"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">prots &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnID/topics/accessions">proteins</a></span>(msnid) <span class="co"># may be more intuitive then accessions</span>
<span class="kw">head</span>(prots)</code></pre></div>
<pre><code>## [1] "CE02347" "CE07055" "CE12728" "CE36358" "CE36359" "CE36360"</code></pre>
<p>The <em><a href="http://bioconductor.org/packages/MSnID">MSnID</a></em> package is aimed at providing convenience functionality to handle MS/MS identifications. Quantification <em>per se</em> is outside of the scope of the package. The only type of quantitation that can be seamlessly tied with MS/MS identification analysis is so-called <em>spectral counting</em> approach. In such an approach a peptide abundance is considered to be directly proportional to the number of matched MS/MS spectra. In its turn protein abunance is proportional to the sum of the number of spectra of the matching peptides. The <em>MSnID</em> object can be converted to an <em>MSnSet</em> object defined in <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> that extends generic Bioconductor <em>eSet</em> class to quantitative proteomics data. The spectral count data can be analyzed with <em><a href="http://bioconductor.org/packages/msmsEDA">msmsEDA</a></em>, <em><a href="http://bioconductor.org/packages/msmsTests">msmsTests</a></em> or <em><a href="http://bioconductor.org/packages/DESeq">DESeq</a></em> packages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnset &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/mzRident2dfr">as</a></span>(msnid, <span class="st">"MSnSet"</span>)
<span class="kw">library</span>(<span class="st">"MSnbase"</span>)
<span class="kw">head</span>(<span class="kw">fData</span>(msnset))</code></pre></div>
<pre><code>##                                                       peptide
## A.AGLKPTQAMVTK.A                             A.AGLKPTQAMVTK.A
## A.AVLEYLAAEVLELAGNAAR.D               A.AVLEYLAAEVLELAGNAAR.D
## A.DCLHCICMR.E                                   A.DCLHCICMR.E
## A.DLFTSIADMQNLLETER.N                   A.DLFTSIADMQNLLETER.N
## A.EKKRKAAETSLMEK.D                         A.EKKRKAAETSLMEK.D
## A.EQLPEKFYGTFDLDHSENFDEYLTAK.G A.EQLPEKFYGTFDLDHSENFDEYLTAK.G
##                                                         accession
## A.AGLKPTQAMVTK.A                                 CE01236, CE30652
## A.AVLEYLAAEVLELAGNAAR.D                          CE04501, CE05477
## A.DCLHCICMR.E                  CE04442, CE17549, CE24850, CE34002
## A.DLFTSIADMQNLLETER.N                                     CE20261
## A.EKKRKAAETSLMEK.D                                        CE27133
## A.EQLPEKFYGTFDLDHSENFDEYLTAK.G                            CE04532</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(msnset))</code></pre></div>
<pre><code>##                                c_elegans_A_3_1_21Apr10_Draco_10-03-04_dta.txt
## A.AGLKPTQAMVTK.A                                                            1
## A.AVLEYLAAEVLELAGNAAR.D                                                     1
## A.DCLHCICMR.E                                                               1
## A.DLFTSIADMQNLLETER.N                                                       1
## A.EKKRKAAETSLMEK.D                                                          1
## A.EQLPEKFYGTFDLDHSENFDEYLTAK.G                                              1</code></pre>
<p>Note, the convertion from <code>MSnID</code> to <code>MSnSet</code> uses peptides as features. The number of redundant peptide observations represent so-called spectral count that can be used for rough quantitative analysis. Summing of all of the peptide counts to a proteins level can be done with <code>combineFeatures</code> function from <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msnset &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/MSnbase/topics/combineFeatures">combineFeatures</a></span>(msnset,
                            <span class="kw">fData</span>(msnset)<span class="op">$</span>accession,
                            <span class="dt">redundancy.handler=</span><span class="st">"unique"</span>,
                            <span class="dt">fun=</span><span class="st">"sum"</span>,
                            <span class="dt">cv=</span><span class="ot">FALSE</span>)
<span class="kw">head</span>(<span class="kw">fData</span>(msnset))</code></pre></div>
<pre><code>##                                     peptide accession
## CE00078                         K.RLPVAPR.G   CE00078
## CE00103 K.LPNDDIGVQVSYLGEPHTFTPEQVLAALLTK.L   CE00103
## CE00134                       I.PAEVAEHLK.A   CE00134
## CE00209            K.ALEGPGPGEDAAHSENNPPR.N   CE00209
## CE00302                  K.LTYFDIHGLAEPIR.L   CE00302
## CE00318        K.ALNALCAQLMTELADALEVLDTDK.S   CE00318</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/tidyeval">exprs</a></span>(msnset))</code></pre></div>
<pre><code>##         c_elegans_A_3_1_21Apr10_Draco_10-03-04_dta.txt
## CE00078                                              4
## CE00103                                              3
## CE00134                                              4
## CE00209                                              8
## CE00302                                              2
## CE00318                                             11</code></pre>
</div>
</div>
</div>
<div id="sec:qc" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:qc" class="anchor"></a>Quality control</h1>
<p>Quality control (QC) is an essential part of any high throughput data driven approach. Bioconductor has a rich history of QC for various genomics data and currently two packages support proteomics QC.</p>
<p><em><a href="http://bioconductor.org/packages/proteoQC">proteoQC</a></em> provides a dedicated a dedicated pipeline that will produce a dynamic and extensive html report. It uses the <em><a href="http://bioconductor.org/packages/rTANDEM">rTANDEM</a></em> package to automate the generation of identification data and uses information about the experimental/replication design.</p>
<p>The <em><a href="http://bioconductor.org/packages/qcmetrics">qcmetrics</a></em> package is a general framework to define QC metrics and bundle them together to generate html or pdf reports. It provides some ready made metrics for MS data and N15 labelled data.</p>
</div>
<div id="sec:annot" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:annot" class="anchor"></a>Annotation</h1>
<p>In this section, we briefly present some Bioconductor annotation infrastructure.</p>
<p>We start with the <em><a href="http://bioconductor.org/packages/hpar">hpar</a></em> package, an interface to the <em>Human Protein Atlas</em> <span class="citation">(Uhlén et al. 2005, Uhlen2010)</span>, to retrieve subcellular localisation information for the <code>ENSG00000002746</code> ensemble gene.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">id &lt;-<span class="st"> "ENSG00000105323"</span>
<span class="kw">library</span>(<span class="st">"hpar"</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/hpar/topics/getHpa">getHpa</a></span>(id, <span class="st">"hpaSubcellularLoc"</span>)</code></pre></div>
<pre><code>##                 Gene Gene.name Reliability    Enhanced Supported Approved
## 2240 ENSG00000105323  HNRNPUL1    Enhanced Nucleoplasm                   
##      Uncertain Single.cell.variation.intensity
## 2240                                          
##      Single.cell.variation.spatial Cell.cycle.dependency
## 2240                                                    
##                         GO.id
## 2240 Nucleoplasm (GO:0005654)</code></pre>
<p>Below, we make use of the human annotation package <em><a href="http://bioconductor.org/packages/org.Hs.eg.db">org.Hs.eg.db</a></em> and the Gene Ontology annotation package <em><a href="http://bioconductor.org/packages/GO.db">GO.db</a></em> to retrieve compatible information with above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"org.Hs.eg.db"</span>)
<span class="kw">library</span>(<span class="st">"GO.db"</span>)
ans &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/AnnotationDbi/topics/AnnotationDb-class">select</a></span>(org.Hs.eg.db,
                             <span class="dt">keys =</span> id,
                             <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">"ENSEMBL"</span>, <span class="st">"GO"</span>, <span class="st">"ONTOLOGY"</span>),
                             <span class="dt">keytype =</span> <span class="st">"ENSEMBL"</span>)</code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ans &lt;-<span class="st"> </span>ans[ans<span class="op">$</span>ONTOLOGY <span class="op">==</span><span class="st"> "CC"</span>, ]
ans</code></pre></div>
<pre><code>##           ENSEMBL         GO EVIDENCE ONTOLOGY
## 4 ENSG00000105323 GO:0005634      IDA       CC
## 5 ENSG00000105323 GO:0005654      IDA       CC
## 6 ENSG00000105323 GO:0005654      TAS       CC</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sapply</span>(<span class="kw">as.list</span>(GOTERM[ans<span class="op">$</span>GO]), slot, <span class="st">"Term"</span>)</code></pre></div>
<pre><code>##    GO:0005634    GO:0005654    GO:0005654 
##     "nucleus" "nucleoplasm" "nucleoplasm"</code></pre>
<p>Finally, this information can also be retrieved from on-line databases using the <em><a href="http://bioconductor.org/packages/biomaRt">biomaRt</a></em> package <span class="citation">(Durinck et al. 2005)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"biomaRt"</span>)
ensembl &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomaRt/topics/useMart">useMart</a></span>(<span class="st">"ensembl"</span>,<span class="dt">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>)
efilter &lt;-<span class="st"> "ensembl_gene_id"</span>
eattr &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"go_id"</span>, <span class="st">"name_1006"</span>, <span class="st">"namespace_1003"</span>)
bmres &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomaRt/topics/getBM">getBM</a></span>(<span class="dt">attributes=</span>eattr, <span class="dt">filters =</span> efilter, <span class="dt">values =</span> id, <span class="dt">mart =</span> ensembl)
bmres[bmres<span class="op">$</span>namespace_<span class="dv">1003</span> <span class="op">==</span><span class="st"> "cellular_component"</span>, <span class="st">"name_1006"</span>]</code></pre></div>
<pre><code>## [1] "nucleoplasm" "nucleus"</code></pre>
</div>
<div id="sec:packages" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:packages" class="anchor"></a>Other packages</h1>
<div id="bioconductor-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#bioconductor-packages" class="anchor"></a>Bioconductor packages</h2>
<p>This section provides a complete list of packages available in the relevant Bioconductor version 3.8 <a href="http://www.bioconductor.org/packages/devel/BiocViews.html%3E"><em>biocView</em></a> categories. the tables below represent the packages for the <code>Proteomics</code> (112 packages), <code>MassSpectrometry</code> (75 packages) and <code>MassSpectrometryData</code> (20 experiment packages) categories.</p>
<div id="htmlwidget-47c7e0ed2c2a8bd560bf" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-47c7e0ed2c2a8bd560bf">{"x":{"filter":"none","data":[["ASEB","bioassayR","biobroom","BioNetStat","biosigner","BRAIN","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","customProDB","cydar","DAPAR","deltaGseg","DEP","DEqMS","diffcyt","DominoEffect","Doscheda","drawProteins","EGSEA","eiR","fCI","fmcsR","GraphPAC","HPAanalyze","hpar","IMMAN","InterMineR","iPAC","IPPD","isobar","kimod","LPEadj","MassSpecWavelet","mlm4omics","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","MSstats","MSstatsQC","MSstatsQCgui","mzID","mzR","NormalyzerDE","omicRexposome","PAA","Path2PPI","pathview","Pbase","PCpheno","PECA","pepXMLTab","PGA","pgca","phosphonormalizer","plgem","PLPE","PowerExplorer","ppiStats","proBAMr","PROcess","procoil","ProCoNA","proFIA","pRoloc","pRolocGUI","prot2D","ProteomicsAnnotationHubData","proteoQC","ProtGenerics","Pviz","qcmetrics","qPLEXanalyzer","QuartPAC","rain","RankProd","RCASPAR","Rchemcpp","Rcpi","readat","ropls","ROTS","RpsiXML","rpx","rTANDEM","sapFinder","ScISI","SDAMS","shinyTANDEM","SLGI","SpacePAC","sparsenetgls","specL","spliceSites","statTarget","SWATH2stats","synapter","tofsims","topdownr","TPP","XINA"],["ASEB","bioassayR","biobroom","BioNetStat","biosigner","BRAIN","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","customProDB","cydar","DAPAR","deltaGseg","DEP","DEqMS","diffcyt","DominoEffect","Doscheda","drawProteins","EGSEA","eiR","fCI","fmcsR","GraphPAC","HPAanalyze","hpar","IMMAN","InterMineR","iPAC","IPPD","isobar","kimod","LPEadj","MassSpecWavelet","mlm4omics","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","MSstats","MSstatsQC","MSstatsQCgui","mzID","mzR","NormalyzerDE","omicRexposome","PAA","Path2PPI","pathview","Pbase","PCpheno","PECA","pepXMLTab","PGA","pgca","phosphonormalizer","plgem","PLPE","PowerExplorer","ppiStats","proBAMr","PROcess","procoil","ProCoNA","proFIA","pRoloc","pRolocGUI","prot2D","ProteomicsAnnotationHubData","proteoQC","ProtGenerics","Pviz","qcmetrics","qPLEXanalyzer","QuartPAC","rain","RankProd","RCASPAR","Rchemcpp","Rcpi","readat","ropls","ROTS","RpsiXML","rpx","rTANDEM","sapFinder","ScISI","SDAMS","shinyTANDEM","SLGI","SpacePAC","sparsenetgls","specL","spliceSites","statTarget","SWATH2stats","synapter","tofsims","topdownr","TPP","XINA"],["Predict Acetylated Lysine Sites","Cross-target analysis of small molecule bioactivity","Turn Bioconductor objects into tidy data frames","Biological Network Analysis","Signature discovery from omics data","Baffling Recursive Algorithm for Isotope distributioN calculations","A mass spectrometry imaging toolbox for statistical analysis","Causal network analysis methods","Training of boolean logic models of signalling networks using prior knowledge networks and perturbation data","R interface to a subset of OpenBabel functionalities","Cheminformatics Toolkit for R","Visualization and management of the protein-protein interaction networks.","Cleavage of Polypeptide Sequences","A package for the clinical proteomic profiling data analysis","Add-on to CellNOptR: Discretized time treatments","Integration of CellNOptR to add missing links","ODE add-on to CellNOptR","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Using Mass Cytometry for Differential Abundance Analyses","Tools for the Differential Analysis of Proteins Abundance with R","deltaGseg","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","Differential discovery in high-dimensional cytometry via high-resolution clustering","Identification and Annotation of Protein Hotspot Residues","A DownStream Chemo-Proteomics Analysis Pipeline","Package to Draw Protein Schematics from Uniprot API output","Ensemble of Gene Set Enrichment Analyses","Accelerated similarity searching of small molecules","f-divergence Cutoff Index for Differential Expression Analysis in Transcriptomics and Proteomics","Mismatch Tolerant Maximum Common Substructure Searching","Identification of Mutational Clusters in Proteins via a Graph Theoretical Approach.","Retrieve and analyze data from the Human Protein Atlas","Human Protein Atlas in R","Interlog protein network reconstruction by Mapping and Mining ANalysis","R Interface with InterMine-Powered Databases","Identification of Protein Amino acid Clustering","Isotopic peak pattern deconvolution for Protein Mass Spectrometry by template matching","Analysis and quantitation of isobarically tagged MSMS proteomics data","A k-tables approach to integrate multiple Omics-Data","A correction of the local pooled error (LPE) method to replace the asymptotic variance adjustment with an unbiased adjustment\nbased on sample size.","Mass spectrum processing by wavelet-based algorithms","Multilevel Model for Multivariate Responses with Missing Values","A shiny GUI for MSGFplus","An interface between R and MS-GF+","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Evaluation of normalization methods and calculation of differential expression analysis statistics","Exposome and omic data associatin and integration analysis","PAA (Protein Array Analyzer)","Prediction of pathway-related protein-protein interaction networks","a tool set for pathway based data integration and visualization","Manipulating and exploring protein and proteomics data","Phenotypes and cellular organizational units","Probe-level Expression Change Averaging","Parsing pepXML files and filter based on peptide FDR.","An package for identification of novel peptides by customized database derived from RNA-Seq","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Compensates for the bias introduced by median normalization in phosphoproteomics","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Local Pooled Error Test for Differential Expression with Paired High-throughput Data","Power Estimation Tool for RNA-Seq and proteomics data","Protein-Protein Interaction Statistical Package","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Prediction of Oligomerization of Coiled Coil Proteins","Protein co-expression network analysis (ProCoNA).","Preprocessing of FIA-HRMS data","A unifying bioinformatics framework for spatial proteomics","Interactive visualisation of spatial proteomics data","Statistical Tools for volume data from 2D Gel Electrophoresis","Transform public proteomics data resources into Bioconductor Data Structures","An R package for proteomics data quality control","S4 generic functions for Bioconductor proteomics infrastructure","Peptide Annotation and Data Visualization using Gviz","A Framework for Quality Control","Tools for qPLEX-RIME data analysis","Identification of mutational clusters in protein quaternary structures.","Rhythmicity Analysis Incorporating Non-parametric Methods","Rank Product method for identifying differentially expressed genes with application in meta-analysis","A package for survival time prediction based on a piecewise baseline hazard Cox regression model.","Similarity measures for chemical compounds","Molecular Informatics Toolkit for Compound-Protein Interaction in Drug Discovery","Functionality to Read and Manipulate SomaLogic ADAT files","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Reproducibility-Optimized Test Statistic","R interface to PSI-MI 2.5 files","R Interface to the ProteomeXchange Repository","Interfaces the tandem protein identification algorithm in R","A package for variant peptides detection and visualization in shotgun proteomics.","In Silico Interactome","Differential Abundant Analysis for Metabolomics and Proteomics Data","Provides a GUI for rTANDEM","Synthetic Lethal Genetic Interaction","Identification of Mutational Clusters in 3D Protein Space via Simulation.","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","A bioconductor package for exploration of alignment gap positions from RNA-seq data","Statistical Analysis of Molecular Profiles","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Import, process and analysis of Time-of-Flight Secondary Ion Mass Spectrometry (ToF-SIMS) imaging data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","Multiplexes isobaric mass tagged-based kinetics data for network analysis"],["1.25.0","1.19.3","1.13.2","1.1.2","1.9.0","1.27.1","1.13.1","1.13.0","1.27.1","1.19.2","3.33.4","1.21.1","1.19.0","1.31.0","1.23.1","1.21.1","1.23.0","1.21.5","1.5.6","1.13.12","1.21.1","1.3.3","0.99.44","1.1.10","1.1.2","1.3.0","1.1.1","1.9.2","1.21.0","1.11.1","1.23.2","1.23.1","0.99.14","1.23.4","1.1.0","1.3.3","1.25.2","1.29.0","1.27.1","1.9.0","1.41.0","1.47.0","0.99.22","1.15.1","1.15.1","1.19.0","1.19.0","2.7.7","1.15.0","3.13.5","1.99.3","1.1.1","1.19.0","2.15.5","0.99.21","1.3.1","1.15.2","1.11.0","1.21.2","0.21.3","1.43.0","1.17.0","1.15.0","1.11.4","1.5.1","1.5.1","1.53.0","1.41.0","1.1.1","1.47.0","1.15.0","1.57.0","2.9.1","1.19.0","1.7.0","1.21.8","1.15.2","1.19.0","1.11.1","1.17.0","1.13.0","1.15.0","1.19.2","0.99.12","1.13.1","1.15.0","3.7.0","1.27.0","2.19.0","1.17.2","1.7.0","1.13.8","1.9.1","2.23.0","1.17.2","1.21.0","1.19.0","1.53.0","1.1.1","1.19.0","1.41.0","1.19.3","0.99.7","1.15.0","1.29.0","1.11.3","1.11.7","2.5.2","1.9.0","1.3.4","3.9.5","0.99.13"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script><div id="htmlwidget-0082cbbd9435e6828015" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0082cbbd9435e6828015">{"x":{"filter":"none","data":[["apComplex","BRAIN","CAMERA","Cardinal","CATALYST","cosmiq","customProDB","DAPAR","DEP","DEqMS","Doscheda","flagme","gaga","INDEED","IPO","isobar","LOBSTAHS","MAIT","MassArray","MassSpecWavelet","Metab","metabomxtr","metaMS","MetCirc","MetNet","mlm4omics","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","msPurity","MSstats","MSstatsQC","MSstatsQCgui","mzID","mzR","PAPi","Pbase","pepXMLTab","PGA","pgca","plgem","proBAMr","PROcess","proFIA","pRoloc","Prostar","proteoQC","ProtGenerics","qcmetrics","qPLEXanalyzer","Rdisop","Risa","RMassBank","rols","ropls","rpx","rTANDEM","sapFinder","SDAMS","shinyTANDEM","SIMAT","sparsenetgls","specL","statTarget","SWATH2stats","synapter","TargetSearch","tofsims","topdownr","TPP","xcms","yamss"],["apComplex","BRAIN","CAMERA","Cardinal","CATALYST","cosmiq","customProDB","DAPAR","DEP","DEqMS","Doscheda","flagme","gaga","INDEED","IPO","isobar","LOBSTAHS","MAIT","MassArray","MassSpecWavelet","Metab","metabomxtr","metaMS","MetCirc","MetNet","mlm4omics","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","msPurity","MSstats","MSstatsQC","MSstatsQCgui","mzID","mzR","PAPi","Pbase","pepXMLTab","PGA","pgca","plgem","proBAMr","PROcess","proFIA","pRoloc","Prostar","proteoQC","ProtGenerics","qcmetrics","qPLEXanalyzer","Rdisop","Risa","RMassBank","rols","ropls","rpx","rTANDEM","sapFinder","SDAMS","shinyTANDEM","SIMAT","sparsenetgls","specL","statTarget","SWATH2stats","synapter","TargetSearch","tofsims","topdownr","TPP","xcms","yamss"],["Estimate protein complex membership using AP-MS protein data","Baffling Recursive Algorithm for Isotope distributioN calculations","Collection of annotation related methods for mass spectrometry data","A mass spectrometry imaging toolbox for statistical analysis","Cytometry dATa anALYSis Tools","cosmiq - COmbining Single Masses Into Quantities","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Tools for the Differential Analysis of Proteins Abundance with R","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","A DownStream Chemo-Proteomics Analysis Pipeline","Analysis of Metabolomics GC/MS Data","GaGa hierarchical model for high-throughput data analysis","An Implementation of Integrated Differential Expression and Differential Network Analysis for Biomarker Candidate Selection","Automated Optimization of XCMS Data Processing parameters","Analysis and quantitation of isobarically tagged MSMS proteomics data","Lipid and Oxylipin Biomarker Screening through Adduct Hierarchy Sequences","Statistical Analysis of Metabolomic Data","Analytical Tools for MassArray Data","Mass spectrum processing by wavelet-based algorithms","Metab: An R Package for a High-Throughput Analysis of Metabolomics Data Generated by GC-MS.","A package to run mixture models for truncated metabolomics data with normal or lognormal distributions","MS-based metabolomics annotation pipeline","Navigating mass spectral similarity in high-resolution MS/MS metabolomics data","Inferring metabolic networks from untargeted high-resolution mass spectrometry data","Multilevel Model for Multivariate Responses with Missing Values","A shiny GUI for MSGFplus","An interface between R and MS-GF+","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Automated Evaluation of Precursor Ion Purity for Mass Spectrometry Based Fragmentation in Metabolomics","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Predict metabolic pathway activity based on metabolomics data","Manipulating and exploring protein and proteomics data","Parsing pepXML files and filter based on peptide FDR.","An package for identification of novel peptides by customized database derived from RNA-Seq","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Preprocessing of FIA-HRMS data","A unifying bioinformatics framework for spatial proteomics","Provides a GUI for DAPAR","An R package for proteomics data quality control","S4 generic functions for Bioconductor proteomics infrastructure","A Framework for Quality Control","Tools for qPLEX-RIME data analysis","Decomposition of Isotopic Patterns","Converting experimental metadata from ISA-tab into Bioconductor data structures","Workflow to process tandem MS files and build MassBank records","An R interface to the Ontology Lookup Service","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","R Interface to the ProteomeXchange Repository","Interfaces the tandem protein identification algorithm in R","A package for variant peptides detection and visualization in shotgun proteomics.","Differential Abundant Analysis for Metabolomics and Proteomics Data","Provides a GUI for rTANDEM","GC-SIM-MS data processing and alaysis tool","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Statistical Analysis of Molecular Profiles","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","A package for the analysis of GC-MS metabolite profiling data","Import, process and analysis of Time-of-Flight Secondary Ion Mass Spectrometry (ToF-SIMS) imaging data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","LC/MS and GC/MS Data Analysis","Tools for high-throughput metabolomics"],["2.47.0","1.27.1","1.37.0","1.13.1","1.5.6","1.15.2","1.21.5","1.13.12","1.3.3","0.99.44","1.3.0","1.37.0","2.27.0","0.99.17","1.7.1","1.27.1","1.7.2","1.15.0","1.33.0","1.47.0","1.15.0","1.15.0","1.17.0","1.11.1","0.99.22","0.99.22","1.15.1","1.15.1","1.19.0","1.19.0","2.7.7","1.15.0","1.7.2","3.13.5","1.99.3","1.1.1","1.19.0","2.15.5","1.21.0","0.21.3","1.15.0","1.11.4","1.5.1","1.53.0","1.15.0","1.57.0","1.7.0","1.21.8","1.13.10","1.17.0","1.13.0","1.19.2","0.99.12","1.41.0","1.23.2","2.9.1","2.9.3","1.13.8","1.17.2","1.21.0","1.19.0","1.1.1","1.19.0","1.13.4","0.99.7","1.15.0","1.11.3","1.11.7","2.5.2","1.37.3","1.9.0","1.3.4","3.9.5","3.3.2","1.7.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script><div id="htmlwidget-3e308f883dd8a9890781" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3e308f883dd8a9890781">{"x":{"filter":"none","data":[["CardinalWorkflows","DAPARdata","faahKO","gcspikelite","iontreeData","metaMSdata","msdata","msPurityData","msqc1","MSstatsBioData","mtbls2","plasFIA","ProData","pRolocdata","PtH2O2lipids","qPLEXdata","RforProteomics","RMassBankData","synapterdata","topdownrdata"],["CardinalWorkflows","DAPARdata","faahKO","gcspikelite","iontreeData","metaMSdata","msdata","msPurityData","msqc1","MSstatsBioData","mtbls2","plasFIA","ProData","pRolocdata","PtH2O2lipids","qPLEXdata","RforProteomics","RMassBankData","synapterdata","topdownrdata"],["Datasets and workflows for the Cardinal mass spectrometry imaging package","Data accompanying the DAPAR and Prostar packages","Saghatelian et al. (2004) FAAH knockout LC/MS data","Spike-in data for GC/MS data and methods within flagme","Data provided to show the usage of functions in iontree package","Example CDF data for the metaMS package","Various Mass Spectrometry raw data example files","Fragmentation spectral libraries and data to test the msPurity package","Sigma mix MSQC1 data","Datasets of published biological studies with DDA or SRM experiments","MetaboLights MTBLS2: Comparative LC/MS-based profiling of silver nitrate-treated Arabidopsis thaliana leaves of wild-type and\ncyp79B2 cyp79B3 double knockout plants. Böttcher et al. (2004)","FIA-HRMS plasma dataset","SELDI-TOF data of Breast cancer samples","Data accompanying the pRoloc package","P. tricornutum HPLC-ESI-MS Lipid Data from van Creveld et al. (2015)","Data accompanying qPLEXanalyzer package","Companion package to the 'Using R and Bioconductor for proteomics data analysis' publication","Test dataset for RMassBank","Data accompanying the synapter package","Example Files for the topdownr R Package"],["1.13.0","1.11.2","1.21.0","1.19.0","1.17.2","1.17.0","0.21.0","1.9.0","1.9.1","1.3.0","1.11.0","1.9.0","1.19.0","1.19.5","1.7.1","0.99.1","1.19.3","1.19.0","1.19.0","1.3.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script><p>The tables can easily be generated with the <code>proteomicsPackages</code>, <code>massSpectrometryPackages</code> and <code>massSpectrometryDataPackages</code> functions. The respective package tables can then be interactively explored using the <code>display</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pp &lt;-<span class="st"> </span><span class="kw"><a href="../reference/proteomicsPackages.html">proteomicsPackages</a></span>()
<span class="kw">display</span>(pp)</code></pre></div>
</div>
<div id="other-cran-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#other-cran-packages" class="anchor"></a>Other CRAN packages</h2>
<p>The CRAN task view on <a href="http://cran.r-project.org/web/views/ChemPhys.html">Chemometrics and Computational Physics</a> is another useful ressource listing 92 packages, including a set of packages for mass spectrometry and proteomics, some of which are illustrated in this document.</p>
<ul>
<li>
<em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> provides tools for quantitative analysis of MALDI-TOF mass spectrometry data, with support for baseline correction, peak detection and plotting of mass spectra.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=OrgMassSpecR">OrgMassSpecR</a></em> is for organic/biological mass spectrometry, with a focus on graphical display, quantification using stable isotope dilution, and protein hydrogen/deuterium exchange experiments.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=FTICRMS">FTICRMS</a></em> provides functions for Analyzing Fourier Transform-Ion Cyclotron Resonance Mass Spectrometry Data.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=titan">titan</a></em> provides a GUI to analyze mass spectrometric data on the relative abundance of two substances from a titration series.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=digeR">digeR</a></em> provides a GUI interface for analysing 2D DIGE data. It allows to perform correlation analysis, score plot, classification, feature selection and power analysis for 2D DIGE experiment data.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=protViz">protViz</a></em> helps with quality checks, visualizations and analysis of mass spectrometry data, coming from proteomics experiments. The package is developed, tested and used at the Functional Genomics Center Zurich.</li>
</ul>
<p>Suggestions for additional R packages are welcome and will be added to the vignette. Please send suggestions and possibly a short description and/or a example utilisation with code to <code>lg390@cam.ac.uk</code>. The only requirement is that the package must be available on an official package channel (CRAN, Bioconductor, R-forge, Omegahat), i.e. not only available through a personal web page.</p>
</div>
</div>
<div id="sec:sessionInfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:sessionInfo" class="anchor"></a>Session information</h1>
<p>All software and respective versions used in this document, as returned by <code>sessionInfo()</code> are detailed below.</p>
<pre><code>## R version 3.5.1 Patched (2018-08-16 r75161)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8          LC_NUMERIC=C                 
##  [3] LC_TIME=en_GB.UTF-8           LC_COLLATE=en_GB.UTF-8       
##  [5] LC_MONETARY=en_GB.UTF-8       LC_MESSAGES=en_GB.UTF-8      
##  [7] LC_PAPER=en_GB.UTF-8          LC_NAME=en_GB.UTF-8          
##  [9] LC_ADDRESS=en_GB.UTF-8        LC_TELEPHONE=en_GB.UTF-8     
## [11] LC_MEASUREMENT=en_GB.UTF-8    LC_IDENTIFICATION=en_GB.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] MSnID_1.14.0             cleaver_1.18.0          
##  [3] OrgMassSpecR_0.5-3       msdata_0.20.0           
##  [5] reshape2_1.4.3           ggplot2_3.0.0           
##  [7] RColorBrewer_1.1-2       RforProteomics_1.19.4   
##  [9] MSGFgui_1.14.0           xlsx_0.6.1              
## [11] MSGFplus_1.14.0          rTANDEM_1.20.0          
## [13] data.table_1.11.4        Rdisop_1.40.0           
## [15] RcppClassic_0.9.11       GO.db_3.6.0             
## [17] org.Hs.eg.db_3.6.0       AnnotationDbi_1.42.1    
## [19] BRAIN_1.26.0             lattice_0.20-35         
## [21] Biostrings_2.48.0        XVector_0.20.0          
## [23] IRanges_2.14.11          S4Vectors_0.18.3        
## [25] PolynomF_1.0-2           hpar_1.22.2             
## [27] rols_2.9.2               IPPD_1.28.0             
## [29] bitops_1.0-6             digest_0.6.17           
## [31] XML_3.98-1.16            Matrix_1.2-14           
## [33] MASS_7.3-50              MALDIquantForeign_0.11.5
## [35] MALDIquant_1.18          rpx_1.16.0              
## [37] mzID_1.18.0              MSnbase_2.7.4           
## [39] ProtGenerics_1.12.0      BiocParallel_1.14.2     
## [41] Biobase_2.40.0           BiocGenerics_0.26.0     
## [43] mzR_2.15.5               Rcpp_0.12.18            
## [45] BiocManager_1.30.2       DT_0.4                  
## [47] BiocStyle_2.8.2         
## 
## loaded via a namespace (and not attached):
##  [1] backports_1.1.2             plyr_1.8.4                 
##  [3] lazyeval_0.2.1              crosstalk_1.0.0            
##  [5] GenomeInfoDb_1.16.0         foreach_1.4.4              
##  [7] BiocInstaller_1.30.0        htmltools_0.3.6            
##  [9] magrittr_1.5                memoise_1.1.0              
## [11] doParallel_1.0.11           limma_3.36.3               
## [13] shinyFiles_0.7.0            matrixStats_0.54.0         
## [15] R.utils_2.7.0               pkgdown_1.1.0              
## [17] prettyunits_1.0.2           colorspace_1.3-2           
## [19] blob_1.1.1                  xfun_0.3                   
## [21] dplyr_0.7.6                 crayon_1.3.4               
## [23] RCurl_1.95-4.11             jsonlite_1.5               
## [25] readBrukerFlexData_1.8.5    graph_1.58.0               
## [27] roxygen2_6.1.0              bindr_0.1.1                
## [29] impute_1.54.0               iterators_1.0.10           
## [31] glue_1.3.0                  gtable_0.2.0               
## [33] zlibbioc_1.26.0             readMzXmlData_2.8.1        
## [35] DelayedArray_0.6.6          R.cache_0.13.0             
## [37] scales_1.0.0                vsn_3.48.1                 
## [39] DBI_1.0.0                   xtable_1.8-3               
## [41] progress_1.2.0              bit_1.1-14                 
## [43] preprocessCore_1.42.0       htmlwidgets_1.2            
## [45] httr_1.3.1                  pkgconfig_2.0.2            
## [47] rJava_0.9-10                R.methodsS3_1.7.1          
## [49] tidyselect_0.2.4            labeling_0.3               
## [51] rlang_0.2.2                 later_0.7.4                
## [53] munsell_0.5.0               biocViews_1.48.3           
## [55] tools_3.5.1                 RSQLite_2.1.1              
## [57] evaluate_0.11               stringr_1.3.1              
## [59] yaml_2.2.0                  knitr_1.20                 
## [61] bit64_0.9-7                 fs_1.2.6                   
## [63] purrr_0.2.5                 bindrcpp_0.2.2             
## [65] ncdf4_1.16                  RBGL_1.56.0                
## [67] mime_0.5                    R.oo_1.22.0                
## [69] xml2_1.2.0                  compiler_3.5.1             
## [71] curl_3.2                    affyio_1.50.0              
## [73] tibble_1.4.2                stringi_1.2.4              
## [75] highr_0.7                   desc_1.2.0                 
## [77] commonmark_1.5              pillar_1.3.0               
## [79] RUnit_0.4.32                GenomicRanges_1.32.6       
## [81] httpuv_1.4.5                R6_2.2.2                   
## [83] pcaMethods_1.72.0           affy_1.58.0                
## [85] bookdown_0.7                promises_1.0.1             
## [87] codetools_0.2-15            assertthat_0.2.0           
## [89] SummarizedExperiment_1.10.1 xlsxjars_0.6.1             
## [91] rprojroot_1.3-2             withr_2.1.2                
## [93] GenomeInfoDbData_1.1.0      hms_0.4.2                  
## [95] rmarkdown_1.10              shiny_1.1.0                
## [97] base64enc_0.1-3</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-synapter">
<p>Bond, Nicholas J., Pavel V. Shliaha, Kathryn S. Lilley, and Laurent Gatto. 2013. “Improving Qualitative and Quantitative Performance for Label Free Proteomics.” <em>J. Proteome Res.</em> doi:<a href="https://doi.org/10.1021/pr300776t">10.1021/pr300776t</a>.</p>
</div>
<div id="ref-Breitwieser2011">
<p>Breitwieser, F P, A Müller, L Dayon, T Köcher, A Hainard, P Pichler, U Schmidt-Erfurth, et al. 2011. “General Statistical Modeling of Data from Protein Relative Expression Isobaric Tags.” <em>J Proteome Res</em> 10 (6): 2758–66. doi:<a href="https://doi.org/10.1021/pr1012784">10.1021/pr1012784</a>.</p>
</div>
<div id="ref-Chambers2012">
<p>Chambers, M C, B Maclean, R Burke, D Amodei, D L Ruderman, S Neumann, L Gatto, et al. 2012. “A Cross-Platform Toolkit for Mass Spectrometry and Proteomics.” <em>Nat Biotechnol</em> 30 (10): 918–20. doi:<a href="https://doi.org/10.1038/nbt.2377">10.1038/nbt.2377</a>.</p>
</div>
<div id="ref-Craig2004">
<p>Craig, R, and R C Beavis. 2004. “TANDEM: Matching Proteins with Tandem Mass Spectra.” <em>Bioinformatics</em> 20 (9): 1466–7. doi:<a href="https://doi.org/10.1093/bioinformatics/bth092">10.1093/bioinformatics/bth092</a>.</p>
</div>
<div id="ref-TargetSearch2009">
<p>Cuadros-Inostroza, Alvaro, Camila Caldana, Henning Redestig, Jan Lisec, Hugo Pena-Cortes, Lothar Willmitzer, and Matthew A Hannah. 2009. “TargetSearch - a Bioconductor Package for the Efficient Pre-Processing of GC-MS Metabolite Profiling Data.” <em>BMC Bioinformatics</em> 10: 428. doi:<a href="https://doi.org/10.1186/1471-2105-10-428">10.1186/1471-2105-10-428</a>.</p>
</div>
<div id="ref-Durinck2005">
<p>Durinck, S, Y Moreau, A Kasprzyk, S Davis, B De Moor, A Brazma, and W Huber. 2005. “BioMart and Bioconductor: A Powerful Link Between Biological Databases and Microarray Data Analysis.” <em>Bioinformatics</em> 21 (16): 3439–40. doi:<a href="https://doi.org/10.1093/bioinformatics/bti525">10.1093/bioinformatics/bti525</a>.</p>
</div>
<div id="ref-Gatto2012">
<p>Gatto, L, and K S Lilley. 2012. “MSnbase – an R/Bioconductor Package for Isobaric Tagged Mass Spectrometry Data Visualization, Processing and Quantitation.” <em>Bioinformatics</em> 28 (2): 288–9. doi:<a href="https://doi.org/10.1093/bioinformatics/btr645">10.1093/bioinformatics/btr645</a>.</p>
</div>
<div id="ref-Gatto:2015">
<p>Gatto, L, L M Breckels, T Naake, and S Gibb. 2015. “Visualisation of proteomics data using R and Bioconductor.” <em>Proteomics</em>, February. doi:<a href="https://doi.org/10.1002/pmic.201400392">10.1002/pmic.201400392</a>.</p>
</div>
<div id="ref-R4Prot2013">
<p>Gatto, Laurent, and Andy Christoforou. 2013. “Using R and Bioconductor for Proteomics Data Analysis.” <em>BBA - Proteins and Proteomics</em>. doi:<a href="https://doi.org/10.1016/j.bbapap.2013.04.032">10.1016/j.bbapap.2013.04.032</a>.</p>
</div>
<div id="ref-Gentleman2004">
<p>Gentleman, Robert C., Vincent J. Carey, Douglas M. Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. “Bioconductor: Open Software Development for Computational Biology and Bioinformatics.” <em>Genome Biol</em> 5 (10): –80. doi:<a href="https://doi.org/10.1186/gb-2004-5-10-r80">10.1186/gb-2004-5-10-r80</a>.</p>
</div>
<div id="ref-Gibb2012">
<p>Gibb, S, and K Strimmer. 2012. “MALDIquant: A Versatile R Package for the Analysis of Mass Spectrometry Data.” <em>Bioinformatics</em> 28 (17): 2270–1. doi:<a href="https://doi.org/10.1093/bioinformatics/bts447">10.1093/bioinformatics/bts447</a>.</p>
</div>
<div id="ref-Kim:2008go">
<p>Kim, Sangtae, Nitin Gupta, and Pavel A Pevzner. 2008. “Spectral probabilities and generating functions of tandem mass spectra: a strike against decoy databases.” <em>Journal of Proteome Research</em> 7 (8): 3354–63.</p>
</div>
<div id="ref-Piehowski2013a">
<p>Piehowski, Paul D., Vladislav A. Petyuk, John D. Sandoval, Kristin E. Burnum, Gary R. Kiebel, Matthew E. Monroe, Gordon A. Anderson, David G Camp 2nd, and Richard D. Smith. 2013. “STEPS: A Grid Search Methodology for Optimized Peptide Identification Filtering of Ms/Ms Database Search Results.” <em>Proteomics</em> 13 (5). Biological Sciences Division; Environmental Molecular Sciences Laboratory, Pacific Northwest National Laboratory, Richland, WA99352, USA.: 766–70. doi:<a href="https://doi.org/10.1002/pmic.201200096">10.1002/pmic.201200096</a>.</p>
</div>
<div id="ref-Smith2006">
<p>Smith, C A, E J Want, G O’Maille, R Abagyan, and G Siuzdak. 2006. “XCMS: Processing Mass Spectrometry Data for Metabolite Profiling Using Nonlinear Peak Alignment, Matching, and Identification.” <em>Anal Chem</em> 78 (3): 779–87. doi:<a href="https://doi.org/10.1021/ac051437y">10.1021/ac051437y</a>.</p>
</div>
<div id="ref-Uhlen2005">
<p>Uhlén, Mathias, Erik Björling, Charlotta Agaton, Cristina Al-Khalili A. Szigyarto, Bahram Amini, Elisabet Andersen, Ann-Catrin C. Andersson, et al. 2005. “A human protein atlas for normal and cancer tissues based on antibody proteomics.” <em>Molecular &amp; Cellular Proteomics : MCP</em> 4 (12). Department of Biotechnology, AlbaNova University Center, Royal Institute of Technology (KTH), SE-106 91 Stockholm, Sweden. mathias.uhlen@biotech.kth.se: 1920–32. doi:<a href="https://doi.org/10.1074/mcp.M500279-MCP200">10.1074/mcp.M500279-MCP200</a>.</p>
</div>
<div id="ref-Vizcaino2014">
<p>Vizcaino, Juan A, Eric W Deutsch, Rui Wang, Attila Csordas, Florian Reisinger, Daniel Rios, Jose A Dianes, et al. 2014. “ProteomeXchange Provides Globally Coordinated Proteomics Data Submission and Dissemination.” <em>Nat Biotech</em> 32 (3): 223–26. doi:<a href="https://doi.org/10.1038/nbt.2839">10.1038/nbt.2839</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Here, we specify <code>mzTab</code> format version 0.9. Recent files have been generated according to the latest specifications, version 1.0, and the <code>version</code> does not need to be specified explicitly.<a href="#fnref1">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#sec:intro">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#sec:rres">General R resources</a></li>
      <li><a href="#sec:biocres">Bioconductor resources</a></li>
      <li><a href="#getting-help">Getting help</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#external-dependencies">External dependencies</a></li>
      <li><a href="#obtaining-the-code">Obtaining the code</a></li>
      <li><a href="#prepare-the-working-environment">Prepare the working environment</a></li>
      </ul>
</li>
      <li>
<a href="#data-standards-and-inputoutput">Data standards and input/output</a><ul class="nav nav-pills nav-stacked">
<li><a href="#sec:mzr">The mzR package</a></li>
      <li><a href="#subsec:mzID">Handling MS2 identification data with <em><a href="http://bioconductor-org/packages/mzID">mzID</a></em></a></li>
      </ul>
</li>
      <li>
<a href="#raw-data-abstraction-with-msnexp-objects">Raw data abstraction with <code>MSnExp</code> objects</a><ul class="nav nav-pills nav-stacked">
<li><a href="#mgf-readwrite-support"><code>mgf</code> read/write support</a></li>
      </ul>
</li>
      <li>
<a href="#quantitative-proteomics">Quantitative proteomics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-mztab-format">The <code>mzTab</code> format</a></li>
      <li><a href="#third-party-data">Third-party data</a></li>
      <li><a href="#working-with-raw-data">Working with raw data</a></li>
      <li><a href="#the-maldiquant-package">The <em><a href="https://CRAN-R-project-org/package=MALDIquant">MALDIquant</a></em> package</a></li>
      <li><a href="#working-with-peptide-sequences">Working with peptide sequences</a></li>
      <li><a href="#the-isobar-package">The <em><a href="http://bioconductor-org/packages/isobar">isobar</a></em> package</a></li>
      <li><a href="#the-dep-package">The <em><a href="http://bioconductor-org/packages/DEP">DEP</a></em> package</a></li>
      <li><a href="#the-synapter-package">The <em><a href="http://bioconductor-org/packages/synapter">synapter</a></em> package</a></li>
      </ul>
</li>
      <li>
<a href="#ms2-spectra-identification">MS2 spectra identification</a><ul class="nav nav-pills nav-stacked">
<li><a href="#x-tandem">X! Tandem</a></li>
      <li><a href="#ms-gf">MS-GF+</a></li>
      <li><a href="#post-search-filtering-of-msms-ids-using-msnid">Post-search Filtering of MS/MS IDs Using <em><a href="http://bioconductor-org/packages/MSnID">MSnID</a></em></a></li>
      </ul>
</li>
      <li><a href="#sec:qc">Quality control</a></li>
      <li><a href="#sec:annot">Annotation</a></li>
      <li>
<a href="#sec:packages">Other packages</a><ul class="nav nav-pills nav-stacked">
<li><a href="#bioconductor-packages">Bioconductor packages</a></li>
      <li><a href="#other-cran-packages">Other CRAN packages</a></li>
      </ul>
</li>
      <li><a href="#sec:sessionInfo">Session information</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
