<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualisation of proteomics data using R and Bioconductor • RforProteomics</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">RforProteomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="RProtVis_files/htmlwidgets-0.9/htmlwidgets.js"></script><script src="RProtVis_files/jquery-1.12.4/jquery.min.js"></script><script src="RProtVis_files/datatables-binding-0.2/datatables.js"></script><link href="RProtVis_files/dt-core-1.10.12/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="RProtVis_files/dt-core-1.10.12/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="RProtVis_files/dt-core-1.10.12/js/jquery.dataTables.min.js"></script><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Visualisation of proteomics data using R and Bioconductor</h1>
                        <h4 class="author">Laurent Gatto</h4>
                                    <h4 class="author">Lisa Breckels</h4>
                                    <h4 class="author">Sebastian Gibb</h4>
                        
          </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This is the companion vignette to the ‘Using R and Bioconductor for proteomics data analysis’ manuscript that presents an overview of R and Bioconductor software for mass spectrometry and proteomics data. It provides the code to reproduce the figures in the paper.</p>
    </div>
    
<div class="contents">
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette illustrates existing  and Bioconductor infrastructure for the visualisation of mass spectrometry and proteomics data. The code details the visualisations presented in</p>
<blockquote>
<p>Gatto L, Breckels LM, Naake T, Gibb S. Visualisation of proteomics data using R and Bioconductor. Proteomics. 2015 Feb 18. doi: 10.1002/pmic.201400392. PubMed PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25690415">25690415</a>.</p>
</blockquote>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ul>
<li>CRAN Task View: Graphic Displays &amp; Dynamic Graphics &amp; Graphic Devices &amp; Visualization: <a href="http://cran.r-project.org/web/views/Graphics.html" class="uri">http://cran.r-project.org/web/views/Graphics.html</a>
</li>
<li>CRAN Task View: Web Technologies and Services: <a href="http://cran.r-project.org/web/views/WebTechnologies.html" class="uri">http://cran.r-project.org/web/views/WebTechnologies.html</a>
</li>
<li>ggplot2 <a href="http://link.springer.com/book/10.1007%2F978-0-387-98141-3">book</a> (syntax is slightly outdated) (<a href="http://ggplot2.org/book/">code</a>), <a href="http://ggplot2.org/">web page</a> and <a href="http://docs.ggplot2.org/current/">on-line docs</a>
</li>
<li>lattice <a href="http://lmdvr.r-forge.r-project.org/figures/figures.html">book</a> and <a href="http://lattice.r-forge.r-project.org/">web page</a>
</li>
<li>
<em>R Graphics</em> <a href="https://www.stat.auckland.ac.nz/~paul/RG2e/">book</a>
</li>
<li>
<a href="http://www.cookbook-r.com/Graphs/">R Cookbook</a> and <a href="http://shop.oreilly.com/product/0636920023135.do">R Graphics Cookbook</a>
</li>
</ul>
</div>
<div id="relevant-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#relevant-packages" class="anchor"></a>Relevant packages</h2>
<p>There are currently <code>r nrow(pp)</code> <a href="http://bioconductor.org/packages/devel/BiocViews.html#___Proteomics">Proteomics</a> and <code>r nrow(msp)</code> <a href="http://bioconductor.org/packages/devel/BiocViews.html#___MassSpectrometry">MassSpectrometry</a> packages in Bioconductor version 3.7. Other non-Bioconductor packages are described in the <code>r Biocexptpkg("RforProteomics")</code> vignette (open it in R with <code><a href="../reference/RforProteomics.html">RforProteomics()</a></code> or read it <a href="file:///home/lg390/dev/00_github/RforProteomics/docs/articles/RforProteomics.html">online</a>.)</p>
<div id="htmlwidget-d580438cae4b440d32bc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d580438cae4b440d32bc">{"x":{"filter":"none","data":[["ASEB","bioassayR","biobroom","biosigner","BRAIN","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","customProDB","cydar","DAPAR","deltaGseg","DEP","Doscheda","drawProteins","EGSEA","fCI","fmcsR","GraphPAC","hpar","InterMineR","iPAC","IPPD","isobar","kimod","LPEadj","MAGeCKFlute","MassSpecWavelet","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","MSstats","MSstatsQC","mzID","mzR","omicRexposome","PAA","PAnnBuilder","Path2PPI","pathview","Pbase","PCpheno","PECA","pepXMLTab","PGA","pgca","phosphonormalizer","plgem","PLPE","ppiStats","proBAMr","PROcess","procoil","ProCoNA","proFIA","pRoloc","pRolocGUI","Prostar","prot2D","ProteomicsAnnotationHubData","proteoQC","ProtGenerics","Pviz","qcmetrics","QuartPAC","rain","RankProd","RCASPAR","Rchemcpp","Rcpi","readat","ropls","ROTS","RpsiXML","rpx","rTANDEM","sapFinder","ScISI","shinyTANDEM","SLGI","SpacePAC","specL","spliceSites","SWATH2stats","synapter","tofsims","topdownr","TPP"],["ASEB","bioassayR","biobroom","biosigner","BRAIN","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","customProDB","cydar","DAPAR","deltaGseg","DEP","Doscheda","drawProteins","EGSEA","fCI","fmcsR","GraphPAC","hpar","InterMineR","iPAC","IPPD","isobar","kimod","LPEadj","MAGeCKFlute","MassSpecWavelet","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","MSstats","MSstatsQC","mzID","mzR","omicRexposome","PAA","PAnnBuilder","Path2PPI","pathview","Pbase","PCpheno","PECA","pepXMLTab","PGA","pgca","phosphonormalizer","plgem","PLPE","ppiStats","proBAMr","PROcess","procoil","ProCoNA","proFIA","pRoloc","pRolocGUI","Prostar","prot2D","ProteomicsAnnotationHubData","proteoQC","ProtGenerics","Pviz","qcmetrics","QuartPAC","rain","RankProd","RCASPAR","Rchemcpp","Rcpi","readat","ropls","ROTS","RpsiXML","rpx","rTANDEM","sapFinder","ScISI","shinyTANDEM","SLGI","SpacePAC","specL","spliceSites","SWATH2stats","synapter","tofsims","topdownr","TPP"],["Predict Acetylated Lysine Sites","Cross-target analysis of small molecule bioactivity","Turn Bioconductor objects into tidy data frames","Signature discovery from omics data","Baffling Recursive Algorithm for Isotope distributioN calculations","A mass spectrometry imaging toolbox for statistical analysis","Causal network analysis methods","Training of boolean logic models of signalling networks using prior knowledge networks and perturbation data","R interface to a subset of OpenBabel functionalities","Cheminformatics Toolkit for R","Visualization and management of the protein-protein interaction networks.","Cleavage of Polypeptide Sequences","A package for the clinical proteomic profiling data analysis","Add-on to CellNOptR: Discretized time treatments","Integration of CellNOptR to add missing links","ODE add-on to CellNOptR","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Using Mass Cytometry for Differential Abundance Analyses","Tools for the Differential Analysis of Proteins Abundance with R","deltaGseg","Differential Enrichment analysis of Proteomics data","A DownStream Chemo-Proteomics Analysis Pipeline","Package to Draw Protein Schematics from Uniprot API output","Ensemble of Gene Set Enrichment Analyses","f-divergence Cutoff Index for Differential Expression Analysis in Transcriptomics and Proteomics","Mismatch Tolerant Maximum Common Substructure Searching","Identification of Mutational Clusters in Proteins via a Graph Theoretical Approach.","Human Protein Atlas in R","R Interface with InterMine-Powered Databases","Identification of Protein Amino acid Clustering","Isotopic peak pattern deconvolution for Protein Mass Spectrometry by template matching","Analysis and quantitation of isobarically tagged MSMS proteomics data","A k-tables approach to integrate multiple Omics-Data","A correction of the local pooled error (LPE) method to replace the asymptotic variance adjustment with an unbiased adjustment\nbased on sample size.","Integrative analysis pipeline for pooled CRISPR functional genetic screens","Mass spectrum processing by wavelet-based algorithms","A shiny GUI for MSGFplus","An interface between R and MS-GF+","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Longitudinal system suitability monitoring and quality control for proteomic experiments","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Exposome and omic data associatin and integration analysis","PAA (Protein Array Analyzer)","Protein annotation data package builder","Prediction of pathway-related protein-protein interaction networks","a tool set for pathway based data integration and visualization","Manipulating and exploring protein and proteomics data","Phenotypes and cellular organizational units","Probe-level Expression Change Averaging","Parsing pepXML files and filter based on peptide FDR.","An package for identification of novel peptides by customized database derived from RNA-Seq","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Compensates for the bias introduced by median normalization in phosphoproteomics","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Local Pooled Error Test for Differential Expression with Paired High-throughput Data","Protein-Protein Interaction Statistical Package","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Prediction of Oligomerization of Coiled Coil Proteins","Protein co-expression network analysis (ProCoNA).","Preprocessing of FIA-HRMS data","A unifying bioinformatics framework for spatial proteomics","Interactive visualisation of spatial proteomics data","Provides a GUI for DAPAR","Statistical Tools for volume data from 2D Gel Electrophoresis","Transform public proteomics data resources into Bioconductor Data Structures","An R package for proteomics data quality control","S4 generic functions for Bioconductor proteomics infrastructure","Peptide Annotation and Data Visualization using Gviz","A Framework for Quality Control","Identification of mutational clusters in protein quaternary structures.","Rhythmicity Analysis Incorporating Non-parametric Methods","Rank Product method for identifying differentially expressed genes with application in meta-analysis","A package for survival time prediction based on a piecewise baseline hazard Cox regression model.","Similarity measures for chemical compounds","Molecular Informatics Toolkit for Compound-Protein Interaction in Drug Discovery","Functionality to Read and Manipulate SomaLogic ADAT files","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Reproducibility-Optimized Test Statistic","R interface to PSI-MI 2.5 files","R Interface to the ProteomeXchange Repository","Interfaces the tandem protein identification algorithm in R","A package for variant peptides detection and visualization in shotgun proteomics.","In Silico Interactome","Provides a GUI for rTANDEM","Synthetic Lethal Genetic Interaction","Identification of Mutational Clusters in 3D Protein Space via Simulation.","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","A bioconductor package for exploration of alignment gap positions from RNA-seq data","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Import, process and analysis of Time-of-Flight Secondary Ion Mass Spectrometry (ToF-SIMS) imaging data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments"],["1.23.0","1.17.0","1.11.0","1.7.0","1.25.0","1.11.2","1.11.0","1.25.1","1.17.2","2.31.1","1.19.0","1.17.0","1.29.0","1.21.0","1.19.0","1.21.0","1.19.0","1.3.1","1.11.5","1.19.0","1.1.2","1.1.0","0.99.9","1.7.1","1.9.0","1.21.0","1.21.0","1.21.0","1.1.0","1.23.0","1.27.0","1.25.0","1.7.0","1.39.0","0.99.17","1.45.0","1.13.0","1.13.0","1.17.0","1.17.0","2.5.3","1.13.0","3.11.0","1.1.1","1.17.0","2.13.3","1.1.1","1.13.0","1.43.0","1.9.0","1.19.0","0.19.0","1.41.0","1.15.0","1.13.0","1.9.0","1.3.0","1.3.0","1.51.0","1.39.0","1.45.0","1.13.0","1.55.0","2.7.0","1.17.0","1.5.0","1.19.0","1.13.0","1.11.5","1.17.0","1.9.0","1.15.0","1.11.0","1.13.0","1.17.0","1.11.0","1.13.0","3.5.0","1.25.0","2.17.0","1.15.1","1.5.0","1.11.0","1.7.4","2.21.0","1.15.0","1.19.0","1.17.0","1.51.0","1.17.0","1.39.0","1.17.0","1.13.3","1.27.1","1.9.1","2.3.0","1.7.0","1.1.4","3.7.1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]},"selection":{"mode":"multiple","selected":null,"target":"row"}},"evals":[],"jsHooks":[]}</script><div id="htmlwidget-b1289b362102f368090e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b1289b362102f368090e">{"x":{"filter":"none","data":[["apComplex","BRAIN","CAMERA","Cardinal","CATALYST","cosmiq","customProDB","DAPAR","DEP","Doscheda","flagme","gaga","iontree","IPO","isobar","LOBSTAHS","MAIT","MassArray","MassSpecWavelet","Metab","metabomxtr","metaMS","MetCirc","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","msPurity","MSstats","MSstatsQC","mzID","mzR","PAPi","Pbase","pepXMLTab","PGA","pgca","plgem","proBAMr","PROcess","proFIA","pRoloc","Prostar","proteoQC","ProtGenerics","qcmetrics","Rdisop","Risa","RMassBank","rols","ropls","rpx","rTANDEM","sapFinder","shinyTANDEM","SIMAT","specL","statTarget","SWATH2stats","synapter","TargetSearch","tofsims","topdownr","TPP","xcms","yamss"],["apComplex","BRAIN","CAMERA","Cardinal","CATALYST","cosmiq","customProDB","DAPAR","DEP","Doscheda","flagme","gaga","iontree","IPO","isobar","LOBSTAHS","MAIT","MassArray","MassSpecWavelet","Metab","metabomxtr","metaMS","MetCirc","MSGFgui","MSGFplus","msmsEDA","msmsTests","MSnbase","MSnID","msPurity","MSstats","MSstatsQC","mzID","mzR","PAPi","Pbase","pepXMLTab","PGA","pgca","plgem","proBAMr","PROcess","proFIA","pRoloc","Prostar","proteoQC","ProtGenerics","qcmetrics","Rdisop","Risa","RMassBank","rols","ropls","rpx","rTANDEM","sapFinder","shinyTANDEM","SIMAT","specL","statTarget","SWATH2stats","synapter","TargetSearch","tofsims","topdownr","TPP","xcms","yamss"],["Estimate protein complex membership using AP-MS protein data","Baffling Recursive Algorithm for Isotope distributioN calculations","Collection of annotation related methods for mass spectrometry data","A mass spectrometry imaging toolbox for statistical analysis","Cytometry dATa anALYSis Tools","cosmiq - COmbining Single Masses Into Quantities","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Tools for the Differential Analysis of Proteins Abundance with R","Differential Enrichment analysis of Proteomics data","A DownStream Chemo-Proteomics Analysis Pipeline","Analysis of Metabolomics GC/MS Data","GaGa hierarchical model for high-throughput data analysis","Data management and analysis of ion trees from ion-trap mass spectrometry","Automated Optimization of XCMS Data Processing parameters","Analysis and quantitation of isobarically tagged MSMS proteomics data","Lipid and Oxylipin Biomarker Screening through Adduct Hierarchy Sequences","Statistical Analysis of Metabolomic Data","Analytical Tools for MassArray Data","Mass spectrum processing by wavelet-based algorithms","Metab: An R Package for a High-Throughput Analysis of Metabolomics Data Generated by GC-MS.","A package to run mixture models for truncated metabolomics data with normal or lognormal distributions","MS-based metabolomics annotation pipeline","Navigating mass spectral similarity in high-resolution MS/MS metabolomics data","A shiny GUI for MSGFplus","An interface between R and MS-GF+","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Automated Evaluation of Precursor Ion Purity for Mass Spectrometry Based Fragmentation in Metabolomics","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Longitudinal system suitability monitoring and quality control for proteomic experiments","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Predict metabolic pathway activity based on metabolomics data","Manipulating and exploring protein and proteomics data","Parsing pepXML files and filter based on peptide FDR.","An package for identification of novel peptides by customized database derived from RNA-Seq","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Preprocessing of FIA-HRMS data","A unifying bioinformatics framework for spatial proteomics","Provides a GUI for DAPAR","An R package for proteomics data quality control","S4 generic functions for Bioconductor proteomics infrastructure","A Framework for Quality Control","Decomposition of Isotopic Patterns","Converting experimental metadata from ISA-tab into Bioconductor data structures","Workflow to process tandem MS files and build MassBank records","An R interface to the Ontology Lookup Service","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","R Interface to the ProteomeXchange Repository","Interfaces the tandem protein identification algorithm in R","A package for variant peptides detection and visualization in shotgun proteomics.","Provides a GUI for rTANDEM","GC-SIM-MS data processing and alaysis tool","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Statistical Analysis of Metabolite Profile","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","A package for the analysis of GC-MS metabolite profiling data","Import, process and analysis of Time-of-Flight Secondary Ion Mass Spectrometry (ToF-SIMS) imaging data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","LC/MS and GC/MS Data Analysis","Tools for high-throughput metabolomics"],["2.45.0","1.25.0","1.35.0","1.11.2","1.3.0","1.13.0","1.19.0","1.11.5","1.1.2","1.1.0","1.35.0","2.25.0","1.23.1","1.5.1","1.25.0","1.5.0","1.13.0","1.31.0","1.45.0","1.13.0","1.13.0","1.15.0","1.7.0","1.13.0","1.13.0","1.17.0","1.17.0","2.5.3","1.13.0","1.5.0","3.11.0","1.1.1","1.17.0","2.13.3","1.19.0","0.19.0","1.13.0","1.9.0","1.3.0","1.51.0","1.13.0","1.55.0","1.5.0","1.19.0","1.11.5","1.15.0","1.11.0","1.17.0","1.39.0","1.21.0","2.7.0","2.7.2","1.11.0","1.15.0","1.19.0","1.17.0","1.17.0","1.11.0","1.13.3","1.9.0","1.9.1","2.3.0","1.35.0","1.7.0","1.1.4","3.7.1","3.1.0","1.5.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]},"selection":{"mode":"multiple","selected":null,"target":"row"}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="ascombes-quartet" class="section level1">
<h1 class="hasAnchor">
<a href="#ascombes-quartet" class="anchor"></a>Ascombe’s quartet</h1>
<table class="table">
<thead><tr>
<th style="text-align:right;">
x1
</th>
<th style="text-align:right;">
x2
</th>
<th style="text-align:right;">
x3
</th>
<th style="text-align:right;">
x4
</th>
<th style="text-align:right;">
y1
</th>
<th style="text-align:right;">
y2
</th>
<th style="text-align:right;">
y3
</th>
<th style="text-align:right;">
y4
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.04
</td>
<td style="text-align:right;">
9.14
</td>
<td style="text-align:right;">
7.46
</td>
<td style="text-align:right;">
6.58
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
6.95
</td>
<td style="text-align:right;">
8.14
</td>
<td style="text-align:right;">
6.77
</td>
<td style="text-align:right;">
5.76
</td>
</tr>
<tr>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7.58
</td>
<td style="text-align:right;">
8.74
</td>
<td style="text-align:right;">
12.74
</td>
<td style="text-align:right;">
7.71
</td>
</tr>
<tr>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.81
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
7.11
</td>
<td style="text-align:right;">
8.84
</td>
</tr>
<tr>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.33
</td>
<td style="text-align:right;">
9.26
</td>
<td style="text-align:right;">
7.81
</td>
<td style="text-align:right;">
8.47
</td>
</tr>
<tr>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
9.96
</td>
<td style="text-align:right;">
8.10
</td>
<td style="text-align:right;">
8.84
</td>
<td style="text-align:right;">
7.04
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7.24
</td>
<td style="text-align:right;">
6.13
</td>
<td style="text-align:right;">
6.08
</td>
<td style="text-align:right;">
5.25
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
4.26
</td>
<td style="text-align:right;">
3.10
</td>
<td style="text-align:right;">
5.39
</td>
<td style="text-align:right;">
12.50
</td>
</tr>
<tr>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
10.84
</td>
<td style="text-align:right;">
9.13
</td>
<td style="text-align:right;">
8.15
</td>
<td style="text-align:right;">
5.56
</td>
</tr>
<tr>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
4.82
</td>
<td style="text-align:right;">
7.26
</td>
<td style="text-align:right;">
6.42
</td>
<td style="text-align:right;">
7.91
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
5.68
</td>
<td style="text-align:right;">
4.74
</td>
<td style="text-align:right;">
5.73
</td>
<td style="text-align:right;">
6.89
</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dv">5</span>, <span class="dv">4</span>)
<span class="kw">colnames</span>(tab) &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">4</span>
<span class="kw">rownames</span>(tab) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"var(x)"</span>, <span class="st">"mean(x)"</span>,
                   <span class="st">"var(y)"</span>, <span class="st">"mean(y)"</span>,
                   <span class="st">"cor(x,y)"</span>)

<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>)
    tab[, i] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">var</span>(anscombe[, i]),
                  <span class="kw">mean</span>(anscombe[, i]),
                  <span class="kw">var</span>(anscombe[, i<span class="op">+</span><span class="dv">4</span>]),
                  <span class="kw">mean</span>(anscombe[, i<span class="op">+</span><span class="dv">4</span>]),
                  <span class="kw">cor</span>(anscombe[, i], anscombe[, i<span class="op">+</span><span class="dv">4</span>]))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>var(x)</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
</tr>
<tr class="even">
<td>mean(x)</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
</tr>
<tr class="odd">
<td>var(y)</td>
<td align="right">4.1272691</td>
<td align="right">4.1276291</td>
<td align="right">4.1226200</td>
<td align="right">4.1232491</td>
</tr>
<tr class="even">
<td>mean(y)</td>
<td align="right">7.5009091</td>
<td align="right">7.5009091</td>
<td align="right">7.5000000</td>
<td align="right">7.5009091</td>
</tr>
<tr class="odd">
<td>cor(x,y)</td>
<td align="right">0.8164205</td>
<td align="right">0.8162365</td>
<td align="right">0.8162867</td>
<td align="right">0.8165214</td>
</tr>
</tbody>
</table>
<p>While the residuals of the linear regression clearly indicate fundamental differences in these data, the most simple and straightforward approach is visualisation to highlight the fundamental differences in the datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ff &lt;-<span class="st"> </span>y <span class="op">~</span><span class="st"> </span>x

mods &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">as.list</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>), <span class="kw">paste0</span>(<span class="st">"lm"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>))

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>) {
    ff[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>] &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">"y"</span>,<span class="st">"x"</span>), i), as.name)
    <span class="kw">plot</span>(ff, <span class="dt">data =</span> anscombe, <span class="dt">pch =</span> <span class="dv">19</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">19</span>), <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">13</span>))
    mods[[i]] &lt;-<span class="st"> </span><span class="kw">lm</span>(ff, <span class="dt">data =</span> anscombe)
    <span class="kw">abline</span>(mods[[i]])
}</code></pre></div>
<p><img src="RProtVis_files/figure-html/anscombefig-1.png" width="672"></p>
<table class="table">
<thead><tr class="header">
<th align="right">lm1</th>
<th align="right">lm2</th>
<th align="right">lm3</th>
<th align="right">lm4</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0.0390000</td>
<td align="right">1.1390909</td>
<td align="right">-0.5397273</td>
<td align="right">-0.421</td>
</tr>
<tr class="even">
<td align="right">-0.0508182</td>
<td align="right">1.1390909</td>
<td align="right">-0.2302727</td>
<td align="right">-1.241</td>
</tr>
<tr class="odd">
<td align="right">-1.9212727</td>
<td align="right">-0.7609091</td>
<td align="right">3.2410909</td>
<td align="right">0.709</td>
</tr>
<tr class="even">
<td align="right">1.3090909</td>
<td align="right">1.2690909</td>
<td align="right">-0.3900000</td>
<td align="right">1.839</td>
</tr>
<tr class="odd">
<td align="right">-0.1710909</td>
<td align="right">0.7590909</td>
<td align="right">-0.6894545</td>
<td align="right">1.469</td>
</tr>
<tr class="even">
<td align="right">-0.0413636</td>
<td align="right">-1.9009091</td>
<td align="right">-1.1586364</td>
<td align="right">0.039</td>
</tr>
<tr class="odd">
<td align="right">1.2393636</td>
<td align="right">0.1290909</td>
<td align="right">0.0791818</td>
<td align="right">-1.751</td>
</tr>
<tr class="even">
<td align="right">-0.7404545</td>
<td align="right">-1.9009091</td>
<td align="right">0.3886364</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="right">1.8388182</td>
<td align="right">0.1290909</td>
<td align="right">-0.8491818</td>
<td align="right">-1.441</td>
</tr>
<tr class="even">
<td align="right">-1.6807273</td>
<td align="right">0.7590909</td>
<td align="right">-0.0805455</td>
<td align="right">0.909</td>
</tr>
<tr class="odd">
<td align="right">0.1794545</td>
<td align="right">-0.7609091</td>
<td align="right">0.2289091</td>
<td align="right">-0.111</td>
</tr>
</tbody>
</table>
</div>
<div id="the-ma-plot-example" class="section level1">
<h1 class="hasAnchor">
<a href="#the-ma-plot-example" class="anchor"></a>The MA plot example</h1>
<p>The following code chunk connects to the <code>PXD000001</code> data set on the ProteomeXchange repository and fetches the <code>mzTab</code> file. After missing values filtering, we extract relevant data (log2 fold-changes and log10 mean expression intensities) into <code>data.frames</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"rpx"</span>)
px1 &lt;-<span class="st"> </span><span class="kw">PXDataset</span>(<span class="st">"PXD000001"</span>)
mztab &lt;-<span class="st"> </span><span class="kw">pxget</span>(px1, <span class="st">"PXD000001_mztab.txt"</span>)</code></pre></div>
<pre><code>## Downloading 1 file</code></pre>
<pre><code>## /home/lg390/dev/RforProteomics/docs/articles/PXD000001_mztab.txt already present.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MSnbase"</span>)
## here, we need to specify the (old) mzTab version 0.9
qnt &lt;-<span class="st"> </span><span class="kw">readMzTabData</span>(mztab, <span class="dt">what =</span> <span class="st">"PEP"</span>, <span class="dt">version =</span> <span class="st">"0.9"</span>)
<span class="kw">sampleNames</span>(qnt) &lt;-<span class="st"> </span><span class="kw">reporterNames</span>(TMT6)
qnt &lt;-<span class="st"> </span><span class="kw">filterNA</span>(qnt)
## may be combineFeatuers

spikes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"P02769"</span>, <span class="st">"P00924"</span>, <span class="st">"P62894"</span>, <span class="st">"P00489"</span>)
protclasses &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">fData</span>(qnt)<span class="op">$</span>accession)
protclasses[<span class="op">!</span>protclasses <span class="op">%in%</span><span class="st"> </span>spikes] &lt;-<span class="st"> "Background"</span>


madata42 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">A =</span> <span class="kw">rowMeans</span>(<span class="kw">log</span>(<span class="kw">exprs</span>(qnt[, <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">2</span>)]), <span class="dv">10</span>)),
                       <span class="dt">M =</span> <span class="kw">log</span>(<span class="kw">exprs</span>(qnt)[, <span class="dv">4</span>], <span class="dv">2</span>) <span class="op">-</span><span class="st"> </span><span class="kw">log</span>(<span class="kw">exprs</span>(qnt)[, <span class="dv">2</span>], <span class="dv">2</span>),
                       <span class="dt">data =</span> <span class="kw">rep</span>(<span class="st">"4vs2"</span>, <span class="kw">nrow</span>(qnt)),
                       <span class="dt">protein =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>accession,
                       <span class="dt">class =</span> protclasses)

madata62 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">A =</span> <span class="kw">rowMeans</span>(<span class="kw">log</span>(<span class="kw">exprs</span>(qnt[, <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">2</span>)]), <span class="dv">10</span>)),
                       <span class="dt">M =</span> <span class="kw">log</span>(<span class="kw">exprs</span>(qnt)[, <span class="dv">6</span>], <span class="dv">2</span>) <span class="op">-</span><span class="st"> </span><span class="kw">log</span>(<span class="kw">exprs</span>(qnt)[, <span class="dv">2</span>], <span class="dv">2</span>),
                       <span class="dt">data =</span> <span class="kw">rep</span>(<span class="st">"6vs2"</span>, <span class="kw">nrow</span>(qnt)),
                       <span class="dt">protein =</span> <span class="kw">fData</span>(qnt)<span class="op">$</span>accession,
                       <span class="dt">class =</span> protclasses)


madata &lt;-<span class="st"> </span><span class="kw">rbind</span>(madata42, madata62)</code></pre></div>
<div id="the-traditional-plotting-system" class="section level2">
<h2 class="hasAnchor">
<a href="#the-traditional-plotting-system" class="anchor"></a>The traditional plotting system</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plot</span>(M <span class="op">~</span><span class="st"> </span>A, <span class="dt">data =</span> madata42, <span class="dt">main =</span> <span class="st">"4vs2"</span>,
     <span class="dt">xlab =</span> <span class="st">"A"</span>, <span class="dt">ylab =</span> <span class="st">"M"</span>, <span class="dt">col =</span> madata62<span class="op">$</span>class)
<span class="kw">plot</span>(M <span class="op">~</span><span class="st"> </span>A, <span class="dt">data =</span> madata62, <span class="dt">main =</span> <span class="st">"6vs2"</span>,
     <span class="dt">xlab =</span> <span class="st">"A"</span>, <span class="dt">ylab =</span> <span class="st">"M"</span>, <span class="dt">col =</span> madata62<span class="op">$</span>class)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig1-1.png" width="672"></p>
</div>
<div id="lattice" class="section level2">
<h2 class="hasAnchor">
<a href="#lattice" class="anchor"></a>lattice</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"lattice"</span>)
latma &lt;-<span class="st"> </span><span class="kw">xyplot</span>(M <span class="op">~</span><span class="st"> </span>A <span class="op">|</span><span class="st"> </span>data, <span class="dt">data =</span> madata,
                <span class="dt">groups =</span> madata<span class="op">$</span>class,
                <span class="dt">auto.key =</span> <span class="ot">TRUE</span>)
<span class="kw">print</span>(latma)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig2-1.png" width="672"></p>
</div>
<div id="ggplot2" class="section level2">
<h2 class="hasAnchor">
<a href="#ggplot2" class="anchor"></a>ggplot2</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"ggplot2"</span>)
ggma &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> A, <span class="dt">y =</span> M, <span class="dt">colour =</span> class), <span class="dt">data =</span> madata,
               <span class="dt">colour =</span> class) <span class="op">+</span>
<span class="st">                   </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st">                       </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>data)
<span class="kw">print</span>(ggma)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig3-1.png" width="672"></p>
</div>
<div id="customization" class="section level2">
<h2 class="hasAnchor">
<a href="#customization" class="anchor"></a>Customization</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"RColorBrewer"</span>)
bcols &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">"Set1"</span>)
cls &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"Background"</span> =<span class="st"> "#12121230"</span>,
         <span class="st">"P02769"</span> =<span class="st"> </span>bcols[<span class="dv">1</span>],
         <span class="st">"P00924"</span> =<span class="st"> </span>bcols[<span class="dv">2</span>],
         <span class="st">"P62894"</span> =<span class="st"> </span>bcols[<span class="dv">3</span>],
         <span class="st">"P00489"</span> =<span class="st"> </span>bcols[<span class="dv">4</span>])</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ggma2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> A, <span class="dt">y =</span> M, <span class="dt">colour =</span> class),
                <span class="dt">data =</span> madata) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="dt">shape =</span> <span class="dv">19</span>) <span class="op">+</span>
<span class="st">                    </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>data) <span class="op">+</span><span class="st"> </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> cls) <span class="op">+</span>
<span class="st">                        </span><span class="kw">guides</span>(<span class="dt">colour =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">alpha =</span> <span class="dv">1</span>)))
<span class="kw">print</span>(ggma2)</code></pre></div>
<p><img src="RProtVis_files/figure-html/macust-1.png" width="672"></p>
</div>
<div id="the-maplot-method-for-msnset-instances" class="section level2">
<h2 class="hasAnchor">
<a href="#the-maplot-method-for-msnset-instances" class="anchor"></a>The <code>MAplot</code> method for <code>MSnSet</code> instances</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">MAplot</span>(qnt, <span class="dt">cex =</span> .<span class="dv">8</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mafigmsnset-1.png" width="672"></p>
</div>
<div id="an-interactive-shiny-app-for-ma-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#an-interactive-shiny-app-for-ma-plots" class="anchor"></a>An interactive <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app for MA plots</h2>
<p>This app is based on Mike Love’s <a href="https://github.com/mikelove/shinyMA">shinyMA</a> application, adapted for a proteomics data. A screen shot is displayed below. To start the application:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/shinyMA.html">shinyMA</a></span>()</code></pre></div>
<div class="figure">
<img src="figures/shinyMA.png" alt="shinyMA screeshot"><p class="caption">shinyMA screeshot</p>
</div>
<p>The application is also available online at <a href="https://lgatto.shinyapps.io/shinyMA/" class="uri">https://lgatto.shinyapps.io/shinyMA/</a>.</p>
<p>See the excellent <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> <a href="http://shiny.rstudio.com/">web page</a> for tutorials.</p>
</div>
<div id="volcano-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#volcano-plots" class="anchor"></a>Volcano plots</h2>
<p>Below, using the <em><a href="http://bioconductor.org/packages/msmsTest">msmsTest</a></em> package, we load a example <code>MSnSet</code> data with spectral couting data (from the <code>r Biocpkg("msmsEDA")</code> package) and run a statistical test to obtain (adjusted) p-values and fold-changes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"msmsEDA"</span>)
<span class="kw">library</span>(<span class="st">"msmsTests"</span>)
<span class="kw">data</span>(msms.dataset)
## Pre-process expression matrix
e &lt;-<span class="st"> </span><span class="kw">pp.msms.data</span>(msms.dataset)
## Models and normalizing condition
null.f &lt;-<span class="st"> "y~batch"</span>
alt.f &lt;-<span class="st"> "y~treat+batch"</span>
div &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">exprs</span>(e), <span class="dv">2</span>, sum)
## Test
res &lt;-<span class="st"> </span><span class="kw">msms.glm.qlll</span>(e, alt.f, null.f, <span class="dt">div =</span> div)
lst &lt;-<span class="st"> </span><span class="kw">test.results</span>(res, e, <span class="kw">pData</span>(e)<span class="op">$</span>treat, <span class="st">"U600"</span>, <span class="st">"U200 "</span>, div,
                    <span class="dt">alpha =</span> <span class="fl">0.05</span>, <span class="dt">minSpC =</span> <span class="dv">2</span>, <span class="dt">minLFC =</span> <span class="kw">log2</span>(<span class="fl">1.8</span>),
                    <span class="dt">method =</span> <span class="st">"BH"</span>)</code></pre></div>
<p>Here, we produce the volcano plot by hand, with the <code>plot</code> function. In the second plot, we limit the x axis limits and add grid lines.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(lst<span class="op">$</span>tres<span class="op">$</span>LogFC, <span class="op">-</span><span class="kw">log10</span>(lst<span class="op">$</span>tres<span class="op">$</span>p.value))</code></pre></div>
<p><img src="RProtVis_files/figure-html/volc1-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(lst<span class="op">$</span>tres<span class="op">$</span>LogFC, <span class="op">-</span><span class="kw">log10</span>(lst<span class="op">$</span>tres<span class="op">$</span>p.value),
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>, <span class="dv">3</span>))
<span class="kw">grid</span>()</code></pre></div>
<p><img src="RProtVis_files/figure-html/volc1-2.png" width="672"></p>
<p>Below, we use the <code>res.volcanoplot</code> function from the <code>r Biocpkg("msmsTests")</code> package. This functions uses the sample annotation stored with the quantitative data in the <code>MSnSet</code> object to colour the samples according to their phenotypes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Plot
<span class="kw">res.volcanoplot</span>(lst<span class="op">$</span>tres,
                <span class="dt">max.pval =</span> <span class="fl">0.05</span>,
                <span class="dt">min.LFC =</span> <span class="dv">1</span>,
                <span class="dt">maxx =</span> <span class="dv">3</span>,
                <span class="dt">maxy =</span> <span class="ot">NULL</span>,
                <span class="dt">ylbls =</span> <span class="dv">4</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/volc2-1.png" width="672"></p>
</div>
<div id="a-pca-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#a-pca-plot" class="anchor"></a>A PCA plot</h2>
<p>Using the <code>counts.pca</code> function from the <em><a href="http://bioconductor.org/packages/msmsEDA">msmsEDA</a></em> package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"msmsEDA"</span>)
<span class="kw">data</span>(msms.dataset)
msnset &lt;-<span class="st"> </span><span class="kw">pp.msms.data</span>(msms.dataset)
lst &lt;-<span class="st"> </span><span class="kw">counts.pca</span>(msnset, <span class="dt">wait =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/msmsedapca-1.png" width="672"><img src="RProtVis_files/figure-html/msmsedapca-2.png" width="672"></p>
<p>It is also possible to generate the PCA data using the <code>prcomp</code>. Below, we extract the coordinates of PC1 and PC2 from the <code>counts.pca</code> result and plot them using the <code>plot</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pcadata &lt;-<span class="st"> </span>lst<span class="op">$</span>pca<span class="op">$</span>x[, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="kw">head</span>(pcadata)</code></pre></div>
<pre><code>##                  PC1       PC2
## U2.2502.1 -120.26080 -53.55270
## U2.2502.2  -99.90618 -53.89979
## U2.2502.3 -127.35928 -49.29906
## U2.2502.4 -166.04611 -39.27557
## U6.2502.1 -127.18423  37.11614
## U6.2502.2 -117.97016  47.03702</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(pcadata[, <span class="dv">1</span>], pcadata[, <span class="dv">2</span>],
     <span class="dt">xlab =</span> <span class="st">"PCA1"</span>, <span class="dt">ylab =</span> <span class="st">"PCA2"</span>)
<span class="kw">grid</span>()</code></pre></div>
<p><img src="RProtVis_files/figure-html/pca-1.png" width="672"></p>
</div>
</div>
<div id="plotting-with-r" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-with-r" class="anchor"></a>Plotting with R</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">kable</span>(plotfuns)</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">plot type</th>
<th align="left">traditional</th>
<th align="left">lattice</th>
<th align="left">ggplot2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">scatterplots</td>
<td align="left">plot</td>
<td align="left">xyplot</td>
<td align="left">geom_point</td>
</tr>
<tr class="even">
<td align="left">histograms</td>
<td align="left">hist</td>
<td align="left">histgram</td>
<td align="left">geom_histogram</td>
</tr>
<tr class="odd">
<td align="left">density plots</td>
<td align="left">plot(density())</td>
<td align="left">densityplot</td>
<td align="left">geom_density</td>
</tr>
<tr class="even">
<td align="left">boxplots</td>
<td align="left">boxplot</td>
<td align="left">bwplot</td>
<td align="left">geom_boxplot</td>
</tr>
<tr class="odd">
<td align="left">violin plots</td>
<td align="left">vioplot::vioplot</td>
<td align="left">bwplot(…, panel = panel.violin)</td>
<td align="left">geom_violin</td>
</tr>
<tr class="even">
<td align="left">line plots</td>
<td align="left">plot, matplot</td>
<td align="left">xyploy, parallelplot</td>
<td align="left">geom_line</td>
</tr>
<tr class="odd">
<td align="left">bar plots</td>
<td align="left">barplot</td>
<td align="left">barchart</td>
<td align="left">geom_bar</td>
</tr>
<tr class="even">
<td align="left">pie charts</td>
<td align="left">pie</td>
<td align="left"></td>
<td align="left">geom_bar with polar coordinates</td>
</tr>
<tr class="odd">
<td align="left">dot plots</td>
<td align="left">dotchart</td>
<td align="left">dotplot</td>
<td align="left">geom_point</td>
</tr>
<tr class="even">
<td align="left">stip plots</td>
<td align="left">stripchart</td>
<td align="left">stripplot</td>
<td align="left">goem_point</td>
</tr>
<tr class="odd">
<td align="left">dendrogramms</td>
<td align="left">plot(hclust())</td>
<td align="left">latticeExtra package</td>
<td align="left">ggdendro package</td>
</tr>
<tr class="even">
<td align="left">heatmaps</td>
<td align="left">image, heatmap</td>
<td align="left">levelplot</td>
<td align="left">geom_tile</td>
</tr>
</tbody>
</table>
<p>Below, we are going to use a data from the <em><a href="http://bioconductor.org/packages/pRolocdata">pRolocdata</a></em> to illustrate the plotting functions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"pRolocdata"</span>)
<span class="kw">data</span>(tan2009r1)</code></pre></div>
<div id="scatter-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#scatter-plots" class="anchor"></a>Scatter plots</h2>
<p>See the MA and volcano plot examples.</p>
<p>The default plot <code>type</code> is <code>p</code>, for points. Other important types are <code>l</code> for lines and <code>h</code> for <em>histogram</em> (see below).</p>
</div>
<div id="historams-and-density-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#historams-and-density-plots" class="anchor"></a>Historams and density plots</h2>
<p>We extract the (normalised) intensities of the first sample</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">exprs</span>(tan2009r1)[, <span class="dv">1</span>]</code></pre></div>
<p>and plot the distribution with a histogram and a density plot next to each other on the same figure (using the <code>mfrow</code> <code>par</code> plotting paramter)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">hist</span>(x)
<span class="kw">plot</span>(<span class="kw">density</span>(x))</code></pre></div>
<p><img src="RProtVis_files/figure-html/histplot-1.png" width="1344"></p>
</div>
<div id="box-plots-and-violin-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#box-plots-and-violin-plots" class="anchor"></a>Box plots and violin plots</h2>
<p>we first extract the 888 proteins by <code>r ncol(tan2009r1)</code> samples data matrix and plot the sample distributions next to each other using <code>boxplot</code> and <code>beanplot</code> (from the <em><a href="https://CRAN.R-project.org/package=beanplot">beanplot</a></em> package).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"beanplot"</span>)
x &lt;-<span class="st"> </span><span class="kw">exprs</span>(tan2009r1)
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">1</span>))
<span class="kw">boxplot</span>(x)
<span class="kw">beanplot</span>(x[, <span class="dv">1</span>], x[, <span class="dv">2</span>], x[, <span class="dv">3</span>], x[, <span class="dv">4</span>], <span class="dt">log =</span> <span class="st">""</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/bxplot-1.png" width="672"></p>
</div>
<div id="line-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#line-plots" class="anchor"></a>Line plots</h2>
<p>below, we produce line plots that describe the protein quantitative profiles for two sets of proteins, namely er and mitochondrial proteins using <code>matplot</code>.</p>
<p>we need to transpose the matrix (with <code>t</code>) and set the type to both (<code>b</code>), to display points and lines, the colours to red and steel blue, the point characters to 1 (an empty point) and the line type to 1 (a solid line).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">er &lt;-<span class="st"> </span><span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers <span class="op">==</span><span class="st"> "ER"</span>
mt &lt;-<span class="st"> </span><span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers <span class="op">==</span><span class="st"> "mitochondrion"</span>

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">1</span>))
<span class="kw">matplot</span>(<span class="kw">t</span>(x[er, ]), <span class="dt">type =</span> <span class="st">"b"</span>, <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">1</span>, <span class="dt">lty =</span> <span class="dv">1</span>)
<span class="kw">matplot</span>(<span class="kw">t</span>(x[mt, ]), <span class="dt">type =</span> <span class="st">"b"</span>, <span class="dt">col =</span> <span class="st">"steelblue"</span>, <span class="dt">pch =</span> <span class="dv">1</span>, <span class="dt">lty =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/matplotex-1.png" width="672"></p>
<p>In the last section, about spatial proteomics, we use the specialised <code>plotDist</code> function from the <em><a href="http://bioconductor.org/packages/pRoloc">pRoloc</a></em> package to generate such figures.</p>
</div>
<div id="bar-and-dot-charts" class="section level2">
<h2 class="hasAnchor">
<a href="#bar-and-dot-charts" class="anchor"></a>Bar and dot charts</h2>
<p>To illustrate bar and dot charts, we cound the number of proteins in the respective class using table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers)
x</code></pre></div>
<pre><code>## 
##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion 
##             7            28            13             8            29 
##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S 
##            21             4            34            15            20 
##  Ribosome 60S       unknown 
##            32           677</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">barplot</span>(x)
<span class="kw">dotchart</span>(<span class="kw">as.numeric</span>(x))</code></pre></div>
<p><img src="RProtVis_files/figure-html/mrkplot-1.png" width="1152"></p>
<!-- ### Strip plots -->
</div>
<div id="heatmaps" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmaps" class="anchor"></a>Heatmaps</h2>
<p>The easiest to produce a complete heatmap is with the <code>heatmap</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">heatmap</span>(<span class="kw">exprs</span>(tan2009r1))</code></pre></div>
<p><img src="RProtVis_files/figure-html/hmap-1.png" width="672"></p>
<p>To produce the a heatmap without the dendrograms, one can use the image function on a matrix or the specialised version for <code>MSnSet</code> objects from the <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
x &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">9</span>, <span class="dt">ncol =</span> <span class="dv">3</span>)
<span class="kw">image</span>(x)
<span class="kw">image</span>(tan2009r1)</code></pre></div>
<p><img src="RProtVis_files/figure-html/image-1.png" width="1344"></p>
<p>See also <em><a href="https://CRAN.R-project.org/package=gplots">gplots</a></em>’s <code>heatmap.2</code> function and the <em><a href="http://bioconductor.org/packages/Heatplus">Heatplus</a></em> Bioconductor package for more advanced heatmaps and the <em><a href="https://CRAN.R-project.org/package=corrplot">corrplot</a></em> package for correlation matrices.</p>
</div>
<div id="dendrograms" class="section level2">
<h2 class="hasAnchor">
<a href="#dendrograms" class="anchor"></a>Dendrograms</h2>
<p>The easiest way to produce and plot a dendrogram is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d &lt;-<span class="st"> </span><span class="kw">dist</span>(<span class="kw">t</span>(<span class="kw">exprs</span>(tan2009r1))) ## distance between samples
hc &lt;-<span class="st"> </span><span class="kw">hclust</span>(d) ## hierarchical clustering
<span class="kw">plot</span>(hc) ## visualisation</code></pre></div>
<p><img src="RProtVis_files/figure-html/dendro-1.png" width="672"></p>
<p>See also <em><a href="https://CRAN.R-project.org/package=dendextend">dendextend</a></em> and this <a href="http://stackoverflow.com/questions/6673162/reproducing-lattice-dendrogram-graph-with-ggplot2">post</a> to illustrate <em><a href="https://CRAN.R-project.org/package=latticeExtra">latticeExtra</a></em> and <em><a href="https://CRAN.R-project.org/package=ggdendro">ggdendro</a></em>.</p>
</div>
<div id="venn-diagrams" class="section level2">
<h2 class="hasAnchor">
<a href="#venn-diagrams" class="anchor"></a>Venn diagrams</h2>
<ul>
<li>The <em><a href="http://bioconductor.org/packages/limma">limma</a></em> package.</li>
<li>The <em><a href="https://CRAN.R-project.org/package=VennDiagram">VennDiagram</a></em> package.</li>
</ul>
</div>
</div>
<div id="visualising-mass-spectrometry-data" class="section level1">
<h1 class="hasAnchor">
<a href="#visualising-mass-spectrometry-data" class="anchor"></a>Visualising mass spectrometry data</h1>
<div id="direct-access-to-the-raw-data" class="section level2">
<h2 class="hasAnchor">
<a href="#direct-access-to-the-raw-data" class="anchor"></a>Direct access to the raw data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"mzR"</span>)
mzf &lt;-<span class="st"> </span><span class="kw">pxget</span>(px1,
             <span class="st">"TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML"</span>)</code></pre></div>
<pre><code>## Downloading 1 file</code></pre>
<pre><code>## /home/lg390/dev/RforProteomics/docs/articles/TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML already present.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ms &lt;-<span class="st"> </span><span class="kw">openMSfile</span>(mzf)

hd &lt;-<span class="st"> </span><span class="kw">header</span>(ms)
ms1 &lt;-<span class="st"> </span><span class="kw">which</span>(hd<span class="op">$</span>msLevel <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)

rtsel &lt;-<span class="st"> </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">30</span> <span class="op">&amp;</span><span class="st"> </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&lt;</span><span class="st"> </span><span class="dv">35</span>
<span class="kw">library</span>(<span class="st">"MSnbase"</span>)
(M &lt;-<span class="st"> </span><span class="kw">MSmap</span>(ms, ms1[rtsel], <span class="dv">521</span>, <span class="dv">523</span>, .<span class="dv">005</span>, hd))</code></pre></div>
<pre><code>## 1</code></pre>
<pre><code>## Object of class "MSmap"
##  Map [75, 401]
##   [1]  Retention time: 30:1 - 34:58 
##   [2]  M/Z: 521 - 523 (res 0.005)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"lattice"</span>)
ff &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">"yellow"</span>, <span class="st">"steelblue"</span>))
<span class="kw">trellis.par.set</span>(<span class="dt">regions=</span><span class="kw">list</span>(<span class="dt">col=</span><span class="kw">ff</span>(<span class="dv">100</span>)))
<span class="kw">plot</span>(M, <span class="dt">aspect =</span> <span class="dv">1</span>, <span class="dt">allTicks =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mapsheat-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">M<span class="op">@</span>map[<span class="kw">msMap</span>(M) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
<span class="kw">plot3D</span>(M, <span class="dt">rgl =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/maps3D-1.png" width="672"></p>
<p>To produce a version that can be reoriented interactively on the screen discplay, use the <em><a href="https://CRAN.R-project.org/package=rgl">rgl</a></em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"rgl"</span>)
<span class="kw">plot3D</span>(M, <span class="dt">rgl =</span> <span class="ot">TRUE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lout &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="dt">ncol =</span> <span class="dv">10</span>, <span class="dt">nrow =</span> <span class="dv">8</span>)
lout[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, ] &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>)
    lout[ii, ] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>)
lout[<span class="dv">5</span>, ] &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">4</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">each =</span> <span class="dv">2</span>)
lout[<span class="dv">6</span>, ] &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">4</span><span class="op">:</span><span class="dv">8</span>, <span class="dt">each =</span> <span class="dv">2</span>)
lout[<span class="dv">7</span>, ] &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">9</span><span class="op">:</span><span class="dv">13</span>, <span class="dt">each =</span> <span class="dv">2</span>)
lout[<span class="dv">8</span>, ] &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">9</span><span class="op">:</span><span class="dv">13</span>, <span class="dt">each =</span> <span class="dv">2</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">i &lt;-<span class="st"> </span>ms1[<span class="kw">which</span>(rtsel)][<span class="dv">1</span>]
j &lt;-<span class="st"> </span>ms1[<span class="kw">which</span>(rtsel)][<span class="dv">2</span>]
ms2 &lt;-<span class="st"> </span>(i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span>(j<span class="op">-</span><span class="dv">1</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">layout</span>(lout)
<span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>))
<span class="kw">plot</span>(<span class="kw">chromatogram</span>(ms)[[<span class="dv">1</span>]], <span class="dt">type =</span> <span class="st">"l"</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> hd[i, <span class="st">"retentionTime"</span>], <span class="dt">col =</span> <span class="st">"red"</span>)
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">0</span>))
<span class="kw">plot</span>(<span class="kw">peaks</span>(ms, i), <span class="dt">type =</span> <span class="st">"l"</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">400</span>, <span class="dv">1000</span>))
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>,
       <span class="dt">legend =</span> <span class="kw">paste0</span>(
           <span class="st">"Acquisition "</span>, hd[i, <span class="st">"acquisitionNum"</span>],  <span class="st">"</span><span class="ch">\n</span><span class="st">"</span>,
           <span class="st">"Retention time "</span>, <span class="kw">formatRt</span>(hd[i, <span class="st">"retentionTime"</span>])))
<span class="kw">abline</span>(<span class="dt">h =</span> <span class="dv">0</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> hd[ms2, <span class="st">"precursorMZ"</span>],
       <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">"#FF000080"</span>,
           <span class="kw">rep</span>(<span class="st">"#12121280"</span>, <span class="dv">9</span>)))
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">1</span>))
<span class="kw">plot</span>(<span class="kw">peaks</span>(ms, i), <span class="dt">type =</span> <span class="st">"l"</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">521</span>, <span class="fl">522.5</span>), <span class="dt">yaxt =</span> <span class="st">"n"</span>)
<span class="kw">abline</span>(<span class="dt">h =</span> <span class="dv">0</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> hd[ms2, <span class="st">"precursorMZ"</span>], <span class="dt">col =</span> <span class="st">"#FF000080"</span>)
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">1</span>))
<span class="cf">for</span> (ii <span class="cf">in</span> ms2) {
    p &lt;-<span class="st"> </span><span class="kw">peaks</span>(ms, ii)
    <span class="kw">plot</span>(p, <span class="dt">xlab =</span> <span class="st">""</span>, <span class="dt">ylab =</span> <span class="st">""</span>, <span class="dt">type =</span> <span class="st">"h"</span>, <span class="dt">cex.axis =</span> .<span class="dv">6</span>)
    <span class="kw">legend</span>(<span class="st">"topright"</span>,
           <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="st">"Prec M/Z</span><span class="ch">\n</span><span class="st">"</span>, <span class="kw">round</span>(hd[ii, <span class="st">"precursorMZ"</span>], <span class="dv">2</span>)),
           <span class="dt">bty =</span> <span class="st">"n"</span>, <span class="dt">cex =</span> .<span class="dv">8</span>)
}</code></pre></div>
<div class="figure">
<img src="RProtVis_files/figure-html/msdetailsplot-1.png" alt="Accesing and plotting MS data." width="672"><p class="caption">
Accesing and plotting MS data.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">M2 &lt;-<span class="st"> </span><span class="kw">MSmap</span>(ms, i<span class="op">:</span>j, <span class="dv">100</span>, <span class="dv">1000</span>, <span class="dv">1</span>, hd)</code></pre></div>
<pre><code>## 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot3D</span>(M2)</code></pre></div>
<p><img src="RProtVis_files/figure-html/maps3D2-1.png" width="672"></p>
<div id="ms-barcoding" class="section level3">
<h3 class="hasAnchor">
<a href="#ms-barcoding" class="anchor"></a>MS barcoding</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>))
<span class="kw">image</span>(<span class="kw">t</span>(<span class="kw">matrix</span>(hd<span class="op">$</span>msLevel, <span class="dv">1</span>, <span class="kw">nrow</span>(hd))),
      <span class="dt">xlab=</span><span class="st">"Retention time"</span>,
      <span class="dt">xaxt=</span><span class="st">"n"</span>, <span class="dt">yaxt=</span><span class="st">"n"</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">"black"</span>,<span class="st">"steelblue"</span>))
k &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="kw">range</span>(hd<span class="op">$</span>retentionTime) <span class="op">/</span><span class="st"> </span><span class="dv">60</span>)
nk &lt;-<span class="st"> </span><span class="dv">5</span>
<span class="kw">axis</span>(<span class="dt">side=</span><span class="dv">1</span>, <span class="dt">at=</span><span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span><span class="op">/</span>nk), <span class="dt">labels=</span><span class="kw">seq</span>(k[<span class="dv">1</span>],k[<span class="dv">2</span>],k[<span class="dv">2</span>]<span class="op">/</span>nk))</code></pre></div>
<p><img src="RProtVis_files/figure-html/barcode-1.png" width="1152"></p>
</div>
<div id="animation" class="section level3">
<h3 class="hasAnchor">
<a href="#animation" class="anchor"></a>Animation</h3>
<p>The following animation scrolls over 5 minutes of retention time for a MZ range between 521 and 523.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"animation"</span>)
an1 &lt;-<span class="st"> </span><span class="cf">function</span>() {
    <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="dv">5</span>, <span class="fl">0.2</span>)) {
        rtsel &lt;-<span class="st"> </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&gt;</span><span class="st"> </span>(<span class="dv">30</span> <span class="op">+</span><span class="st"> </span>i) <span class="op">&amp;</span>
<span class="st">            </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&lt;</span><span class="st"> </span>(<span class="dv">35</span> <span class="op">+</span><span class="st"> </span>i)
        M &lt;-<span class="st"> </span><span class="kw">MSmap</span>(ms, ms1[rtsel], <span class="dv">521</span>, <span class="dv">523</span>, .<span class="dv">005</span>, hd)
        M<span class="op">@</span>map[<span class="kw">msMap</span>(M) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
        <span class="kw">print</span>(<span class="kw">plot3D</span>(M, <span class="dt">rgl =</span> <span class="ot">FALSE</span>))
    }
}

<span class="kw">saveGIF</span>(<span class="kw">an1</span>(), <span class="dt">movie.name =</span> <span class="st">"msanim1.gif"</span>)</code></pre></div>
<div class="figure">
<img src="figures/msanim1.gif" alt="MS animation 1"><p class="caption">MS animation 1</p>
</div>
<p>The code chunk below scrolls of a slice of retention times while keeping the retention time constant between 30 and 35 minutes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">an2 &lt;-<span class="st"> </span><span class="cf">function</span>() {
    <span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq</span>(<span class="dv">0</span>, <span class="fl">2.5</span>, <span class="fl">0.1</span>)) {
        rtsel &lt;-<span class="st"> </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&gt;</span><span class="st"> </span><span class="dv">30</span> <span class="op">&amp;</span><span class="st"> </span>hd<span class="op">$</span>retentionTime[ms1] <span class="op">/</span><span class="st"> </span><span class="dv">60</span> <span class="op">&lt;</span><span class="st"> </span><span class="dv">35</span>
        mz1 &lt;-<span class="st"> </span><span class="dv">520</span> <span class="op">+</span><span class="st"> </span>i
        mz2 &lt;-<span class="st"> </span><span class="dv">522</span> <span class="op">+</span><span class="st"> </span>i
        M &lt;-<span class="st"> </span><span class="kw">MSmap</span>(ms, ms1[rtsel], mz1, mz2, .<span class="dv">005</span>, hd)
        M<span class="op">@</span>map[<span class="kw">msMap</span>(M) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>] &lt;-<span class="st"> </span><span class="ot">NA</span>
        <span class="kw">print</span>(<span class="kw">plot3D</span>(M, <span class="dt">rgl =</span> <span class="ot">FALSE</span>))
    }
}

<span class="kw">saveGIF</span>(<span class="kw">an2</span>(), <span class="dt">movie.name =</span> <span class="st">"msanim2.gif"</span>)</code></pre></div>
<div class="figure">
<img src="figures/msanim2.gif" alt="MS animation 2"><p class="caption">MS animation 2</p>
</div>
</div>
</div>
<div id="the-msnbase-infrastructure" class="section level2">
<h2 class="hasAnchor">
<a href="#the-msnbase-infrastructure" class="anchor"></a>The <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> infrastructure</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MSnbase"</span>)
<span class="kw">data</span>(itraqdata)
itraqdata2 &lt;-<span class="st"> </span><span class="kw">pickPeaks</span>(itraqdata, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)
<span class="kw">plot</span>(itraqdata[[<span class="dv">25</span>]], <span class="dt">full =</span> <span class="ot">TRUE</span>, <span class="dt">reporters =</span> iTRAQ4)</code></pre></div>
<p><img src="RProtVis_files/figure-html/msnbviz-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">oma =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>))
<span class="kw">plot</span>(itraqdata2[[<span class="dv">25</span>]], itraqdata2[[<span class="dv">28</span>]], <span class="dt">sequences =</span> <span class="kw">rep</span>(<span class="st">"IMIDLDGTENK"</span>, <span class="dv">2</span>))</code></pre></div>
<p><img src="RProtVis_files/figure-html/msnbviz-2.png" width="672"></p>
</div>
<div id="the-protviz-package" class="section level2">
<h2 class="hasAnchor">
<a href="#the-protviz-package" class="anchor"></a>The <em><a href="https://CRAN.R-project.org/package=protViz">protViz</a></em> package</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"protViz"</span>)
<span class="kw">data</span>(msms)

fi &lt;-<span class="st"> </span><span class="kw">fragmentIon</span>(<span class="st">"TAFDEAIAELDTLNEESYK"</span>)
fi.cyz &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">cbind</span>(<span class="dt">c=</span>fi[[<span class="dv">1</span>]]<span class="op">$</span>c, <span class="dt">y=</span>fi[[<span class="dv">1</span>]]<span class="op">$</span>y, <span class="dt">z=</span>fi[[<span class="dv">1</span>]]<span class="op">$</span>z))
     
p &lt;-<span class="st"> </span><span class="kw">peakplot</span>(<span class="st">"TAFDEAIAELDTLNEESYK"</span>,
              <span class="dt">spec =</span> msms[[<span class="dv">1</span>]],
              <span class="dt">fi =</span> fi.cyz,
              <span class="dt">itol =</span> <span class="fl">0.6</span>,
              <span class="dt">ion.axes =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/protviz-1.png" width="672"></p>
<p>The <code>peakplot</code> function return the annotation of the MSMS spectrum that is plotted:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(p)</code></pre></div>
<pre><code>## List of 7
##  $ mZ.Da.error : num [1:57] 215.3 144.27 -2.8 -17.06 2.03 ...
##  $ mZ.ppm.error: num [1:57] 1808046 758830 -8306 -37724 3501 ...
##  $ idx         : int [1:57] 1 1 1 3 16 24 41 52 67 88 ...
##  $ label       : chr [1:57] "c1" "c2" "c3" "c4" ...
##  $ score       : num -1
##  $ sequence    : chr "TAFDEAIAELDTLNEESYK"
##  $ fragmentIon :'data.frame':    19 obs. of  3 variables:
##   ..$ c: num [1:19] 119 190 337 452 581 ...
##   ..$ y: num [1:19] 147 310 397 526 655 ...
##   ..$ z: num [1:19] 130 293 380 509 638 ...</code></pre>
</div>
<div id="preprocessing-of-maldi-ms-spectra" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing-of-maldi-ms-spectra" class="anchor"></a>Preprocessing of MALDI-MS spectra</h2>
<p>The following code chunks demonstrate the usage of the mass spectrometry preprocessing and plotting routines in the <code>r CRANpkg("MALDIquant")</code> package. <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> uses the traditional graphics system. Therefore <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> overloads the traditional functions <code>plot</code>, <code>lines</code> and <code>points</code> for its own data types. These data types represents spectrum and peak lists as S4 classes. Please see the <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> <a href="http://cran.r-project.org/web/packages/MALDIquant/vignettes/MALDIquant-intro.pdf">vignette</a> and the corresponding <a href="http://strimmerlab.org/software/maldiquant/">website</a> for more details.</p>
<p>After loading some example data a simple <code>plot</code> draws the raw spectrum.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MALDIquant"</span>)

<span class="kw">data</span>(<span class="st">"fiedler2009subset"</span>, <span class="dt">package=</span><span class="st">"MALDIquant"</span>)

<span class="kw">plot</span>(fiedler2009subset[[<span class="dv">14</span>]])</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqraw-1.png" width="672"></p>
<p>After some preprocessing, namely variance stabilization and smoothing, we use <code>lines</code> to draw our baseline estimate in our processed spectrum.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">transformedSpectra &lt;-<span class="st"> </span><span class="kw">transformIntensity</span>(fiedler2009subset, <span class="dt">method =</span> <span class="st">"sqrt"</span>)
smoothedSpectra &lt;-<span class="st"> </span><span class="kw">smoothIntensity</span>(transformedSpectra, <span class="dt">method =</span> <span class="st">"SavitzkyGolay"</span>)

<span class="kw">plot</span>(smoothedSpectra[[<span class="dv">14</span>]])
<span class="kw">lines</span>(<span class="kw">estimateBaseline</span>(smoothedSpectra[[<span class="dv">14</span>]]), <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">col =</span> <span class="st">"red"</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqestimatebaseline-1.png" width="672"></p>
<p>After removing the background removal we could use <code>plot</code> again to draw our baseline corrected spectrum.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rbSpectra &lt;-<span class="st"> </span><span class="kw">removeBaseline</span>(smoothedSpectra)
<span class="kw">plot</span>(rbSpectra[[<span class="dv">14</span>]])</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqremovebaseline-1.png" width="672"></p>
<p><code>detectPeaks</code> returns a <code>MassPeaks</code> object that offers the same traditional graphics functions. The next code chunk demonstrates how to mark the detected peaks in a spectrum.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cbSpectra &lt;-<span class="st"> </span><span class="kw">calibrateIntensity</span>(rbSpectra, <span class="dt">method =</span> <span class="st">"TIC"</span>)
peaks &lt;-<span class="st"> </span><span class="kw">detectPeaks</span>(cbSpectra, <span class="dt">SNR =</span> <span class="dv">5</span>)

<span class="kw">plot</span>(cbSpectra[[<span class="dv">14</span>]])
<span class="kw">points</span>(peaks[[<span class="dv">14</span>]], <span class="dt">col =</span> <span class="st">"red"</span>, <span class="dt">pch =</span> <span class="dv">4</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqpeaks-1.png" width="672"></p>
<p>Additional there is a special function <code>labelPeaks</code> that allows to draw the <em>M/Z</em> values above the corresponding peaks. Next we mark the 5 top peaks in the spectrum.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top5 &lt;-<span class="st"> </span><span class="kw">intensity</span>(peaks[[<span class="dv">14</span>]]) <span class="op">%in%</span><span class="st"> </span><span class="kw">sort</span>(<span class="kw">intensity</span>(peaks[[<span class="dv">14</span>]]),
                                         <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]
<span class="kw">labelPeaks</span>(peaks[[<span class="dv">14</span>]], <span class="dt">index =</span> top5, <span class="dt">avoidOverlap =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqlabelpeaks-1.png" width="672"></p>
<p>Often multiple spectra have to be recalibrated to be comparable. Therefore <em><a href="https://CRAN.R-project.org/package=MALDIquant">MALDIquant</a></em> warps the spectra according to so called reference or landmark peaks. For debugging the <code>determineWarpingFunctions</code> function offers some warping plots. Here we show only the last 4 plots:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))
warpingFunctions &lt;-
<span class="st">    </span><span class="kw">determineWarpingFunctions</span>(peaks,
                              <span class="dt">tolerance =</span> <span class="fl">0.001</span>,
                              <span class="dt">plot =</span> <span class="ot">TRUE</span>,
                              <span class="dt">plotInteractive =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqwarp-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))
warpedSpectra &lt;-<span class="st"> </span><span class="kw">warpMassSpectra</span>(cbSpectra, warpingFunctions)
warpedPeaks &lt;-<span class="st"> </span><span class="kw">warpMassPeaks</span>(peaks, warpingFunctions)</code></pre></div>
<p>In the next code chunk we visualise the need and the effect of the recalibration.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sel &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">10</span>, <span class="dv">14</span>, <span class="dv">16</span>)
xlim &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">4180</span>, <span class="dv">4240</span>)
ylim &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">1.9e-3</span>)
lty &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">6</span>)

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plot</span>(cbSpectra[[<span class="dv">1</span>]], <span class="dt">xlim =</span> xlim, <span class="dt">ylim =</span> ylim, <span class="dt">type =</span> <span class="st">"n"</span>)

<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq</span>(<span class="dt">along =</span> sel)) {
  <span class="kw">lines</span>(peaks[[sel[i]]], <span class="dt">lty =</span> lty[i], <span class="dt">col =</span> i)
  <span class="kw">lines</span>(cbSpectra[[sel[i]]], <span class="dt">lty =</span> lty[i], <span class="dt">col =</span> i)
}

<span class="kw">plot</span>(cbSpectra[[<span class="dv">1</span>]], <span class="dt">xlim =</span> xlim, <span class="dt">ylim =</span> ylim, <span class="dt">type =</span> <span class="st">"n"</span>)

<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq</span>(<span class="dt">along =</span> sel)) {
  <span class="kw">lines</span>(warpedPeaks[[sel[i]]], <span class="dt">lty =</span> lty[i], <span class="dt">col =</span> i)
  <span class="kw">lines</span>(warpedSpectra[[sel[i]]], <span class="dt">lty =</span> lty[i], <span class="dt">col =</span> i)
}</code></pre></div>
<p><img src="RProtVis_files/figure-html/mqwarped-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</code></pre></div>
<p>The code chunks above generate plots that are very similar to the figure 7 in the corresponding paper <em>“Visualisation of proteomics data using R”</em>. Please find the code to exactly reproduce the figure at: <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/R/createFigure1_color.R" class="uri">https://github.com/sgibb/MALDIquantExamples/blob/master/R/createFigure1_color.R</a></p>
</div>
</div>
<div id="genomic-and-protein-sequences" class="section level1">
<h1 class="hasAnchor">
<a href="#genomic-and-protein-sequences" class="anchor"></a>Genomic and protein sequences</h1>
<p>These visualisations originate from the <code>Pbase</code> <a href="http://bioconductor.org/packages/devel/bioc/vignettes/Pbase/inst/doc/Pbase-data.html"><code>Pbase-data</code></a> and <a href="http://bioconductor.org/packages/devel/bioc/vignettes/Pbase/inst/doc/mapping.html"><code>mapping</code></a> vignettes.</p>
</div>
<div id="mass-spectrometry-imaging" class="section level1">
<h1 class="hasAnchor">
<a href="#mass-spectrometry-imaging" class="anchor"></a>Mass spectrometry imaging</h1>
<p>The following code chunk downloads a MALDI imaging dataset from a mouse kidney shared by <a href="http://figshare.com/articles/MALDI_Imaging_Mass_Spectrometry_of_a_Mouse_Kidney/735961">Adrien Nyakas and Stefan Schurch</a> and generates a plot with the mean spectrum and three slices of interesting <em>M/Z</em> regions.</p>
<!-- Not running this code chunk as the access to the data (75M) -->
<!-- on figshare seems to hang or become very slow at times -->
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"MALDIquant"</span>)
<span class="kw">library</span>(<span class="st">"MALDIquantForeign"</span>)

spectra &lt;-<span class="st"> </span><span class="kw">importBrukerFlex</span>(<span class="st">"http://files.figshare.com/1106682/MouseKidney_IMS_testdata.zip"</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)

spectra &lt;-<span class="st"> </span><span class="kw">smoothIntensity</span>(spectra, <span class="st">"SavitzkyGolay"</span>,  <span class="dt">halfWindowSize =</span> <span class="dv">8</span>)
spectra &lt;-<span class="st"> </span><span class="kw">removeBaseline</span>(spectra, <span class="dt">method =</span> <span class="st">"TopHat"</span>, <span class="dt">halfWindowSize =</span> <span class="dv">16</span>)
spectra &lt;-<span class="st"> </span><span class="kw">calibrateIntensity</span>(spectra, <span class="dt">method =</span> <span class="st">"TIC"</span>)
avgSpectrum &lt;-<span class="st"> </span><span class="kw">averageMassSpectra</span>(spectra)
avgPeaks &lt;-<span class="st"> </span><span class="kw">detectPeaks</span>(avgSpectrum, <span class="dt">SNR =</span> <span class="dv">5</span>)

avgPeaks &lt;-<span class="st"> </span>avgPeaks[<span class="kw">intensity</span>(avgPeaks) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.0015</span>]

oldPar &lt;-<span class="st"> </span><span class="kw">par</span>(<span class="dt">no.readonly =</span> <span class="ot">TRUE</span>)
<span class="kw">layout</span>(<span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>), <span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">byrow =</span> <span class="ot">TRUE</span>))
<span class="kw">plot</span>(avgSpectrum, <span class="dt">main =</span> <span class="st">"mean spectrum"</span>,
     <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">3000</span>, <span class="dv">6000</span>), <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.007</span>))
<span class="kw">lines</span>(avgPeaks, <span class="dt">col =</span> <span class="st">"red"</span>)
<span class="kw">labelPeaks</span>(avgPeaks, <span class="dt">cex =</span> <span class="dv">1</span>)

<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="fl">1.5</span>, <span class="fl">0.5</span>))
<span class="kw">plotMsiSlice</span>(spectra,
             <span class="dt">center =</span> <span class="kw">mass</span>(avgPeaks),
             <span class="dt">tolerance =</span> <span class="dv">1</span>,
             <span class="dt">plotInteractive =</span> <span class="ot">TRUE</span>)
<span class="kw">par</span>(oldPar)</code></pre></div>
<p><img src="figures/mqmsi-1.png" alt="mqmsi">)]</p>
<div id="an-interactive-shiny-app-for-imaging-mass-spectrometry" class="section level2">
<h2 class="hasAnchor">
<a href="#an-interactive-shiny-app-for-imaging-mass-spectrometry" class="anchor"></a>An interactive <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app for Imaging mass spectrometry</h2>
<p>There is also an interactive <a href="https://github.com/sgibb/ims-shiny">MALDIquant IMS shiny app</a> for demonstration purposes. A screen shot is displayed below. To start the application:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"shiny"</span>)
<span class="kw">runGitHub</span>(<span class="st">"sgibb/ims-shiny"</span>)</code></pre></div>
<div class="figure">
<img src="figures/ims-shiny.png" alt="ims-shiny screeshot"><p class="caption">ims-shiny screeshot</p>
</div>
</div>
</div>
<div id="spatial-proteomics" class="section level1">
<h1 class="hasAnchor">
<a href="#spatial-proteomics" class="anchor"></a>Spatial proteomics</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">"pRoloc"</span>)
<span class="kw">library</span>(<span class="st">"pRolocdata"</span>)

<span class="kw">data</span>(tan2009r1)

## these params use class weights
fn &lt;-<span class="st"> </span><span class="kw">dir</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="dt">package =</span> <span class="st">"pRoloc"</span>),
          <span class="dt">full.names =</span> <span class="ot">TRUE</span>, <span class="dt">pattern =</span> <span class="st">"params2.rda"</span>)
<span class="kw">load</span>(fn)

<span class="kw">setStockcol</span>(<span class="ot">NULL</span>)
<span class="kw">setStockcol</span>(<span class="kw">paste0</span>(<span class="kw">getStockcol</span>(), <span class="dv">90</span>))

w &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="kw">fData</span>(tan2009r1)[, <span class="st">"pd.markers"</span>])
(w &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">/</span>w[<span class="kw">names</span>(w) <span class="op">!=</span><span class="st"> "unknown"</span>])</code></pre></div>
<pre><code>## 
##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion 
##    0.14285714    0.05000000    0.16666667    0.12500000    0.07142857 
##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S 
##    0.05000000    0.25000000    0.06666667    0.09090909    0.07142857 
##  Ribosome 60S 
##    0.04000000</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tan2009r1 &lt;-<span class="st"> </span><span class="kw">svmClassification</span>(tan2009r1, params2,
                               <span class="dt">class.weights =</span> w,
                               <span class="dt">fcol =</span> <span class="st">"pd.markers"</span>)
ptsze &lt;-<span class="st"> </span><span class="kw">exp</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>svm.scores) <span class="op">-</span><span class="st"> </span><span class="dv">1</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lout &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="kw">rep</span>(<span class="dv">5</span>, <span class="dv">4</span>)), <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">nrow =</span> <span class="dv">2</span>)
<span class="kw">layout</span>(lout)
cls &lt;-<span class="st"> </span><span class="kw">getStockcol</span>()
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>))
<span class="kw">plotDist</span>(tan2009r1[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>PLSDA <span class="op">==</span><span class="st"> "mitochondrion"</span>), ],
         <span class="dt">markers =</span> <span class="kw">featureNames</span>(tan2009r1)[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers.orig <span class="op">==</span><span class="st"> "mitochondrion"</span>)],
         <span class="dt">mcol =</span> cls[<span class="dv">5</span>])
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="st">"mitochondrion"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>)
<span class="kw">plotDist</span>(tan2009r1[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>PLSDA <span class="op">==</span><span class="st"> "ER/Golgi"</span>), ],
         <span class="dt">markers =</span> <span class="kw">featureNames</span>(tan2009r1)[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers.orig <span class="op">==</span><span class="st"> "ER"</span>)],
         <span class="dt">mcol =</span> cls[<span class="dv">2</span>])
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="st">"ER"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>)
<span class="kw">plotDist</span>(tan2009r1[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>PLSDA <span class="op">==</span><span class="st"> "ER/Golgi"</span>), ],
         <span class="dt">markers =</span> <span class="kw">featureNames</span>(tan2009r1)[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers.orig <span class="op">==</span><span class="st"> "Golgi"</span>)],
         <span class="dt">mcol =</span> cls[<span class="dv">3</span>])
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="st">"Golgi"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>)
<span class="kw">plotDist</span>(tan2009r1[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>PLSDA <span class="op">==</span><span class="st"> "PM"</span>), ],
         <span class="dt">markers =</span> <span class="kw">featureNames</span>(tan2009r1)[<span class="kw">which</span>(<span class="kw">fData</span>(tan2009r1)<span class="op">$</span>markers.orig <span class="op">==</span><span class="st"> "PM"</span>)],
         <span class="dt">mcol =</span> cls[<span class="dv">8</span>])
<span class="kw">legend</span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="st">"PM"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>)
<span class="kw">plot2D</span>(tan2009r1, <span class="dt">fcol =</span> <span class="st">"svm"</span>, <span class="dt">cex =</span> ptsze, <span class="dt">method =</span> <span class="st">"kpca"</span>)
<span class="kw">addLegend</span>(tan2009r1, <span class="dt">where =</span> <span class="st">"bottomleft"</span>, <span class="dt">fcol =</span> <span class="st">"svm"</span>, <span class="dt">bty =</span> <span class="st">"n"</span>)</code></pre></div>
<p><img src="RProtVis_files/figure-html/spatplot-1.png" width="1152"></p>
<p>See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/pRoloc/inst/doc/pRoloc-tutorial.pdf"><code>pRoloc-tutorial</code></a> vignette (pdf) from the <em><a href="http://bioconductor.org/packages/pRoloc">pRoloc</a></em> package for details about spatial proteomics data analysis and visualisation.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">sessionInfo</span>(), <span class="dt">locale =</span> <span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## R Under development (unstable) (2017-12-12 r73903)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.0
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] beanplot_1.2             ggplot2_2.2.1.9000      
##  [3] lattice_0.20-35          e1071_1.6-8             
##  [5] msmsTests_1.17.0         msmsEDA_1.17.0          
##  [7] pRolocdata_1.17.0        pRoloc_1.19.0           
##  [9] MLInterfaces_1.59.0      cluster_2.0.6           
## [11] annotate_1.57.2          XML_3.98-1.9            
## [13] AnnotationDbi_1.41.4     IRanges_2.13.7          
## [15] S4Vectors_0.17.18        MALDIquantForeign_0.11.1
## [17] MALDIquant_1.17          RColorBrewer_1.1-2      
## [19] xtable_1.8-2             rpx_1.15.0              
## [21] knitr_1.18               DT_0.2                  
## [23] BiocInstaller_1.29.3     protViz_0.2.45          
## [25] RforProteomics_1.15.1    MSnbase_2.5.4           
## [27] ProtGenerics_1.11.0      BiocParallel_1.13.1     
## [29] mzR_2.13.3               Rcpp_0.12.14            
## [31] Biobase_2.39.1           BiocGenerics_0.25.1     
## [33] BiocStyle_2.7.2         
## 
## loaded via a namespace (and not attached):
##   [1] R.utils_2.6.0                 RUnit_0.4.31                 
##   [3] tidyselect_0.2.3              RSQLite_2.0                  
##   [5] htmlwidgets_0.9               grid_3.5.0                   
##   [7] trimcluster_0.1-2             lpSolve_5.6.13               
##   [9] Rtsne_0.13                    rda_1.0.2-2                  
##  [11] munsell_0.4.3                 codetools_0.2-15             
##  [13] preprocessCore_1.41.0         withr_2.1.1.9000             
##  [15] colorspace_1.3-2              Category_2.45.2              
##  [17] highr_0.6                     robustbase_0.92-8            
##  [19] dimRed_0.1.0                  mzID_1.17.0                  
##  [21] labeling_0.3                  mnormt_1.5-5                 
##  [23] hwriter_1.3.2                 bit64_0.9-7                  
##  [25] ggvis_0.4.3                   rprojroot_1.3-2              
##  [27] ipred_0.9-6                   randomForest_4.6-12          
##  [29] diptest_0.75-7                R6_2.2.2                     
##  [31] doParallel_1.0.11             gridSVG_1.6-0                
##  [33] locfit_1.5-9.1                flexmix_2.3-14               
##  [35] DRR_0.0.3                     bitops_1.0-6                 
##  [37] assertthat_0.2.0              scales_0.5.0.9000            
##  [39] nnet_7.3-12                   gtable_0.2.0                 
##  [41] affy_1.57.0                   biocViews_1.47.0             
##  [43] ddalpha_1.3.1                 timeDate_3042.101            
##  [45] rlang_0.1.6                   CVST_0.2-1                   
##  [47] genefilter_1.61.0             RcppRoll_0.2.2               
##  [49] splines_3.5.0                 lazyeval_0.2.1               
##  [51] ModelMetrics_1.1.0            impute_1.53.0                
##  [53] hexbin_1.27.1                 broom_0.4.3                  
##  [55] yaml_2.1.16                   reshape2_1.4.3               
##  [57] threejs_0.3.1                 crosstalk_1.0.0              
##  [59] backports_1.1.2               httpuv_1.3.5                 
##  [61] qvalue_2.11.0                 RBGL_1.55.0                  
##  [63] caret_6.0-78                  tools_3.5.0                  
##  [65] lava_1.5.1                    psych_1.7.8                  
##  [67] gplots_3.0.1                  affyio_1.49.1                
##  [69] readBrukerFlexData_1.8.5      proxy_0.4-21                 
##  [71] plyr_1.8.4                    base64enc_0.1-3              
##  [73] progress_1.1.2                zlibbioc_1.25.0              
##  [75] purrr_0.2.4                   RCurl_1.95-4.10              
##  [77] prettyunits_1.0.2             rpart_4.1-12                 
##  [79] viridis_0.4.1                 sampling_2.8                 
##  [81] sfsmisc_1.1-1                 magrittr_1.5                 
##  [83] pcaMethods_1.71.0             mvtnorm_1.0-6                
##  [85] whisker_0.3-2                 mime_0.5                     
##  [87] evaluate_0.10.1               mclust_5.4                   
##  [89] gridExtra_2.3                 compiler_3.5.0               
##  [91] biomaRt_2.35.5                tibble_1.4.1                 
##  [93] KernSmooth_2.23-15            R.oo_1.21.0                  
##  [95] htmltools_0.3.6               tidyr_0.7.2                  
##  [97] lubridate_1.7.1               DBI_0.7                      
##  [99] MASS_7.3-48                   fpc_2.1-10                   
## [101] Matrix_1.2-12                 vsn_3.47.0                   
## [103] R.methodsS3_1.7.1             gdata_2.18.0                 
## [105] mlbench_2.1-1                 bindr_0.1                    
## [107] gower_0.1.2                   igraph_1.1.2                 
## [109] pkgconfig_2.0.1               foreign_0.8-70               
## [111] readMzXmlData_2.8.1           recipes_0.1.2                
## [113] xml2_1.1.1                    foreach_1.4.4                
## [115] prodlim_1.6.1                 stringr_1.2.0                
## [117] digest_0.6.13                 pls_2.6-0                    
## [119] graph_1.57.1                  rmarkdown_1.8                
## [121] edgeR_3.21.5                  dendextend_1.6.0             
## [123] GSEABase_1.41.2               curl_3.1                     
## [125] kernlab_0.9-25                shiny_1.0.5                  
## [127] gtools_3.5.0                  modeltools_0.2-21            
## [129] nlme_3.1-135                  jsonlite_1.5                 
## [131] bindrcpp_0.2                  interactiveDisplay_1.17.0    
## [133] viridisLite_0.2.0             limma_3.35.5                 
## [135] pillar_1.0.1                  httr_1.3.1                   
## [137] DEoptimR_1.0-8                survival_2.41-3              
## [139] interactiveDisplayBase_1.17.0 glue_1.2.0                   
## [141] FNN_1.1                       gbm_2.1.3                    
## [143] prabclus_2.2-6                iterators_1.0.9              
## [145] bit_1.1-12                    class_7.3-14                 
## [147] stringi_1.1.6                 blob_1.1.0                   
## [149] caTools_1.17.1                memoise_1.1.0                
## [151] dplyr_0.7.4</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#references">References</a></li>
      <li><a href="#relevant-packages">Relevant packages</a></li>
      </ul>
</li>
      <li><a href="#ascombes-quartet">Ascombe’s quartet</a></li>
      <li>
<a href="#the-ma-plot-example">The MA plot example</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-traditional-plotting-system">The traditional plotting system</a></li>
      <li><a href="#lattice">lattice</a></li>
      <li><a href="#ggplot2">ggplot2</a></li>
      <li><a href="#customization">Customization</a></li>
      <li><a href="#the-maplot-method-for-msnset-instances">The <code>MAplot</code> method for <code>MSnSet</code> instances</a></li>
      <li><a href="#an-interactive-shiny-app-for-ma-plots">An interactive <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app for MA plots</a></li>
      <li><a href="#volcano-plots">Volcano plots</a></li>
      <li><a href="#a-pca-plot">A PCA plot</a></li>
      </ul>
</li>
      <li>
<a href="#plotting-with-r">Plotting with R</a><ul class="nav nav-pills nav-stacked">
<li><a href="#scatter-plots">Scatter plots</a></li>
      <li><a href="#historams-and-density-plots">Historams and density plots</a></li>
      <li><a href="#box-plots-and-violin-plots">Box plots and violin plots</a></li>
      <li><a href="#line-plots">Line plots</a></li>
      <li><a href="#bar-and-dot-charts">Bar and dot charts</a></li>
      <li><a href="#heatmaps">Heatmaps</a></li>
      <li><a href="#dendrograms">Dendrograms</a></li>
      <li><a href="#venn-diagrams">Venn diagrams</a></li>
      </ul>
</li>
      <li>
<a href="#visualising-mass-spectrometry-data">Visualising mass spectrometry data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#direct-access-to-the-raw-data">Direct access to the raw data</a></li>
      <li><a href="#the-msnbase-infrastructure">The <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> infrastructure</a></li>
      <li><a href="#the-protviz-package">The <em><a href="https://CRAN.R-project.org/package=protViz">protViz</a></em> package</a></li>
      <li><a href="#preprocessing-of-maldi-ms-spectra">Preprocessing of MALDI-MS spectra</a></li>
      </ul>
</li>
      <li><a href="#genomic-and-protein-sequences">Genomic and protein sequences</a></li>
      <li>
<a href="#mass-spectrometry-imaging">Mass spectrometry imaging</a><ul class="nav nav-pills nav-stacked">
<li><a href="#an-interactive-shiny-app-for-imaging-mass-spectrometry">An interactive <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> app for Imaging mass spectrometry</a></li>
      </ul>
</li>
      <li><a href="#spatial-proteomics">Spatial proteomics</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
