<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Proteomics and MS biocView packages — proteomicsPackages • RforProteomics</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Proteomics and MS biocView packages — proteomicsPackages" />

<meta property="og:description" content="Searches for all the packages with the &quot;Proteomics&quot;
  (software), &quot;MassSpectrometry&quot; (software) and
  &quot;MassSepctrometryData&quot; (data) packages and return their
  names, titles and versions as a data.frame. The
  (unexported but documented) underlying functions are
  RforProteomics:::getPackagesInBiocView (to find relevant
  package) and RforProteomics:::packageDF
  (data.frame formatting)." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RforProteomics</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.19.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RforProteomics.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/RProtVis.html">Visualisation of proteomics data using R and Bioconductor</a>
    </li>
    <li>
      <a href="../articles/biocprot.html">Proteomics packages in Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Proteomics and MS biocView packages</h1>
    
    <div class="hidden name"><code>proteomicsPackages.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Searches for all the packages with the <code>"Proteomics"</code>
  (software), <code>"MassSpectrometry"</code> (software) and
  <code>"MassSepctrometryData"</code> (data) packages and return their
  names, titles and versions as a <code>data.frame</code>. The
  (unexported but documented) underlying functions are
  <code>RforProteomics:::getPackagesInBiocView</code> (to find relevant
  package) and <code>RforProteomics:::packageDF</code>
  (<code>data.frame</code> formatting).</p>
    
    </div>

    <pre class="usage"><span class='fu'>proteomicsPackages</span>(<span class='no'>biocv</span>, <span class='kw'>cache</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)

<span class='fu'>massSpectrometryPackages</span>(<span class='no'>biocv</span>, <span class='kw'>cache</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)

<span class='fu'>massSpectrometryDataPackages</span>(<span class='no'>biocv</span>, <span class='kw'>cache</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>biocv</th>
      <td><p>A <code>character</code> with the Bioconductor version to
search for relevant packages. If missing, the running version
is used.</p></td>
    </tr>
    <tr>
      <th>cache</th>
      <td><p>A <code>logical</code> indicating whether cached package
information should be used. Default is <code>FALSE</code>. All
except development versions are up-to-date.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>data.frame</code> with the respective package names,
    titles and versions.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>head</span>(<span class='no'>pp</span> <span class='kw'>&lt;-</span> <span class='fu'>proteomicsPackages</span>(<span class='st'>"3.0"</span>))</div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>CopyNumberVariants</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>ConnectTools, HighThroughputSequencing</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencing</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Cancer</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Statistics, Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>MultipleComparisons</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>CopyNumber</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics, Enrichment</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>SNPsAndGeneticVariability</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Clustering.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencing</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics, Cancer, FlowCytData, StemCells, DensityGating</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Gating</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>GraphsAndNetworks</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencing</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics, MultipleComparisons</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Cancer</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics, Statistics, MultipleComparisons</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>AMDIS, GCMS</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>GraphsAndNetwork</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencingData</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>GraphsAndNetwork</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>GO </span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Infastructure</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NoViewProvided</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HTA2.0</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics, Visualisation</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencing</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NetworkAnalysis, GraphsAndNetworks</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>NetworkAnalysis</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Cancer, StemCells, HIV</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Statistics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>Bioinformatics</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Dropping unknown biocViews terms:</span>
#&gt; <span class='warning'>HighThroughputSequencing</span></div><div class='output co'>#&gt;             Package
#&gt; ASEB           ASEB
#&gt; bioassayR bioassayR
#&gt; BRAIN         BRAIN
#&gt; CellNOptR CellNOptR
#&gt; ChemmineR ChemmineR
#&gt; cisPath     cisPath
#&gt;                                                                                                                   Title
#&gt; ASEB                                                                                    Predict Acetylated Lysine Sites
#&gt; bioassayR                                                                            R library for Bioactivity analysis
#&gt; BRAIN                                                Baffling Recursive Algorithm for Isotope distributioN calculations
#&gt; CellNOptR Training of boolean logic models of signalling networks using prior knowledge networks and perturbation data.
#&gt; ChemmineR                                                                                 Cheminformatics Toolkit for R
#&gt; cisPath                                       Visualization and management of the protein-protein interaction networks.
#&gt;           Version
#&gt; ASEB       1.10.0
#&gt; bioassayR   1.4.3
#&gt; BRAIN      1.12.0
#&gt; CellNOptR  1.12.0
#&gt; ChemmineR  2.18.1
#&gt; cisPath     1.6.4</div><div class='input'><span class='no'>ppc</span> <span class='kw'>&lt;-</span> <span class='fu'>proteomicsPackages</span>(<span class='st'>"3.0"</span>, <span class='kw'>cache</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='fu'>all.equal</span>(<span class='no'>pp</span>, <span class='no'>ppc</span>)</div><div class='output co'>#&gt; [1] "Component “Version”: 18 string mismatches"</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Laurent Gatto

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

